---
title: "synthesis"
author: "SPT"
date: "2021_11_02"
output: github_document
---

### Step 1: Load packages and data

```{r, echo = `FALSE`}
# Install packages
#devtools::install_github("MathiasHarrer/dmetar")
#devtools::install_github("daniel1noble/metaDigitise")
#install.packages("devtools")
#install.packages("cli")

# Load packages
library(tidyverse)
library(meta)
library(metafor)
library(esc)
library(gdata)
library(searcher)
library(metaDigitise)
#library(dmetar)

# Reset global environment
rm(list = ls())

# Change working directory
setwd("/Users/simontye/Research/Projects/Meta_Analysis/2021_MetaAnalysis")

# Load Web of Science data (WoS)
wos.comp <- read.csv(file = "data/papers/wos_comp.csv",  head = TRUE, sep = ",")
wos.pred <- read.csv(file = "data/papers/wos_pred.csv", head = TRUE, sep = ",")
wos.mut  <- read.csv(file = "data/papers/wos_mutual.csv", head = TRUE, sep = ",")
chamber  <- read.csv(file = "data/papers/chamberlain.csv", head = TRUE, sep = ",")

# Plot theme
pretty <- theme_classic(base_size = 10) +
  theme(axis.title.y      = element_text(margin = margin(r = 10)),
        axis.title.x      = element_text(margin = margin(t = 10)), 
        axis.text         = element_text(color = "black"), 
        axis.ticks        = element_line(color = "black"),
        axis.ticks.length = unit(.25, "cm"), 
        legend.key.size   = unit(1.5,  'lines'),
        legend.position   = "right")
```

```{r, echo = `FALSE`}
# Function to insert rows for papers with multiple effect sizes
add.rows <- function(data,
                     new.row,
                     r) {
  data[seq(r + 1, nrow(data) + 1),] <- data[seq(r, nrow(data)),]
  data[r, ] <- new.row
  data
}

# Function to include or exclude an article
ma.eval <- function(input,
                    article.id,
                    include,
                    comment) {
  # Add study details in effect size df
  input[article.id, c(8:9)] <- c(include, comment)
  # Choose output
  return(input)
}

# Function to calculate effect sizes
ma.es <- function(input,
                  output,
                  article.id,
                  study.data) {
  if(study.data["measure"] == "Mean" && study.data["direction"] == "-") {
  study.es <- esc_mean_sd(grp1m  = as.numeric(study.data[c("mean.e")]),
                          grp1sd = as.numeric(study.data[c("error.e")]),
                          grp1n  = as.numeric(study.data[c("n.e")]),
                          grp2m  = as.numeric(study.data[c("mean.c")]),
                          grp2sd = as.numeric(study.data[c("error.c")]),
                          grp2n  = as.numeric(study.data[c("n.c")]))    
  } 
  if(study.data["measure"] == "Mean" && study.data["direction"] == "+") {
  study.es <- esc_mean_sd(grp1m  = as.numeric(study.data[c("mean.c")]),
                          grp1sd = as.numeric(study.data[c("error.c")]),
                          grp1n  = as.numeric(study.data[c("n.c")]),
                          grp2m  = as.numeric(study.data[c("mean.e")]),
                          grp2sd = as.numeric(study.data[c("error.e")]),
                          grp2n  = as.numeric(study.data[c("n.e")]))
  }
  if(study.data["measure"] != "Mean") {
  study.es <- data.frame(es    = NA,
                         se    = NA,
                         var   = NA,
                         ci.lo = NA,
                         ci.hi = NA,
                         w     = NA)
  }
  # Add study details
  output <- add.rows(output, c(input[article.id, 1:8],
                               study.data[1:24],
                               as.numeric(study.es$es),
                               as.numeric(study.es$se),
                               as.numeric(study.es$var),
                               as.numeric(study.es$ci.lo),
                               as.numeric(study.es$ci.hi),
                               as.numeric(study.es$w)),
                     sum(nrow(output)) + 1)
  # Remove empty columns
  output <- filter(output, !is.na(id))
  # Choose output
  return(output)
}

# Function to search Google for paper title
ma.search <- function(input, article.id) {
  # Filter article
  tmp <- filter(input, id == as.numeric(article.id))
  # Search Google
  search_google(query = tmp$title, rlang = FALSE)
  # Choose output
  return(tmp)
}

# Function to monitor progress
ma.status <- function(wos, es) {
  # Number of articles
  articles <- as.numeric(nrow(wos))
  # Number of evaluated articles
  eval <- wos %>%
    filter(!is.na(include))
  eval <- as.numeric(nrow(eval))
  # Number of good articles
  good <- wos %>%
    filter(include == "yes" | include == "prob yes")
  good <- as.numeric(nrow(good))
  # Number of bad articles
  bad <- eval - good
  # Number of effect sizes
  es <- as.numeric(nrow(es))
  # Percent complete
  percent <- round((eval / articles * 100), 2)
  
  # Create output
  paper.count   <- paste0("1. Completed ", percent, "% (", eval, "/699) of article evaluations.")
  good.count    <- paste0("2. Found ", good, " articles that may have relevant information.")
  bad.count     <- paste0("3. Found ", bad, " articles that do not have relevant information.")
  #es.count      <- paste0("4. Calculated ", es, " effect size(s).")
  # Choose output
  return(cat(paper.count,
             good.count,
             bad.count,
             #es.count,
             sep = "\n"))
}

# Function to print or search abstract
ma.abstract <- function(wos, article.id) {
  # Print or search for abstract
  article <- ifelse(is.na(wos[article.id, "abstract"]),
                    ma.search(wos, article.id),
                    wos[article.id, "abstract"])
  # Choose output
  return(article)
}

# Plot theme
pretty <- theme_classic(base_size = 10) +
  theme(axis.title.y      = element_text(margin = margin(r = 10)),
        axis.title.x      = element_text(margin = margin(t = 10)), 
        axis.text         = element_text(color = "black"), 
        axis.ticks        = element_line(color = "black"),
        axis.ticks.length = unit(.25, "cm"), 
        legend.key.size   = unit(1.5,  'lines'),
        legend.position   = "right")
```

### Step 2: Format data

```{r echo = `FALSE`}
# Combine WoS data
wos <- wos.comp %>%
  rbind(., wos.pred, wos.mut) %>%
  rename_all(tolower) %>%
  dplyr::select(., c(publication.year, article.title, authors,
                     source.title, book.series.title, abstract)) %>%
  dplyr::rename(title   = article.title,
                journal = source.title,
                book    = book.series.title,
                year    = publication.year) %>%
  arrange(year, title) %>%
  mutate_all(na_if,"") %>%
  mutate(id      = row_number(),
         include = as.character(NA),
         comment = as.character(NA)) %>%
  dplyr::select(., c(id, year, title, authors,
                     journal, book, abstract,
                     include, comment))

# Format Chamberlain data
chamber <- chamber %>%
  rename_all(tolower) %>%
  dplyr::rename(year     = publication.year,
                authors  = author,
                journal  = publication.title,
                abstract = abstract.note) %>%
  dplyr::select(., c(year, title, authors, journal, abstract)) %>%
  mutate_all(na_if,"") %>%
  mutate(id      = row_number(),
         book    = as.character(NA),
         include = "yes",
         comment = as.character(NA)) %>%
  dplyr::select(., c(id, year, title, authors,
                     journal, book, abstract,
                     include, comment))

# Create empty df to store effect sizes
es <- data.frame(id            = as.character(NA),
                 year          = as.character(NA),
                 title         = as.character(NA),
                 authors       = as.character(NA),
                 journal       = as.character(NA),
                 book          = as.character(NA),
                 abstract      = as.character(NA),
                 include       = as.character(NA),
                 comment       = as.character(NA),
                 es.id         = as.character(NA),
                 focal.species = as.character(NA),
                 focal.n       = as.character(NA),
                 focal.density = as.character(NA),
                 focal.size    = as.character(NA),
                 mut.species   = as.character(NA),
                 mut.n         = as.character(NA),
                 mut.density   = as.character(NA),
                 mut.size      = as.character(NA),
                 comp.species  = as.character(NA), 
                 comp.n        = as.character(NA),
                 comp.density  = as.character(NA),
                 comp.size     = as.character(NA),
                 pred.species  = as.character(NA), 
                 pred.n        = as.character(NA),  
                 pred.density  = as.character(NA),
                 pred.size     = as.character(NA),
                 interaction   = as.character(NA),   
                 trait         = as.character(NA),         
                 treatment     = as.character(NA),
                 location      = as.character(NA),      
                 measure       = as.character(NA),       
                 direction     = as.character(NA),
                 es            = as.character(NA),
                 se            = as.character(NA),
                 var           = as.character(NA),
                 ci.lo         = as.character(NA),
                 ci.hi         = as.character(NA),
                 w             = as.character(NA))

# Remove objects
rm(wos.comp, wos.pred, wos.mut)
```

### Step 3: Evaluate articles

```{r, echo = `FALSE`}
# Evaluate articles
wos <- ma.eval(wos, 1,    "no", "economics")
wos <- ma.eval(wos, 2,    "no", "evo")
wos <- ma.eval(wos, 3,    "no", "agricultural model")
wos <- ma.eval(wos, 4,    "yes", "mut, field exp")
wos <- ma.eval(wos, 5,    "yes", "comp, field exp")
wos <- ma.eval(wos, 6,    "yes", "comp, resource pulse, field exp")
wos <- ma.eval(wos, 7,    "yes", "comp, field exp")
wos <- ma.eval(wos, 8,    "no", "economics")
wos <- ma.eval(wos, 9,    "prob no", "virus")
wos <- ma.eval(wos, 10,   "prob no", "intra feeding behavior")
wos <- ma.eval(wos, 11,   "yes", "pred, field exp")
wos <- ma.eval(wos, 12,   "no", "electronics")
wos <- ma.eval(wos, 13,   "no", "english")
wos <- ma.eval(wos, 14,   "prob yes", "comp, field exp")
wos <- ma.eval(wos, 15,   "prob yes", "comp, field exp")
wos <- ma.eval(wos, 16,   "prob yes", "comp, field exp")
wos <- ma.eval(wos, 17,   "maybe", "comp, co2, field exp")
wos <- ma.eval(wos, 18,   "prob yes", "mut, field exp")
wos <- ma.eval(wos, 19,   "maybe", "read")
wos <- ma.eval(wos, 20,   "yes", "field exp")
wos <- ma.eval(wos, 21,   "yes", "field exp")
wos <- ma.eval(wos, 22,   "yes", "field exp")
wos <- ma.eval(wos, 23,   "yes", "field exp")
wos <- ma.eval(wos, 24,   "no", "economics")
wos <- ma.eval(wos, 25,   "maybe", "read")
wos <- ma.eval(wos, 26,   "no", "obituary")
wos <- ma.eval(wos, 27,   "yes", "comp, field exp")
wos <- ma.eval(wos, 28,   "yes", "comp, field exp")
wos <- ma.eval(wos, 29,   "prob no", "obs")
wos <- ma.eval(wos, 30,   "prob yes", "removal, field exp")
wos <- ma.eval(wos, 31,   "no", "weird")
wos <- ma.eval(wos, 32,   "prob yes", "removal")
wos <- ma.eval(wos, 33,   "no", "human")
wos <- ma.eval(wos, 34,   "prob yes", "field exp")
wos <- ma.eval(wos, 35,   "prob yes", "field exp")
wos <- ma.eval(wos, 36,   "no", "human, model")
wos <- ma.eval(wos, 37,   "no", "model")
wos <- ma.eval(wos, 38,   "no", "chemistry")
wos <- ma.eval(wos, 39,   "yes", "plants")
wos <- ma.eval(wos, 40,   "yes", "plants")
wos <- ma.eval(wos, 41,   "no", "model")
wos <- ma.eval(wos, 42,   "prob yes", "field exp")
wos <- ma.eval(wos, 43,   "no", "economics")
wos <- ma.eval(wos, 44,   "no", "economics")
wos <- ma.eval(wos, 45,   "prob yes", "mut, field exp")
wos <- ma.eval(wos, 46,   "prob yes", "mut, field exp")
wos <- ma.eval(wos, 47,   "no", "electronics")
wos <- ma.eval(wos, 48,   "maybe", "read")
wos <- ma.eval(wos, 49,   "no", "economics")
wos <- ma.eval(wos, 50,   "no", "model")
wos <- ma.eval(wos, 51,   "maybe", "polyploid")
wos <- ma.eval(wos, 52,   "no", "human")
wos <- ma.eval(wos, 53,   "no", "economics")
wos <- ma.eval(wos, 54,   "no", "economics")
wos <- ma.eval(wos, 55,   "yes", "field exp")
wos <- ma.eval(wos, 56,   "no", "model")
wos <- ma.eval(wos, 57,   "yes", "field exp")
wos <- ma.eval(wos, 58,   "prob yes", "read")
wos <- ma.eval(wos, 59,   "maybe", "obs")
wos <- ma.eval(wos, 60,   "prob yes", "field exp")
wos <- ma.eval(wos, 61,   "yes", "field exp")
wos <- ma.eval(wos, 62,   "yes", "protein")
wos <- ma.eval(wos, 63,   "yes", "field exp")
wos <- ma.eval(wos, 64,   "prob yes", "removal, field exp")
wos <- ma.eval(wos, 65,   "prob yes", "field exp")
wos <- ma.eval(wos, 66,   "yes", "review, read")
wos <- ma.eval(wos, 67,   "no", "economics")
wos <- ma.eval(wos, 68,   "no", "humans")
wos <- ma.eval(wos, 69,   "no", "humans")
wos <- ma.eval(wos, 70,   "maybe", "read")
wos <- ma.eval(wos, 71,   "prob yes", "field exp")
wos <- ma.eval(wos, 72,   "no", "human")
wos <- ma.eval(wos, 73,   "prob no", "read, hermaphrodites")
wos <- ma.eval(wos, 74,   "prob no", "sperm comp")
wos <- ma.eval(wos, 75,   "prob yes", "comp, field exp")
wos <- ma.eval(wos, 76,   "no", "chemistry")
wos <- ma.eval(wos, 77,   "no", "chemistry")
wos <- ma.eval(wos, 78,   "no", "evolution")
wos <- ma.eval(wos, 79,   "yes", "field exp")
wos <- ma.eval(wos, 80,   "prob yes", "field exp")
wos <- ma.eval(wos, 81,   "prob yes", "read, old tree clearing")
wos <- ma.eval(wos, 82,   "prob yes", "field exp")
wos <- ma.eval(wos, 83,   "no", "human")
wos <- ma.eval(wos, 84,   "no", "what?")
wos <- ma.eval(wos, 85,   "maybe", "read")
wos <- ma.eval(wos, 86,   "prob yes", "field exp")
wos <- ma.eval(wos, 87,   "no", "human")
wos <- ma.eval(wos, 88,   "no", "human")
wos <- ma.eval(wos, 89,   "no", "economics")
wos <- ma.eval(wos, 90,   "no", "economics")
wos <- ma.eval(wos, 91,   "maybe", "cannibalism")
wos <- ma.eval(wos, 92,   "no", "economics")
wos <- ma.eval(wos, 93,   "prob yes", "plants, read")
wos <- ma.eval(wos, 94,   "no", "human")
wos <- ma.eval(wos, 95,   "prob no", "virus")
wos <- ma.eval(wos, 96,   "no", "human")
wos <- ma.eval(wos, 97,   "maybe", "pest species")
wos <- ma.eval(wos, 98,   "prob no", "obs")
wos <- ma.eval(wos, 99,   "prob yes", "field exp")
wos <- ma.eval(wos, 100,  "maybe", "pollen")
wos <- ma.eval(wos, 101,  "no", "economics")
wos <- ma.eval(wos, 102,  "no", "human")
wos <- ma.eval(wos, 103,  "maybe", "comp, field exp")
wos <- ma.eval(wos, 104,  "maybe", "polymorphism, intra")
wos <- ma.eval(wos, 105,  "yes", "comp, field exp")
wos <- ma.eval(wos, 106,  "no", "human")
wos <- ma.eval(wos, 107,  "prob yes", "fire exc, field exp")
wos <- ma.eval(wos, 108,  "maybe", "seems complex comm")
wos <- ma.eval(wos, 109,  "prob no", "self-compatbiilty")
wos <- ma.eval(wos, 110,  "maybe", "seems obs")
wos <- ma.eval(wos, 111,  "no", "human")
wos <- ma.eval(wos, 112,  "no", "human")
wos <- ma.eval(wos, 113,  "maybe", "seems complex comm")
wos <- ma.eval(wos, 114,  "prob yes", "comp, field exp")
wos <- ma.eval(wos, 115,  "prob yes", "comp, field exp")
wos <- ma.eval(wos, 116,  "prob yes", "mut, field exp")
wos <- ma.eval(wos, 117,  "prob yes", "mut, field exp")
wos <- ma.eval(wos, 118,  "no", "model")
wos <- ma.eval(wos, 119,  "prob yes", "comp, field exp")
wos <- ma.eval(wos, 120,  "no", "chemistry")
wos <- ma.eval(wos, 121,  "maybe", "neighbor removal")
wos <- ma.eval(wos, 122,  "no", "weird")
wos <- ma.eval(wos, 123,  "prob yes", "mut, field exp")
wos <- ma.eval(wos, 124,  "no", "electronics")
wos <- ma.eval(wos, 125,  "prob yes", "pred, field exp")
wos <- ma.eval(wos, 126,  "prob yes", "mut, field exp")
wos <- ma.eval(wos, 127,  "prob yes", "mut, field exp")
wos <- ma.eval(wos, 128,  "no", "model")
wos <- ma.eval(wos, 129,  "no", "model")
wos <- ma.eval(wos, 130,  "no", "sperm comp, review")
wos <- ma.eval(wos, 131,  "no", "economics")
wos <- ma.eval(wos, 132,  "maybe", "hermaphrodite")
wos <- ma.eval(wos, 133,  "prob yes", "mut, field exp")
wos <- ma.eval(wos, 134,  "prob yes", "comp, field exp")
wos <- ma.eval(wos, 135,  "prob yes", "comp, field exp")
wos <- ma.eval(wos, 136,  "prob no", "obs")
wos <- ma.eval(wos, 137,  "no", "weird")
wos <- ma.eval(wos, 138,  "no", "economics")
wos <- ma.eval(wos, 139,  "no", "economics")
wos <- ma.eval(wos, 140,  "prob yes", "comp, field exp")
wos <- ma.eval(wos, 141,  "prob yes", "comp, field exp")
wos <- ma.eval(wos, 142,  "prob yes", "comp, field exp")
wos <- ma.eval(wos, 143,  "no", "human, robot")
wos <- ma.eval(wos, 144,  "prob yes", "mut, field exp")
wos <- ma.eval(wos, 145,  "maybe", "self-comp")
wos <- ma.eval(wos, 146,  "no", "model")
wos <- ma.eval(wos, 147,  "prob yes", "mut, field exp")
wos <- ma.eval(wos, 148,  "no", "human")
wos <- ma.eval(wos, 149,  "no", "human")
wos <- ma.eval(wos, 150,  "prob yes", "pred, field exp")
wos <- ma.eval(wos, 151,  "maybe", "seems intra")
wos <- ma.eval(wos, 152,  "no", "sperm comp")
wos <- ma.eval(wos, 153,  "no", "soil comp for molecules")
wos <- ma.eval(wos, 154,  "prob yes", "mut, field exp")
wos <- ma.eval(wos, 155,  "prob yes", "mut, field exp")
wos <- ma.eval(wos, 156,  "prob no", "seems intra")
wos <- ma.eval(wos, 157,  "prob yes", "pred, field exp")
wos <- ma.eval(wos, 158,  "prob yes", "pred, field exp")
wos <- ma.eval(wos, 159,  "no", "economics")
wos <- ma.eval(wos, 160,  "maybe", "mixed tree stands")
wos <- ma.eval(wos, 161,  "maybe", "propaglue comp")
wos <- ma.eval(wos, 162,  "no", "model")
wos <- ma.eval(wos, 163,  "maybe", "read")
wos <- ma.eval(wos, 164,  "no", "sperm comp")
wos <- ma.eval(wos, 165,  "no", "method")
wos <- ma.eval(wos, 166,  "prob no", "hermaphrodite")
wos <- ma.eval(wos, 167,  "no", "volleyball?")
wos <- ma.eval(wos, 168,  "no", "economics")
wos <- ma.eval(wos, 169,  "no", "human")
wos <- ma.eval(wos, 170,  "prob yes", "field exp")
wos <- ma.eval(wos, 171,  "prob no", "intra, geno")
wos <- ma.eval(wos, 172,  "no", "economics")
wos <- ma.eval(wos, 173,  "prob yes", "mut, field exp")
wos <- ma.eval(wos, 174,  "yes", "comp, pred, field exp")
wos <- ma.eval(wos, 175,  "no", "economimcs")
wos <- ma.eval(wos, 176,  "yes", "comp, field exp")
wos <- ma.eval(wos, 177,  "maybe", "crossing, field exp")
wos <- ma.eval(wos, 178,  "no", "economimcs")
wos <- ma.eval(wos, 179,  "maybe", "read")
wos <- ma.eval(wos, 180,  "no", "chemistry")
wos <- ma.eval(wos, 181,  "prob yes", "comp, field exp")
wos <- ma.eval(wos, 182,  "no", "what?")
wos <- ma.eval(wos, 183,  "no", "economimcs")
wos <- ma.eval(wos, 184,  "maybe", "mut, field exp")
wos <- ma.eval(wos, 185,  "prob yes", "comp, field exp")
wos <- ma.eval(wos, 186,  "prob yes", "comp, field exp")
wos <- ma.eval(wos, 187,  "no", "economics")
wos <- ma.eval(wos, 188,  "no", "human")
wos <- ma.eval(wos, 189,  "yes", "comp, field exp")
wos <- ma.eval(wos, 190,  "yes", "comp, field exp")
wos <- ma.eval(wos, 191,  "yes", "comp, field exp")
wos <- ma.eval(wos, 192,  "maybe", "mut, comp, field exp")
wos <- ma.eval(wos, 193,  "prob yes", "pred, comp, field exp")
wos <- ma.eval(wos, 194,  "no", "economics")
wos <- ma.eval(wos, 195,  "no", "economics")
wos <- ma.eval(wos, 196,  "prob yes", "comp, field exp")
wos <- ma.eval(wos, 197,  "prob yes", "pred, field exp")
wos <- ma.eval(wos, 198,  "prob yes", "mut, field exp")
wos <- ma.eval(wos, 199,  "no", "human")
wos <- ma.eval(wos, 200,  "prob no", "intra sex")
wos <- ma.eval(wos, 201,  "prob yes", "pred, field exp")
wos <- ma.eval(wos, 202,  "prob no", "intra, field exp")
wos <- ma.eval(wos, 203,  "prob no", "intra, field exp")
wos <- ma.eval(wos, 204,  "prob yes", "comp, field exp")
wos <- ma.eval(wos, 205,  "prob no", "pollution, field exp")
wos <- ma.eval(wos, 206,  "prob no", "introduction, field exp")
wos <- ma.eval(wos, 207,  "no", "human")
wos <- ma.eval(wos, 208,  "prob yes", "comp, field exp")
wos <- ma.eval(wos, 209,  "prob yes", "pred, field exp")
wos <- ma.eval(wos, 210,  "no", "human")
wos <- ma.eval(wos, 211,  "prob yes", "pre, field exp")
wos <- ma.eval(wos, 212,  "prob no", "polymorph")
wos <- ma.eval(wos, 213,  "no", "chemistry")
wos <- ma.eval(wos, 214,  "maybe", "succession, field exp")
wos <- ma.eval(wos, 215,  "no", "human")
wos <- ma.eval(wos, 216,  "no", "economics")
wos <- ma.eval(wos, 217,  "no", "human")
wos <- ma.eval(wos, 218,  "maybe", "read, ma of bird food manipulation")
wos <- ma.eval(wos, 219,  "no", "human")
wos <- ma.eval(wos, 220,  "prob yes", "mut, field exp")
wos <- ma.eval(wos, 221,  "yes", "all, field exp")
wos <- ma.eval(wos, 222,  "no", "same as 221")
wos <- ma.eval(wos, 223,  "prob no", "intrasex, exp")
wos <- ma.eval(wos, 224,  "no", "review, econ")
wos <- ma.eval(wos, 225,  "yes", "pred, field exp")
wos <- ma.eval(wos, 226,  "yes", "comp, field exp")
wos <- ma.eval(wos, 227,  "yes", "comp, field exp")
wos <- ma.eval(wos, 228,  "no", "human")
wos <- ma.eval(wos, 229,  "no", "human")
wos <- ma.eval(wos, 230,  "no", "human")
wos <- ma.eval(wos, 231,  "maybe", "lab exp")
wos <- ma.eval(wos, 232,  "no", "human")
wos <- ma.eval(wos, 233,  "maybe", "seems intra")
wos <- ma.eval(wos, 234,  "no", "human")
wos <- ma.eval(wos, 235,  "prob yes", "pred, field exp")
wos <- ma.eval(wos, 236,  "no", "human")
wos <- ma.eval(wos, 237,  "no", "cellular")
wos <- ma.eval(wos, 238,  "prob yes", "pred, field exp")
wos <- ma.eval(wos, 239,  "no", "human")
wos <- ma.eval(wos, 240,  "yes", "pred, field exp")
wos <- ma.eval(wos, 241,  "no", "intra, field exp")
wos <- ma.eval(wos, 242,  "maybe", "lab exp")
wos <- ma.eval(wos, 243,  "no", "economics")
wos <- ma.eval(wos, 244,  "prob yes", "mut, field exp")
wos <- ma.eval(wos, 245,  "no", "human")
wos <- ma.eval(wos, 246,  "no", "economics")
wos <- ma.eval(wos, 247,  "no", "economics")
wos <- ma.eval(wos, 248,  "yes", "pred, field exp")
wos <- ma.eval(wos, 249,  "prob yes", "mut, field exp")
wos <- ma.eval(wos, 250,  "prob yes", "comp, field exp")
wos <- ma.eval(wos, 251,  "maybe", "obs, fire")
wos <- ma.eval(wos, 252,  "no", "human")
wos <- ma.eval(wos, 253,  "no", "theory")
wos <- ma.eval(wos, 254,  "no", "human")
wos <- ma.eval(wos, 255,  "prob yes", "mut, field exp")
wos <- ma.eval(wos, 256,  "no", "model")
wos <- ma.eval(wos, 257,  "no", "obs")
wos <- ma.eval(wos, 258,  "no", "physics")
wos <- ma.eval(wos, 259,  "yes", "pred, field exp")
wos <- ma.eval(wos, 260,  "maybe", "comp, field exp")
wos <- ma.eval(wos, 261,  "no", "human")
wos <- ma.eval(wos, 262,  "no", "model")
wos <- ma.eval(wos, 263,  "maybe", "seems complex comm")
wos <- ma.eval(wos, 264,  "no", "economics")
wos <- ma.eval(wos, 265,  "prob no", "obs")
wos <- ma.eval(wos, 266,  "no", "tranportation")
wos <- ma.eval(wos, 267,  "no", "chemistry")
wos <- ma.eval(wos, 268,  "prob no", "cultivar")
wos <- ma.eval(wos, 269,  "prob no", "intra")
wos <- ma.eval(wos, 270,  "prob yes", "mut, field exp")
wos <- ma.eval(wos, 271,  "prob yes", "mut, field exp")
wos <- ma.eval(wos, 272,  "prob yes", "mut, field exp")
wos <- ma.eval(wos, 273,  "prob yes", "all, field exp")
wos <- ma.eval(wos, 274,  "prob yes", "all, field exp")
wos <- ma.eval(wos, 275,  "no", "economics")
wos <- ma.eval(wos, 276,  "prob yes", "lab exp")
wos <- ma.eval(wos, 277,  "no", "review")
wos <- ma.eval(wos, 278,  "maybe", "read")
wos <- ma.eval(wos, 279,  "no", "obs")
wos <- ma.eval(wos, 280,  "maybe", "read, mangrove, nutrients")
wos <- ma.eval(wos, 281,  "yes", "comp, field exp")
wos <- ma.eval(wos, 282,  "no", "human")
wos <- ma.eval(wos, 283,  "yes", "comp")
wos <- ma.eval(wos, 284,  "no", "chemistry")
wos <- ma.eval(wos, 285,  "no", "human")
wos <- ma.eval(wos, 286,  "no", "economics")
wos <- ma.eval(wos, 287,  "yes", "comp, field exp")
wos <- ma.eval(wos, 288,  "no", "economics")
wos <- ma.eval(wos, 289,  "yes", "comp, field exp")
wos <- ma.eval(wos, 290,  "maybe", "mut, field exp")
wos <- ma.eval(wos, 291,  "no", "human")
wos <- ma.eval(wos, 292,  "no", "human")
wos <- ma.eval(wos, 293,  "no", "human")
wos <- ma.eval(wos, 294,  "prob yes", "comp, field exp")
wos <- ma.eval(wos, 295,  "prob yes", "comp, field exp")
wos <- ma.eval(wos, 296,  "maybe", "ploids")
wos <- ma.eval(wos, 297,  "prob yes", "mut, field exp")
wos <- ma.eval(wos, 298,  "prob yes", "lab exp")
wos <- ma.eval(wos, 299,  "no", "evo")
wos <- ma.eval(wos, 300,  "no", "evo")
wos <- ma.eval(wos, 301,  "prob no", "seems intra")
wos <- ma.eval(wos, 302,  "prob yes", "mut, field exp")
wos <- ma.eval(wos, 303,  "prob no", "seems intra")
wos <- ma.eval(wos, 304,  "prob yes", "mut, field exp")
wos <- ma.eval(wos, 305,  "no", "human")
wos <- ma.eval(wos, 306,  "no", "human")
wos <- ma.eval(wos, 307,  "no", "human")
wos <- ma.eval(wos, 308,  "no", "human")
wos <- ma.eval(wos, 309,  "maybe", "read")
wos <- ma.eval(wos, 310,  "no", "economics")
wos <- ma.eval(wos, 311,  "prob no", "seems intra")
wos <- ma.eval(wos, 312,  "prob yes", "comp, field exp")
wos <- ma.eval(wos, 313,  "prob yes", "comp, field exp")
wos <- ma.eval(wos, 314,  "yes", "field exp")
wos <- ma.eval(wos, 315,  "no", "human")
wos <- ma.eval(wos, 316,  "no", "human")
wos <- ma.eval(wos, 317,  "prob no", "seems intra")
wos <- ma.eval(wos, 318,  "prob no", "pred, complex comm")
wos <- ma.eval(wos, 319,  "prob yes", "comp, lab exp")
wos <- ma.eval(wos, 320,  "maybe", "comp, complex comm, field exp")
wos <- ma.eval(wos, 320,  "maybe", "comp, complex comm, field exp")
wos <- ma.eval(wos, 321,  "prob yes", "pred, field exp")
wos <- ma.eval(wos, 322,  "no", "economics")
wos <- ma.eval(wos, 323,  "no", "obs")
wos <- ma.eval(wos, 324,  "no", "obs")
wos <- ma.eval(wos, 325,  "no", "economics")
wos <- ma.eval(wos, 326,  "no", "human")
wos <- ma.eval(wos, 327,  "prob yes", "pred, field exp")
wos <- ma.eval(wos, 328,  "maybe", "read, review")
wos <- ma.eval(wos, 329,  "no", "policy")
wos <- ma.eval(wos, 330,  "prob yes", "pred, lab exp")
wos <- ma.eval(wos, 331,  "no", "intra, herbicide")
wos <- ma.eval(wos, 332,  "no", "policy")
wos <- ma.eval(wos, 333,  "no", "intra, field exp")
wos <- ma.eval(wos, 334,  "no", "complex comm, field exp")
wos <- ma.eval(wos, 335,  "prob yes", "inter, field exp")
wos <- ma.eval(wos, 336,  "no", "obs")
wos <- ma.eval(wos, 337,  "maybe", "inter, field exp")
wos <- ma.eval(wos, 338,  "yes", "inter, field exp")
wos <- ma.eval(wos, 339,  "no", "economics")
wos <- ma.eval(wos, 340,  "maybe", "reivew, ma on evo defense cost")
wos <- ma.eval(wos, 341,  "no", "human")
wos <- ma.eval(wos, 342,  "no", "chemistry")
wos <- ma.eval(wos, 343,  "no", "human")
wos <- ma.eval(wos, 344,  "no", "human")
wos <- ma.eval(wos, 345,  "no", "ecosystem")
wos <- ma.eval(wos, 346,  "prob yes", "mut, field exp")
wos <- ma.eval(wos, 347,  "prob yes", "mut, field exp")
wos <- ma.eval(wos, 348,  "no", "human")
wos <- ma.eval(wos, 349,  "no", "human")
wos <- ma.eval(wos, 350,  "no", "human")
wos <- ma.eval(wos, 351,  "maybe", "seems intra, song callback")
wos <- ma.eval(wos, 352,  "no", "human")
wos <- ma.eval(wos, 353,  "no", "chemistry")
wos <- ma.eval(wos, 354,  "no", "human")
wos <- ma.eval(wos, 355,  "no", "human")
wos <- ma.eval(wos, 356,  "prob yes", "inter, field exp")
wos <- ma.eval(wos, 357,  "no", "chemistry")
wos <- ma.eval(wos, 358,  "maybe", "review of invasive pollinators, i think")
wos <- ma.eval(wos, 359,  "no", "human")
wos <- ma.eval(wos, 360,  "yes", "inter, exp")
wos <- ma.eval(wos, 361,  "no", "human")
wos <- ma.eval(wos, 361,  "no", "human")
wos <- ma.eval(wos, 362,  "prob yes", "pred, field exp")
wos <- ma.eval(wos, 363,  "no", "human")
wos <- ma.eval(wos, 364,  "prob yes", "mut, field exp")
wos <- ma.eval(wos, 365,  "prob yes", "herb, field exp")
wos <- ma.eval(wos, 366,  "no", "human")
wos <- ma.eval(wos, 367,  "prob yes", "herb, field exp")
wos <- ma.eval(wos, 368,  "maybe", "viral, lab exp")
wos <- ma.eval(wos, 369,  "yes", "pred, field exp")
wos <- ma.eval(wos, 369,  "yes", "pred, field exp")
wos <- ma.eval(wos, 370,  "maybe", "inter, complex comm, field exp")
wos <- ma.eval(wos, 371,  "prob no", "intra, field exp")
wos <- ma.eval(wos, 372,  "no", "human")
wos <- ma.eval(wos, 373,  "no", "human")
wos <- ma.eval(wos, 374,  "prob yes", "inter, field exp")
wos <- ma.eval(wos, 375,  "prob no", "inter, complex comm, field exp")
wos <- ma.eval(wos, 376,  "prob no", "inter, complex comm, field exp")
wos <- ma.eval(wos, 377,  "no", "human")
wos <- ma.eval(wos, 378,  "no", "human")
wos <- ma.eval(wos, 379,  "prob yes", "inter, field exp")
wos <- ma.eval(wos, 380,  "prob yes", "ma of native/invasive pol, field exp")
wos <- ma.eval(wos, 381,  "no", "same as 380")
wos <- ma.eval(wos, 382,  "prob yes", "inter, field exp")
wos <- ma.eval(wos, 383,  "prob yes", "inter, field exp")
wos <- ma.eval(wos, 384,  "maybe", "parasitism, field exp")
wos <- ma.eval(wos, 385,  "prob yes", "inter, field exp")
wos <- ma.eval(wos, 386,  "maybe", "complex comm, field exp")
wos <- ma.eval(wos, 387,  "prob no", "intra, field exp")
wos <- ma.eval(wos, 388,  "no", "human")
wos <- ma.eval(wos, 389,  "no", "genes")
wos <- ma.eval(wos, 390,  "no", "cancer")
wos <- ma.eval(wos, 391,  "no", "teaching guide")
wos <- ma.eval(wos, 392,  "maybe", "read, ma of biological control")
wos <- ma.eval(wos, 393,  "no", "intra, field exp")
wos <- ma.eval(wos, 394,  "no", "genes")
wos <- ma.eval(wos, 395,  "no", "human")
wos <- ma.eval(wos, 396,  "no", "human")
wos <- ma.eval(wos, 397,  "prob yes", "mut, field exp")
wos <- ma.eval(wos, 398,  "prob yes", "mut, field exp")
wos <- ma.eval(wos, 399,  "no", "human")
wos <- ma.eval(wos, 400,  "no", "chemistry")
wos <- ma.eval(wos, 401,  "no", "human")
wos <- ma.eval(wos, 402,  "prob yes", "mut, field exp")
wos <- ma.eval(wos, 403,  "no", "human")
wos <- ma.eval(wos, 404,  "no", "human")
wos <- ma.eval(wos, 405,  "no", "human")
wos <- ma.eval(wos, 406,  "no", "human")
wos <- ma.eval(wos, 407,  "prob yes", "inter, field exp")
wos <- ma.eval(wos, 408,  "no", "economics")
wos <- ma.eval(wos, 409,  "no", "human")
wos <- ma.eval(wos, 410,  "maybe", "read, community phylogenetics")
wos <- ma.eval(wos, 411,  "no", "economics")
wos <- ma.eval(wos, 412,  "prob yes", "pred, field exp")
wos <- ma.eval(wos, 413,  "no", "human")
wos <- ma.eval(wos, 414,  "no", "human")
wos <- ma.eval(wos, 415,  "prob no", "intra, field exp")
wos <- ma.eval(wos, 416,  "no", "human")
wos <- ma.eval(wos, 417,  "no", "economics")
wos <- ma.eval(wos, 418,  "prob yes", "inter, field exp")
wos <- ma.eval(wos, 419,  "no", "models")
wos <- ma.eval(wos, 419,  "no", "models")
wos <- ma.eval(wos, 420,  "no", "models")
wos <- ma.eval(wos, 421,  "no", "economics")
wos <- ma.eval(wos, 422,  "prob yes", "inter, field exp")
wos <- ma.eval(wos, 423,  "prob yes", "inter, field exp")
wos <- ma.eval(wos, 424,  "prob yes", "inter, field exp")
wos <- ma.eval(wos, 425,  "no", "model")
wos <- ma.eval(wos, 426,  "no", "intra, field exp")
wos <- ma.eval(wos, 427,  "maybe", "inter, complex comm, field exp")
wos <- ma.eval(wos, 428,  "no", "human")
wos <- ma.eval(wos, 429,  "no", "pesticide")
wos <- ma.eval(wos, 430,  "prob yes", "mut, field exp")
wos <- ma.eval(wos, 431,  "prob yes", "pred, field exp")
wos <- ma.eval(wos, 432,  "prob no", "complex comm, field exp")
wos <- ma.eval(wos, 433,  "no", "intra, field exp")
wos <- ma.eval(wos, 434,  "prob yes", "mut, field exp")
wos <- ma.eval(wos, 435,  "no", "human")
wos <- ma.eval(wos, 436,  "maybe", "complex comm")
wos <- ma.eval(wos, 437,  "prob yes", "inter, field exp")
wos <- ma.eval(wos, 438,  "no", "human")
wos <- ma.eval(wos, 439,  "maybe", "seems intra")
wos <- ma.eval(wos, 440,  "prob yes", "mut and pred, field exp")
wos <- ma.eval(wos, 441,  "prob yes", "mut, field exp")
wos <- ma.eval(wos, 442,  "prob yes", "mut, field exp")
wos <- ma.eval(wos, 443,  "maybe", "parasitism, field exp")
wos <- ma.eval(wos, 444,  "no", "wildfire, field exp")
wos <- ma.eval(wos, 445,  "maybe", "mut, complex comm, field exp")
wos <- ma.eval(wos, 446,  "prob yes", "mut, field exp")
wos <- ma.eval(wos, 447,  "prob yes", "mut, field exp")
wos <- ma.eval(wos, 448,  "prob yes", "mut, field exp")
wos <- ma.eval(wos, 449,  "prob yes", "mut, field exp")
wos <- ma.eval(wos, 450,  "prob yes", "inter, field exp")
wos <- ma.eval(wos, 451,  "prob yes", "inter, field exp")
wos <- ma.eval(wos, 452,  "maybe", "inter, disturbance, field exp")
wos <- ma.eval(wos, 453,  "prob yes", "inter, field exp")
wos <- ma.eval(wos, 454,  "no", "obs")
wos <- ma.eval(wos, 455,  "prob no", "complex comm, model")
wos <- ma.eval(wos, 456,  "maybe", "fuel cell tech?")
wos <- ma.eval(wos, 457,  "no", "human")
wos <- ma.eval(wos, 458,  "maybe", "mut, field exp")
wos <- ma.eval(wos, 459,  "no", "human")
wos <- ma.eval(wos, 460,  "no", "model")
wos <- ma.eval(wos, 461,  "no", "review, ecosystem services")
wos <- ma.eval(wos, 462,  "no", "review, resurrection ecology")
wos <- ma.eval(wos, 463,  "prob no", "inter, obs")
wos <- ma.eval(wos, 464,  "no", "human")
wos <- ma.eval(wos, 465,  "prob no", "complex comm")
wos <- ma.eval(wos, 466,  "no", "codon bias")
wos <- ma.eval(wos, 467,  "no", "model")
wos <- ma.eval(wos, 468,  "prob yes", "inter, field exp")
wos <- ma.eval(wos, 469,  "no", "human")
wos <- ma.eval(wos, 470,  "no", "human")
wos <- ma.eval(wos, 471,  "no", "review")
wos <- ma.eval(wos, 472,  "yes", "mut, co2, field exp")
wos <- ma.eval(wos, 473,  "no", "review")
wos <- ma.eval(wos, 474,  "prob yes", "inter, field exp")
wos <- ma.eval(wos, 475,  "yes", "fungi ma")
wos <- ma.eval(wos, 476,  "no", "chemistry")
wos <- ma.eval(wos, 477,  "maybe", "seems obs")
wos <- ma.eval(wos, 478,  "maybe", "seems obs")
wos <- ma.eval(wos, 479,  "no", "human")
wos <- ma.eval(wos, 480,  "no", "biofuel")
wos <- ma.eval(wos, 481,  "maybe", "pred, complex comm, maybe on control")
wos <- ma.eval(wos, 482,  "no", "human")
wos <- ma.eval(wos, 483,  "maybe", "inter, complex comm, maybe pairwise")
wos <- ma.eval(wos, 484,  "maybe", "inter, complex comm, maybe pairwise")
wos <- ma.eval(wos, 485,  "prob yes", "pred, field exp")
wos <- ma.eval(wos, 486,  "no", "human")
wos <- ma.eval(wos, 487,  "prob yes", "inter, field exp")
wos <- ma.eval(wos, 488,  "no", "human")
wos <- ma.eval(wos, 489,  "no", "economics")
wos <- ma.eval(wos, 490,  "maybe", "types of comp")
wos <- ma.eval(wos, 491,  "prob yes", "mut, field exp")
wos <- ma.eval(wos, 492,  "maybe", "inter and intra, field exp")
wos <- ma.eval(wos, 493,  "prob yes", "inter, field exp")
wos <- ma.eval(wos, 494,  "maybe", "mut, complex comm, field exp")
wos <- ma.eval(wos, 495,  "no", "human")
wos <- ma.eval(wos, 496,  "maybe", "pred vocals, field exp")
wos <- ma.eval(wos, 497,  "no", "human")
wos <- ma.eval(wos, 498,  "maybe", "inter, lab exp")
wos <- ma.eval(wos, 499,  "prob yes", "inter, field exp")
wos <- ma.eval(wos, 500,  "no", "human")
wos <- ma.eval(wos, 501,  "no", "model")
wos <- ma.eval(wos, 502,  "no", "human")
wos <- ma.eval(wos, 503,  "prob yes", "inter, field exp")
wos <- ma.eval(wos, 504,  "maybe", "seems too complex")
wos <- ma.eval(wos, 505,  "prob no", "parasite, field exp")
wos <- ma.eval(wos, 506,  "no", "no control, field exp")
wos <- ma.eval(wos, 507,  "maybe", "lab exp")
wos <- ma.eval(wos, 508,  "no", "human")
wos <- ma.eval(wos, 509,  "no", "model")
wos <- ma.eval(wos, 510,  "no", "model")
wos <- ma.eval(wos, 511,  "maybe", "pred excl, complex comm")
wos <- ma.eval(wos, 512,  "prob no", "pred vocal, obs")
wos <- ma.eval(wos, 513,  "no", "obs")
wos <- ma.eval(wos, 514,  "maybe", "seems yes but complex comm and obs")
wos <- ma.eval(wos, 515,  "no", "what?")
wos <- ma.eval(wos, 516,  "no", "sperm?")
wos <- ma.eval(wos, 517,  "no", "human")
wos <- ma.eval(wos, 518,  "no", "human")
wos <- ma.eval(wos, 519,  "maybe", "seems yes but complex comm and obs")
wos <- ma.eval(wos, 520,  "yes", "pred, inter")
wos <- ma.eval(wos, 521,  "yes", "pred, inter")
wos <- ma.eval(wos, 522,  "no", "intra")
wos <- ma.eval(wos, 523,  "no", "self comp")
wos <- ma.eval(wos, 524,  "maybe", "prey choice")
wos <- ma.eval(wos, 525,  "maybe", "intra pred/prey")
wos <- ma.eval(wos, 526,  "no", "poll review")
wos <- ma.eval(wos, 527,  "no", "pred rem review")
wos <- ma.eval(wos, 528,  "no", "human")
wos <- ma.eval(wos, 529,  "no", "human")
wos <- ma.eval(wos, 530,  "no", "human")
wos <- ma.eval(wos, 531,  "no", "poll obs")
wos <- ma.eval(wos, 532,  "no", "theory")
wos <- ma.eval(wos, 533,  "no", "mut obs")
wos <- ma.eval(wos, 534,  "no", "politics")
wos <- ma.eval(wos, 535,  "no", "fungi crazy")
wos <- ma.eval(wos, 536,  "no", "human")
wos <- ma.eval(wos, 537,  "no", "human")
wos <- ma.eval(wos, 538,  "no", "human")
wos <- ma.eval(wos, 539,  "no", "human")
wos <- ma.eval(wos, 540,  "prob yes", "pred maybe no control, exp")
wos <- ma.eval(wos, 541,  "no", "model")
wos <- ma.eval(wos, 542,  "no", "human")
wos <- ma.eval(wos, 543,  "prob no", "fake pred")
wos <- ma.eval(wos, 544,  "no", "obs, growth rings")
wos <- ma.eval(wos, 545,  "prob no", "complex comm, colon")
wos <- ma.eval(wos, 546,  "prob no", "complex comm, mut")
wos <- ma.eval(wos, 547,  "no", "methods")
wos <- ma.eval(wos, 548,  "no", "eco-evo")
wos <- ma.eval(wos, 549,  "no", "complex comm")
wos <- ma.eval(wos, 550,  "prob yes", "pred, field exp")
wos <- ma.eval(wos, 551,  "no", "human")
wos <- ma.eval(wos, 552,  "maybe", "inter maybe no control")
wos <- ma.eval(wos, 553,  "maybe", "same as 552")
wos <- ma.eval(wos, 554,  "no", "clay models, cool")
wos <- ma.eval(wos, 555,  "no", "human")
wos <- ma.eval(wos, 556,  "prob no", "complex comm")
wos <- ma.eval(wos, 557,  "no", "human")
wos <- ma.eval(wos, 558,  "no", "human")
wos <- ma.eval(wos, 559,  "no", "poll obs, model")
wos <- ma.eval(wos, 560,  "no", "human")
wos <- ma.eval(wos, 561,  "no", "human")
wos <- ma.eval(wos, 562,  "no", "human")
wos <- ma.eval(wos, 563,  "no", "human")
wos <- ma.eval(wos, 564,  "no", "intra, mut")
wos <- ma.eval(wos, 565,  "no", "human")
wos <- ma.eval(wos, 566,  "no", "eco-evo")
wos <- ma.eval(wos, 567,  "no", "eco-evo")
wos <- ma.eval(wos, 568,  "no", "mut, complex comm")
wos <- ma.eval(wos, 569,  "no", "eco-evo")
wos <- ma.eval(wos, 570,  "no", "eco-evo")
wos <- ma.eval(wos, 571,  "no", "eco-evo")
wos <- ma.eval(wos, 572,  "no", "obs")
wos <- ma.eval(wos, 573,  "no", "obs")
wos <- ma.eval(wos, 574,  "maybe", "eco-evo")
wos <- ma.eval(wos, 575,  "no", "field exp, model")
wos <- ma.eval(wos, 576,  "no", "human")
wos <- ma.eval(wos, 577,  "prob no", "fert, mut")
wos <- ma.eval(wos, 578,  "no", "kang rat review")
wos <- ma.eval(wos, 579,  "no", "human")
wos <- ma.eval(wos, 580,  "no", "mut theory")
wos <- ma.eval(wos, 581,  "maybe", "playback stimuli, inter")
wos <- ma.eval(wos, 582,  "no", "genes")
wos <- ma.eval(wos, 583,  "no", "enrichment, complex comm")
wos <- ma.eval(wos, 584,  "no", "ma of transplants")
wos <- ma.eval(wos, 585,  "no", "human")
wos <- ma.eval(wos, 586,  "no", "human")
wos <- ma.eval(wos, 587,  "no", "human")
wos <- ma.eval(wos, 588,  "no", "eco-evo")
wos <- ma.eval(wos, 589,  "prob no", "complex comm, mut")
wos <- ma.eval(wos, 590,  "no", "intra")
wos <- ma.eval(wos, 591,  "prob yes", "pred chem cue")
wos <- ma.eval(wos, 592,  "no", "economics")
wos <- ma.eval(wos, 593,  "no", "obs, spider, light pollution")
wos <- ma.eval(wos, 594,  "no", "eco-evo")
wos <- ma.eval(wos, 595,  "no", "intra, genotypes")
wos <- ma.eval(wos, 596,  "no", "eco-evo")
wos <- ma.eval(wos, 597,  "no", "complex comm, model, field exp")
wos <- ma.eval(wos, 598,  "prob yes", "mut, inter, field exp")
wos <- ma.eval(wos, 599,  "no", "model")
wos <- ma.eval(wos, 600,  "no", "human")
wos <- ma.eval(wos, 601,  "no", "obs, complex comm")
wos <- ma.eval(wos, 602,  "no", "human")
wos <- ma.eval(wos, 603,  "prob yes", "pred, field exp")
wos <- ma.eval(wos, 604,  "no", "proteins")
wos <- ma.eval(wos, 605,  "prob yes", "complex comm but removal")
wos <- ma.eval(wos, 606,  "prob no", "enrichment")
wos <- ma.eval(wos, 607,  "prob yes", "inter, tree growth, warming")
wos <- ma.eval(wos, 608,  "prob yes", "long term field exp, all treatments")
wos <- ma.eval(wos, 609,  "no", "eco service review")
wos <- ma.eval(wos, 610,  "yes", "inter, plant, field exp")
wos <- ma.eval(wos, 611,  "maybe", "complex comm, pest species, field exp")
wos <- ma.eval(wos, 612,  "maybe", "prey choice, field exp")
wos <- ma.eval(wos, 613,  "prob yes", "inter fish cues, field exp")
wos <- ma.eval(wos, 614,  "no", "human")
wos <- ma.eval(wos, 615,  "maybe", "complex comm")
wos <- ma.eval(wos, 616,  "no", "chemistry")
wos <- ma.eval(wos, 617,  "no", "human")
wos <- ma.eval(wos, 618,  "no", "human")
wos <- ma.eval(wos, 619,  "prob no", "no control, isolates")
wos <- ma.eval(wos, 620,  "no", "plasticity")
wos <- ma.eval(wos, 621,  "no", "obs, poll, temp")
wos <- ma.eval(wos, 622,  "no", "plasticity")
wos <- ma.eval(wos, 623,  "no", "human")
wos <- ma.eval(wos, 624,  "no", "camera trap review")
wos <- ma.eval(wos, 625,  "no", "eco-evo")
wos <- ma.eval(wos, 626,  "maybe", "read, seems intra and clone/petri dish")
wos <- ma.eval(wos, 627,  "no", "theory")
wos <- ma.eval(wos, 628,  "no", "pollen comp")
wos <- ma.eval(wos, 629,  "prob no", "eco-evo, tri-trophic")
wos <- ma.eval(wos, 630,  "no", "chemistry")
wos <- ma.eval(wos, 631,  "no", "mut, disturb, enrich")
wos <- ma.eval(wos, 632,  "no", "genetics")
wos <- ma.eval(wos, 633,  "no", "sperm")
wos <- ma.eval(wos, 634,  "no", "genetics")
wos <- ma.eval(wos, 635,  "prob yes", "inter, pairwise")
wos <- ma.eval(wos, 636,  "no", "evo")
wos <- ma.eval(wos, 637,  "no", "human")
wos <- ma.eval(wos, 638,  "no", "economics")
wos <- ma.eval(wos, 639,  "no", "disturb, complex comm")
wos <- ma.eval(wos, 640,  "no", "eco-evo")
wos <- ma.eval(wos, 641,  "no", "economics")
wos <- ma.eval(wos, 642,  "no", "model")
wos <- ma.eval(wos, 643,  "no", "eco-evo")
wos <- ma.eval(wos, 644,  "prob no", "pred transplant, no control")
wos <- ma.eval(wos, 645,  "no", "obs")
wos <- ma.eval(wos, 646,  "no", "economics")
wos <- ma.eval(wos, 647,  "no", "obs")
wos <- ma.eval(wos, 648,  "no", "chemistry")
wos <- ma.eval(wos, 649,  "no", "obs")
wos <- ma.eval(wos, 650,  "prob no", "strains of amf")
wos <- ma.eval(wos, 651,  "no", "obs")
wos <- ma.eval(wos, 652,  "no", "economics")
wos <- ma.eval(wos, 653,  "prob yes", "read, tri-trophic, maybe no control")
wos <- ma.eval(wos, 654,  "prob no", "comp removal, owl, prob no but read")
wos <- ma.eval(wos, 655,  "maybe", "complex comm, inter")
wos <- ma.eval(wos, 656,  "prob yes", "crops w/ or w/o amf")
wos <- ma.eval(wos, 657,  "no", "human")
wos <- ma.eval(wos, 658,  "maybe", "mut, amf, maybe no control, shade/light")
wos <- ma.eval(wos, 659,  "no", "obs")
wos <- ma.eval(wos, 660,  "no", "intra, sex comp exp")
wos <- ma.eval(wos, 661,  "no", "human")
wos <- ma.eval(wos, 662,  "maybe", "am vs em fungi, plants")
wos <- ma.eval(wos, 663,  "prob yes", "invas, comp, field exp")
wos <- ma.eval(wos, 664,  "prob no", "invas, comp, complex comm, field exp")
wos <- ma.eval(wos, 665,  "no", "economics")
wos <- ma.eval(wos, 666,  "maybe", "inter, vocal, field exp")
wos <- ma.eval(wos, 667,  "no", "strains, enrichment")
wos <- ma.eval(wos, 668,  "prob yes", "pred, comp, field exp")
wos <- ma.eval(wos, 669,  "no", "economics")
wos <- ma.eval(wos, 670,  "no", "enrich, obs")
wos <- ma.eval(wos, 671,  "prob no", "flower manip, poll")
wos <- ma.eval(wos, 672,  "no", "speciation")
wos <- ma.eval(wos, 673,  "no", "same as 672")
wos <- ma.eval(wos, 674,  "no", "human")
wos <- ma.eval(wos, 675,  "no", "amf, emf, complex comm")
wos <- ma.eval(wos, 676,  "no", "economics")
wos <- ma.eval(wos, 677,  "prob yes", "mut, removal")
wos <- ma.eval(wos, 678,  "no", "economics")
wos <- ma.eval(wos, 679,  "maybe", "lat grad pred, prob no but cool")
wos <- ma.eval(wos, 680,  "maybe", "pred trait, exp")
wos <- ma.eval(wos, 681,  "no", "economics")
wos <- ma.eval(wos, 682,  "no", "economics")
wos <- ma.eval(wos, 683,  "no", "economics")
wos <- ma.eval(wos, 684,  "no", "complex comm")
wos <- ma.eval(wos, 685,  "no", "human")
wos <- ma.eval(wos, 686,  "no", "poll obs")
wos <- ma.eval(wos, 687,  "prob no", "colon, complex comm")
wos <- ma.eval(wos, 688,  "no", "human")
wos <- ma.eval(wos, 689,  "no", "human")
wos <- ma.eval(wos, 690,  "no", "comm composition, obs")
wos <- ma.eval(wos, 691,  "prob yes", "inter, field exp")
wos <- ma.eval(wos, 692,  "no", "human")
wos <- ma.eval(wos, 693,  "prob yes", "inter, field exp")
wos <- ma.eval(wos, 694,  "prob yes", "pred excl, field exp")
wos <- ma.eval(wos, 695,  "prob yes", "inter vocal, field exp")
wos <- ma.eval(wos, 696,  "prob yes", "pred pres, maybe abs, field exp")
wos <- ma.eval(wos, 697,  "no", "economics")
wos <- ma.eval(wos, 698,  "no", "intra, polymorph")
wos <- ma.eval(wos, 699,  "maybe", "fungicide, inter")
```

```{r, echo = `FALSE`}
# Changed opinions
wos <- ma.eval(wos, 5,    "no", "obs")
wos <- ma.eval(wos, 6,    "prob no", "complex comm")
wos <- ma.eval(wos, 7,    "prob no", "no pred control")
wos <- ma.eval(wos, 20,   "no", "intra, no pred control")
wos <- ma.eval(wos, 21,   "maybe", "read")
wos <- ma.eval(wos, 23,   "no", "intra")
wos <- ma.eval(wos, 28,   "no", "nematicide")
wos <- ma.eval(wos, 40,   "prob no", "no comp control")
wos <- ma.eval(wos, 55,   "maybe", "foreign lang")
wos <- ma.eval(wos, 57,   "prob no", "complex comm")
wos <- ma.eval(wos, 61,   "prob no", "no error")
wos <- ma.eval(wos, 62,   "no", "proteins")
wos <- ma.eval(wos, 63,   "maybe", "obs")
wos <- ma.eval(wos, 66,   "maybe", "review")
wos <- ma.eval(wos, 175,  "prob no", "no error")
wos <- ma.eval(wos, 221,  "maybe", "temporal patterns")
wos <- ma.eval(wos, 227,  "prob no", "complex comm")
wos <- ma.eval(wos, 240,  "prob no", "evo")
wos <- ma.eval(wos, 248,  "prob yes", "evo")
wos <- ma.eval(wos, 281,  "prob yes", "read, review")
wos <- ma.eval(wos, 283,  "maybe", "read")
wos <- ma.eval(wos, 287,  "maybe", "read")
wos <- ma.eval(wos, 289,  "maybe", "read")
wos <- ma.eval(wos, 11,   "prob not", "complex comm, excl exp")
wos <- ma.eval(wos, 22,   "prob not", "eco-evo, tri-trophic")
wos <- ma.eval(wos, 27,   "prob not", "complex comm")
wos <- ma.eval(wos, 176,  "prob not", "inter, complex comm")
wos <- ma.eval(wos, 189,  "prob not", "inter, complex comm")
wos <- ma.eval(wos, 225,  "prob not", "ungulate exlc, complex comm")
wos <- ma.eval(wos, 521,  "no", "plasticity")
wos <- ma.eval(wos, 610,  "no", "methods")
```

### Step 4: Extract data from text

```{r, echo = `FALSE`}
# Set order of evaluations
wos$include <- factor(as.factor(wos$include),
                      levels = c("yes", "prob yes","maybe", "prob no", "no"))

# Reorder articles based on evaluations
wos <- wos %>%
  .[order(.$include),]

###########################################
# Priority #1: "yes" and "prob yes"
wos.yes <- wos %>%
  filter(include == "yes" | include == "prob yes")
```

```{r, echo = `FALSE`}
###########################################
# Schadler2010
# Extract
# Fig. 1

###########################################
# Vogt2010
# Extract
# Fig3a, Fig3b
# Log ratio response and raw response, have raw response in Fig3 extracted

###########################################
# Grey2011
# Extract
# Fig5, maybe pop growth rates, con + hetero or none though

###########################################
# Santos2012
# Extracted Fig2a, Fig2b

###########################################
# Marino2013
# Extracted Fig2a, Fig2b

###########################################
# Nilsson2014
# Check about transformations, weird stats presented

###########################################
# Shah2014
# Check about transformations for Relative Interaction Intensity for Figs 1 and 2

###########################################
# Bontrager2014
# Extracted fig 1 and 5, think about including time series

###########################################
# Martinez2015
# Extract figure 2

###########################################
# Kenison2016
# Extract figure 2, think about time series

###########################################
# Parepa2016
# Missing stats, just model, it seems

###########################################
# Suraci2017
# Maybe fig 2

###########################################
# Nobbs2017
# Can prob extract figs

###########################################
# Dong2018
# Check, MA of CO2 with some AMF

###########################################
# Ramirez2018
# Check, MA of AMF

###########################################
# Welk2019
# Extract from fig 1 

###########################################
# Welk2019
# Extract from fig 1 

###########################################

#wos.yes[17, c("title", "year", "authors", "id")]


#
#colnames(wos.yes)
#str(es)
#ma.es

## Create empty df to store effect sizes
#es <- data.frame(id            = as.character(NA),
#                 year          = as.character(NA),
#                 title         = as.character(NA),
#                 authors       = as.character(NA),
#                 journal       = as.character(NA),
#                 book          = as.character(NA),
#                 abstract      = as.character(NA),
#                 include       = as.character(NA),
#                 comment       = as.character(NA),
#                 es.id         = as.character(NA),
#                 focal.species = as.character(NA),
#                 focal.n       = as.character(NA),
#                 focal.density = as.character(NA),
#                 focal.size    = as.character(NA),
#                 mut.species   = as.character(NA),
#                 mut.n         = as.character(NA),
#                 mut.density   = as.character(NA),
#                 mut.size      = as.character(NA),
#                 comp.species  = as.character(NA), 
#                 comp.n        = as.character(NA),
#                 comp.density  = as.character(NA),
#                 comp.size     = as.character(NA),
#                 pred.species  = as.character(NA), 
#                 pred.n        = as.character(NA),  
#                 pred.density  = as.character(NA),
#                 pred.size     = as.character(NA),
#                 interaction   = as.character(NA),   
#                 trait         = as.character(NA),         
#                 treatment     = as.character(NA),
#                 location      = as.character(NA),      
#                 measure       = as.character(NA),       
#                 direction     = as.character(NA),
#                 es            = as.character(NA),
#                 se            = as.character(NA),
#                 var           = as.character(NA),
#                 ci.lo         = as.character(NA),
#                 ci.hi         = as.character(NA),
#                 w             = as.character(NA))

```

### Step 5: Extract data from figures

```{r, echo = FALSE}
# Save file names
filenames <- list.files("/Users/simontye/Research/Projects/Meta_Analysis/2021_MetaAnalysis/papers/figs_new")

# Exclude metaDigitise folder
filenames <- filenames[!filenames %in% grep(paste0("caldat"),
                                            filenames,
                                            value = T)]

# File names and article ids
filenames <- data.frame(filename   = as.factor(all_of(filenames)),
                        article.id = c(
                          # Aristov2015
                          #c(rep(235, 1)),
                          # Barton2014
                          #c(rep(193, 1)),
                          # Bontrager2014
                          c(rep(226, 5)),
                          # Burkle2015
                          c(rep(272, 3)),
                          # Cipriotti2010
                          c(rep(35, 2)),
                          # Corli2019
                          c(rep(499, 2)),
                          # Gibb2011a
                          #c(rep(64, 1)),
                          # Gibb2011b
                          #c(rep(65, 1)),
                          # Gonzalez2018
                          #c(rep(440, 3)),
                          # Grass2018
                          c(rep(442, 3)),
                          # Grey2011
                          #c(rep(XXX, XXX)),
                          # Gundale2012
                          c(rep(115, 5)),
                          # Guo2012
                          c(rep(114, 17)),  
                          # He2021
                          #c(rep(653, 4)),
                          # Jackson2016
                          c(rep(295, 1)),
                          # Kenison2016,
                          c(rep(314, 6)),
                          # Leps2018
                          c(rep(487, 1)),
                          # Marino2013
                          c(rep(174, 2)),
                          # Martinez2015
                          c(rep(259, 3)),
                          # Nilsson2014
                          #c(rep(190, 1)),
                          # Overly2015
                          c(rep(271, 4)),
                          # Partzsch2017
                          c(rep(468, 2)),
                          # Petipas2014
                          c(rep(225, 1)),
                          # Pizaco2011
                          c(rep(46, 1)),
                          # Raak2017
                          c(rep(450, 2)),
                          # Santos2012
                          c(rep(105, 2)),
                          # Schadler2010
                          c(rep(4, 1)),
                          # Schulze2012
                          c(rep(99, 2)),
                          # Segre2013
                          c(rep(189, 4)),
                          # Shah2014
                          #c(rep(191, 4)),
                          # Trekels2013
                          c(rep(157, 3)),
                          # Tremlett2019
                          c(rep(598, 1)),
                          # VanDerNiet2015
                          c(rep(249, 1)),
                          # Vogt2010
                          c(rep(39, 2)),
                          # Welk2019
                          c(rep(520, 8))))

####################### Complete to Overby2015

# Extract figure data
fig.data <- metaDigitise("/Users/simontye/Research/Projects/Meta_Analysis/2021_MetaAnalysis/papers/figs_new")

# Options to continue to data entry
2
1

# Join WoS and figure data
fig.data <- fig.data %>%
  full_join(., filenames, by = c("filename")) %>%
  mutate(group_id = as.character(group_id))

# Split identifying information into separate columns
fig.detail <- strsplit(fig.data$group_id,',', fixed = FALSE)

# Convert list to df
fig.detail <- fig.detail %>%
  do.call(rbind.data.frame, .) %>%
  # Remove trailing/leading spaces
  mutate(across(where(is.character), str_trim))

# Rename details columns (group_#)
for(i in 1:ncol(fig.detail)){
  # Isolate column
  colnames(fig.detail)[i] <- c(paste0("group.", i))
}

# Add details
fig.data <- fig.data %>%
  cbind(., fig.detail) %>%
  mutate(group_id = NULL)

# Remove details
rm(filenames, fig.detail, i)
```

# Bontrager 2014

```{r, echo = `FALSE`}
# Bontrager2014_Fig1.png
fig.data.1 <- fig.data %>%
  filter(filename == "Bontrager2014_Fig1.png") %>%
  dplyr::rowwise() %>%
  mutate(
    ### Control
    # n
    n.c    = ifelse(group.1 == "Oenanthe" && group.2 == "cleared", n, NA),
    n.c    = ifelse(group.1 == "Salix"    && group.2 == "cleared", n, n.c),
    n.c    = ifelse(group.1 == "Scirpus"  && group.2 == "cleared", n, n.c),
    # mean
    mean.c = ifelse(group.1 == "Oenanthe" && group.2 == "cleared", mean, NA),
    mean.c = ifelse(group.1 == "Salix"    && group.2 == "cleared", mean, mean.c),
    mean.c = ifelse(group.1 == "Scirpus"  && group.2 == "cleared", mean, mean.c),
    # sd
    sd.c   = ifelse(group.1 == "Oenanthe" && group.2 == "cleared", sd, NA),
    sd.c   = ifelse(group.1 == "Salix"    && group.2 == "cleared", sd, sd.c),
    sd.c   = ifelse(group.1 == "Scirpus"  && group.2 == "cleared", sd, sd.c),
    ### Experimental
    # n
    n.e    = ifelse(group.1 == "Oenanthe" && group.2 == "not cleared", n, NA),
    n.e    = ifelse(group.1 == "Salix"    && group.2 == "not cleared", n, n.e),
    n.e    = ifelse(group.1 == "Scirpus"  && group.2 == "not cleared", n, n.e),
    # mean
    mean.e = ifelse(group.1 == "Oenanthe" && group.2 == "not cleared", mean, NA),
    mean.e = ifelse(group.1 == "Salix"    && group.2 == "not cleared", mean, mean.e),
    mean.e = ifelse(group.1 == "Scirpus"  && group.2 == "not cleared", mean, mean.e),
    # sd
    sd.e   = ifelse(group.1 == "Oenanthe" && group.2 == "not cleared", sd, NA),
    sd.e   = ifelse(group.1 == "Salix"    && group.2 == "not cleared", sd, sd.e),
    sd.e   = ifelse(group.1 == "Scirpus"  && group.2 == "not cleared", sd, sd.e)) %>%
  mutate(
    # Remove old stats columns
    mean       = NULL,
    sd         = NULL,
    n          = NULL,
    r          = NULL,
    se         = NULL,
    error_type = NULL,
    plot_type  = NULL) %>%
  # Merge control and experimental rows
  group_by(filename, variable, article.id, group.1) %>%
  dplyr::summarize(n.c     = max(n.c, na.rm = TRUE),
                   mean.c  = max(mean.c, na.rm = TRUE),
                   sd.c    = max(sd.c, na.rm = TRUE),
                   n.e     = max(n.e, na.rm = TRUE),
                   mean.e  = max(mean.e, na.rm = TRUE),
                   sd.e    = max(sd.e, na.rm = TRUE),
                   group.2 = "cleared",
                   group.3 = as.character(NA),
                   group.4 = as.character(NA)) %>%
  # Replace any missing values (Inf) with NAs
  mutate_at(., vars (-group_cols()), function(x) ifelse(is.infinite(x), NA, x))

# Bontrager2014_Fig5a.png
fig.data.2 <- fig.data %>%
  filter(filename == "Bontrager2014_Fig5a.png") %>%
  dplyr::rowwise() %>%
  mutate(
    ### Control
    # n
    n.c    = ifelse(group.1 == "Oenanthe" && group.2 == "cleared", n, NA),
    n.c    = ifelse(group.1 == "Salix"    && group.2 == "cleared", n, n.c),
    # mean
    mean.c = ifelse(group.1 == "Oenanthe" && group.2 == "cleared", mean, NA),
    mean.c = ifelse(group.1 == "Salix"    && group.2 == "cleared", mean, mean.c),
    # sd
    sd.c = ifelse(group.1 == "Oenanthe" && group.2 == "cleared", sd, NA),
    sd.c = ifelse(group.1 == "Salix"    && group.2 == "cleared", sd, sd.c),
    ### Experimental
    # n
    n.e    = ifelse(group.1 == "Oenanthe" && group.2 == "not cleared", n, NA),
    n.e    = ifelse(group.1 == "Salix"    && group.2 == "not cleared", n, n.e),
    # mean
    mean.e = ifelse(group.1 == "Oenanthe" && group.2 == "not cleared", mean, NA),
    mean.e = ifelse(group.1 == "Salix"    && group.2 == "not cleared", mean, mean.e),
    # sd
    sd.e   = ifelse(group.1 == "Oenanthe" && group.2 == "not cleared", sd, NA),
    sd.e   = ifelse(group.1 == "Salix"    && group.2 == "not cleared", sd, sd.e)) %>%
  mutate(
    # Remove old stats columns
    mean       = NULL,
    sd         = NULL,
    n          = NULL,
    r          = NULL,
    se         = NULL,
    error_type = NULL,
    plot_type  = NULL) %>%
  # Merge control and experimental rows
  group_by(filename, variable, article.id, group.1) %>%
  dplyr::summarize(n.c     = max(n.c, na.rm = TRUE),
                   mean.c  = max(mean.c, na.rm = TRUE),
                   sd.c    = max(sd.c, na.rm = TRUE),
                   n.e     = max(n.e, na.rm = TRUE),
                   mean.e  = max(mean.e, na.rm = TRUE),
                   sd.e    = max(sd.e, na.rm = TRUE),
                   group.2 = "cleared",
                   group.3 = as.character(NA),
                   group.4 = as.character(NA)) %>%
  # Replace any missing values (Inf) with NAs
  mutate_at(., vars (-group_cols()), function(x) ifelse(is.infinite(x), NA, x))

# Bontrager2014_Fig5b.png
fig.data.3 <- fig.data %>%
  filter(filename == "Bontrager2014_Fig5b.png") %>%
  dplyr::rowwise() %>%
  mutate(
    ### Control
    # n
    n.c    = ifelse(group.1 == "Oenanthe" && group.2 == "cleared", n, NA),
    n.c    = ifelse(group.1 == "Salix"    && group.2 == "cleared", n, n.c),
    n.c    = ifelse(group.1 == "Scirpus"  && group.2 == "cleared", n, n.c),
    # mean
    mean.c = ifelse(group.1 == "Oenanthe" && group.2 == "cleared", mean, NA),
    mean.c = ifelse(group.1 == "Salix"    && group.2 == "cleared", mean, mean.c),
    mean.c = ifelse(group.1 == "Scirpus"  && group.2 == "cleared", mean, mean.c),
    # sd
    sd.c   = ifelse(group.1 == "Oenanthe" && group.2 == "cleared", sd, NA),
    sd.c   = ifelse(group.1 == "Salix"    && group.2 == "cleared", sd, sd.c),
    sd.c   = ifelse(group.1 == "Scirpus"  && group.2 == "cleared", sd, sd.c),
    ### Experimental
    # n
    n.e    = ifelse(group.1 == "Oenanthe" && group.2 == "not cleared", n, NA),
    n.e    = ifelse(group.1 == "Salix"    && group.2 == "not cleared", n, n.e),
    n.e    = ifelse(group.1 == "Scirpus"  && group.2 == "not cleared", n, n.e),
    # mean
    mean.e = ifelse(group.1 == "Oenanthe" && group.2 == "not cleared", mean, NA),
    mean.e = ifelse(group.1 == "Salix"    && group.2 == "not cleared", mean, mean.e),
    mean.e = ifelse(group.1 == "Scirpus"  && group.2 == "not cleared", mean, mean.e),
    # sd
    sd.e   = ifelse(group.1 == "Oenanthe" && group.2 == "not cleared", sd, NA),
    sd.e   = ifelse(group.1 == "Salix"    && group.2 == "not cleared", sd, sd.e),
    sd.e   = ifelse(group.1 == "Scirpus"  && group.2 == "not cleared", sd, sd.e)) %>%
  mutate(
    # Remove old stats columns
    mean       = NULL,
    sd         = NULL,
    n          = NULL,
    r          = NULL,
    se         = NULL,
    error_type = NULL,
    plot_type  = NULL) %>%
  # Merge control and experimental rows
  group_by(filename, variable, article.id, group.1) %>%
  dplyr::summarize(n.c     = max(n.c, na.rm = TRUE),
                   mean.c  = max(mean.c, na.rm = TRUE),
                   sd.c    = max(sd.c, na.rm = TRUE),
                   n.e     = max(n.e, na.rm = TRUE),
                   mean.e  = max(mean.e, na.rm = TRUE),
                   sd.e    = max(sd.e, na.rm = TRUE),
                   group.2 = "cleared",
                   group.3 = as.character(NA),
                   group.4 = as.character(NA)) %>%
  # Replace any missing values (Inf) with NAs
  mutate_at(., vars (-group_cols()), function(x) ifelse(is.infinite(x), NA, x))

# Bontrager2014_Fig5c.png
fig.data.4 <- fig.data %>%
  filter(filename == "Bontrager2014_Fig5c.png") %>%
  dplyr::rowwise() %>%
  mutate(
    ### Control
    # n
    n.c    = ifelse(group.1 == "Oenanthe" && group.2 == "cleared", n, NA),
    # mean
    mean.c = ifelse(group.1 == "Oenanthe" && group.2 == "cleared", mean, NA),
    # sd
    sd.c   = ifelse(group.1 == "Oenanthe" && group.2 == "cleared", sd, NA),
    ### Experimental
    # n
    n.e    = ifelse(group.1 == "Oenanthe" && group.2 == "not cleared", n, NA),
    # mean
    mean.e = ifelse(group.1 == "Oenanthe" && group.2 == "not cleared", mean, NA),
    # sd
    sd.e   = ifelse(group.1 == "Oenanthe" && group.2 == "not cleared", sd, NA)) %>%
  mutate(
    # Remove old stats columns
    mean       = NULL,
    sd         = NULL,
    n          = NULL,
    r          = NULL,
    se         = NULL,
    error_type = NULL,
    plot_type  = NULL) %>%
  # Merge control and experimental rows
  group_by(filename, variable, article.id, group.1) %>%
  dplyr::summarize(n.c     = max(n.c, na.rm = TRUE),
                   mean.c  = max(mean.c, na.rm = TRUE),
                   sd.c    = max(sd.c, na.rm = TRUE),
                   n.e     = max(n.e, na.rm = TRUE),
                   mean.e  = max(mean.e, na.rm = TRUE),
                   sd.e    = max(sd.e, na.rm = TRUE),
                   group.2 = "cleared",
                   group.3 = as.character(NA),
                   group.4 = as.character(NA)) %>%
  # Replace any missing values (Inf) with NAs
  mutate_at(., vars (-group_cols()), function(x) ifelse(is.infinite(x), NA, x))

# Bontrager2014_Fig5d.png
fig.data.5 <- fig.data %>%
  filter(filename == "Bontrager2014_Fig5d.png") %>%
  dplyr::rowwise() %>%
  mutate(
    ### Control
    # n
    n.c    = ifelse(group.1 == "Oenanthe" && group.2 == "cleared", n, NA),
    # mean
    mean.c = ifelse(group.1 == "Oenanthe" && group.2 == "cleared", mean, NA),
    # sd
    sd.c   = ifelse(group.1 == "Oenanthe" && group.2 == "cleared", sd, NA),
    ### Experimental
    # n
    n.e    = ifelse(group.1 == "Oenanthe" && group.2 == "not cleared", n, NA),
    # mean
    mean.e = ifelse(group.1 == "Oenanthe" && group.2 == "not cleared", mean, NA),
    # sd
    sd.e   = ifelse(group.1 == "Oenanthe" && group.2 == "not cleared", sd, NA)) %>%
  mutate(
    # Remove old stats columns
    mean       = NULL,
    sd         = NULL,
    n          = NULL,
    r          = NULL,
    se         = NULL,
    error_type = NULL,
    plot_type  = NULL) %>%
  # Merge control and experimental rows
  group_by(filename, variable, article.id, group.1) %>%
  dplyr::summarize(n.c     = max(n.c, na.rm = TRUE),
                   mean.c  = max(mean.c, na.rm = TRUE),
                   sd.c    = max(sd.c, na.rm = TRUE),
                   n.e     = max(n.e, na.rm = TRUE),
                   mean.e  = max(mean.e, na.rm = TRUE),
                   sd.e    = max(sd.e, na.rm = TRUE),
                   group.2 = "cleared",
                   group.3 = as.character(NA),
                   group.4 = as.character(NA)) %>%
  # Replace any missing values (Inf) with NAs
  mutate_at(., vars (-group_cols()), function(x) ifelse(is.infinite(x), NA, x))

# Combine fig.data
working.fig.data <- fig.data.1 %>%
  full_join(., fig.data.2) %>%
  full_join(., fig.data.3) %>%
  full_join(., fig.data.4) %>%
  full_join(., fig.data.5)

# Remove objects
rm(fig.data.1, fig.data.2, fig.data.3,
   fig.data.4, fig.data.5)
```

# Burkle 2015

```{r, echo = `FALSE`}
# Burkle2015_Fig1a.png
fig.data.1 <- fig.data %>%
  filter(filename == "Burkle2015_Fig1a.png") %>%
  dplyr::rowwise() %>%
  mutate(
    ### Control
    # n
    n.c    = ifelse(group.1 == "Linum lewisii"      && group.2 == "w/o myco", n, NA),
    n.c    = ifelse(group.1 == "Bromus marginatus"  && group.2 == "w/o myco", n, n.c),
    n.c    = ifelse(group.1 == "Dactylis glomerata" && group.2 == "w/o myco", n, n.c),
    # mean
    mean.c = ifelse(group.1 == "Linum lewisii"      && group.2 == "w/o myco", mean, NA),
    mean.c = ifelse(group.1 == "Bromus marginatus"  && group.2 == "w/o myco", mean, mean.c),
    mean.c = ifelse(group.1 == "Dactylis glomerata" && group.2 == "w/o myco", mean, mean.c),
    # sd
    sd.c   = ifelse(group.1 == "Linum lewisii"      && group.2 == "w/o myco", sd, NA),
    sd.c   = ifelse(group.1 == "Bromus marginatus"  && group.2 == "w/o myco", sd, sd.c),
    sd.c   = ifelse(group.1 == "Dactylis glomerata" && group.2 == "w/o myco", sd, sd.c),
    ### Experimental
    # n
    n.e    = ifelse(group.1 == "Linum lewisii"      && group.2 == "w/ myco", n, NA),
    n.e    = ifelse(group.1 == "Bromus marginatus"  && group.2 == "w/ myco", n, n.e),
    n.e    = ifelse(group.1 == "Dactylis glomerata" && group.2 == "w/ myco", n, n.e),
    # mean
    mean.e = ifelse(group.1 == "Linum lewisii"      && group.2 == "w/ myco", mean, NA),
    mean.e = ifelse(group.1 == "Bromus marginatus"  && group.2 == "w/ myco", mean, mean.e),
    mean.e = ifelse(group.1 == "Dactylis glomerata" && group.2 == "w/ myco", mean, mean.e),
    # sd
    sd.e   = ifelse(group.1 == "Linum lewisii"      && group.2 == "w/ myco", sd, NA),
    sd.e   = ifelse(group.1 == "Bromus marginatus"  && group.2 == "w/ myco", sd, sd.e),
    sd.e   = ifelse(group.1 == "Dactylis glomerata" && group.2 == "w/ myco", sd, sd.e)) %>%
  mutate(
    # Remove old stats columns
    mean       = NULL,
    sd         = NULL,
    n          = NULL,
    r          = NULL,
    se         = NULL,
    error_type = NULL,
    plot_type  = NULL) %>%
  # Merge control and experimental rows
  group_by(filename, variable, article.id, group.1) %>%
  dplyr::summarize(n.c     = max(n.c, na.rm = TRUE),
                   mean.c  = max(mean.c, na.rm = TRUE),
                   sd.c    = max(sd.c, na.rm = TRUE),
                   n.e     = max(n.e, na.rm = TRUE),
                   mean.e  = max(mean.e, na.rm = TRUE),
                   sd.e    = max(sd.e, na.rm = TRUE),
                   group.2 = "myco",
                   group.3 = as.character(NA),
                   group.4 = as.character(NA)) %>%
  # Replace any missing values (Inf) with NAs
  mutate_at(., vars (-group_cols()), function(x) ifelse(is.infinite(x), NA, x))

# Burkle2015_Fig1b.png
fig.data.2 <- fig.data %>%
  filter(filename == "Burkle2015_Fig1b.png") %>%
  dplyr::rowwise() %>%
  mutate(
    ### Control
    # n
    n.c    = ifelse(group.1 == "Linum lewisii"      && group.2 == "w/o myco", n, NA),
    n.c    = ifelse(group.1 == "B marginatus"       && group.2 == "w/o myco", n, n.c),
    n.c    = ifelse(group.1 == "Dactylis glomerata" && group.2 == "w/o myco", n, n.c),
    # mean
    mean.c = ifelse(group.1 == "Linum lewisii"      && group.2 == "w/o myco", mean, NA),
    mean.c = ifelse(group.1 == "B marginatus"       && group.2 == "w/o myco", mean, mean.c),
    mean.c = ifelse(group.1 == "Dactylis glomerata" && group.2 == "w/o myco", mean, mean.c),
    # sd
    sd.c   = ifelse(group.1 == "Linum lewisii"      && group.2 == "w/o myco", sd, NA),
    sd.c   = ifelse(group.1 == "B marginatus"       && group.2 == "w/o myco", sd, sd.c),
    sd.c   = ifelse(group.1 == "Dactylis glomerata" && group.2 == "w/o myco", sd, sd.c),
    ### Experimental
    # n
    n.e    = ifelse(group.1 == "Linum lewisii"      && group.2 == "w/ myco", n, NA),
    n.e    = ifelse(group.1 == "B marginatus"       && group.2 == "w/ myco", n, n.e),
    n.e    = ifelse(group.1 == "Dactylis glomerata" && group.2 == "w/ myco", n, n.e),
    # mean
    mean.e = ifelse(group.1 == "Linum lewisii"      && group.2 == "w/ myco", mean, NA),
    mean.e = ifelse(group.1 == "B marginatus"       && group.2 == "w/ myco", mean, mean.e),
    mean.e = ifelse(group.1 == "Dactylis glomerata" && group.2 == "w/ myco", mean, mean.e),
    # sd
    sd.e   = ifelse(group.1 == "Linum lewisii"      && group.2 == "w/ myco", sd, NA),
    sd.e   = ifelse(group.1 == "B marginatus"       && group.2 == "w/ myco", sd, sd.e),
    sd.e   = ifelse(group.1 == "Dactylis glomerata" && group.2 == "w/ myco", sd, sd.e)) %>%
  # Fix species name
  mutate(group.1 = ifelse(group.1 == "B marginatus", "Bromus marginatus", group.1)) %>%
  mutate(
    # Remove old stats columns
    mean       = NULL,
    sd         = NULL,
    n          = NULL,
    r          = NULL,
    se         = NULL,
    error_type = NULL,
    plot_type  = NULL) %>%
  # Merge control and experimental rows
  group_by(filename, variable, article.id, group.1) %>%
  dplyr::summarize(n.c     = max(n.c, na.rm = TRUE),
                   mean.c  = max(mean.c, na.rm = TRUE),
                   sd.c    = max(sd.c, na.rm = TRUE),
                   n.e     = max(n.e, na.rm = TRUE),
                   mean.e  = max(mean.e, na.rm = TRUE),
                   sd.e    = max(sd.e, na.rm = TRUE),
                   group.2 = "myco",
                   group.3 = as.character(NA),
                   group.4 = as.character(NA)) %>%
  # Replace any missing values (Inf) with NAs
  mutate_at(., vars (-group_cols()), function(x) ifelse(is.infinite(x), NA, x))

# Burkle2015_Fig1c.png
fig.data.3 <- fig.data %>%
  filter(filename == "Burkle2015_Fig1c.png") %>%
  dplyr::rowwise() %>%
  mutate(
    ### Control
    # n
    n.c    = ifelse(group.1 == "Linum lewisii"      && group.2 == "w/o myco", n, NA),
    n.c    = ifelse(group.1 == "Bromus marginatus"  && group.2 == "w/o myco", n, n.c),
    n.c    = ifelse(group.1 == "Dactylis glomerata" && group.2 == "w/o myco", n, n.c),
    # mean
    mean.c = ifelse(group.1 == "Linum lewisii"      && group.2 == "w/o myco", mean, NA),
    mean.c = ifelse(group.1 == "Bromus marginatus"  && group.2 == "w/o myco", mean, mean.c),
    mean.c = ifelse(group.1 == "Dactylis glomerata" && group.2 == "w/o myco", mean, mean.c),
    # sd
    sd.c   = ifelse(group.1 == "Linum lewisii"      && group.2 == "w/o myco", sd, NA),
    sd.c   = ifelse(group.1 == "Bromus marginatus"  && group.2 == "w/o myco", sd, sd.c),
    sd.c   = ifelse(group.1 == "Dactylis glomerata" && group.2 == "w/o myco", sd, sd.c),
    ### Experimental
    # n
    n.e    = ifelse(group.1 == "Linum lewisii"      && group.2 == "w/ myco", n, NA),
    n.e    = ifelse(group.1 == "Bromus marginatus"  && group.2 == "w/ myco", n, n.e),
    n.e    = ifelse(group.1 == "Dactylis glomerata" && group.2 == "w/ myco", n, n.e),
    # mean
    mean.e = ifelse(group.1 == "Linum lewisii"      && group.2 == "w/ myco", mean, NA),
    mean.e = ifelse(group.1 == "Bromus marginatus"  && group.2 == "w/ myco", mean, mean.e),
    mean.e = ifelse(group.1 == "Dactylis glomerata" && group.2 == "w/ myco", mean, mean.e),
    # sd
    sd.e   = ifelse(group.1 == "Linum lewisii"      && group.2 == "w/ myco", sd, NA),
    sd.e   = ifelse(group.1 == "Bromus marginatus"  && group.2 == "w/ myco", sd, sd.e),
    sd.e   = ifelse(group.1 == "Dactylis glomerata" && group.2 == "w/ myco", sd, sd.e)) %>%
  # Fix species name
  mutate(group.1 = ifelse(group.1 == "B marginatus", "Bromus marginatus", group.1)) %>%
  mutate(
    # Remove old stats columns
    mean       = NULL,
    sd         = NULL,
    n          = NULL,
    r          = NULL,
    se         = NULL,
    error_type = NULL,
    plot_type  = NULL) %>%
  # Merge control and experimental rows
  group_by(filename, variable, article.id, group.1) %>%
  dplyr::summarize(n.c     = max(n.c, na.rm = TRUE),
                   mean.c  = max(mean.c, na.rm = TRUE),
                   sd.c    = max(sd.c, na.rm = TRUE),
                   n.e     = max(n.e, na.rm = TRUE),
                   mean.e  = max(mean.e, na.rm = TRUE),
                   sd.e    = max(sd.e, na.rm = TRUE),
                   group.2 = "myco",
                   group.3 = as.character(NA),
                   group.4 = as.character(NA)) %>%
  # Replace any missing values (Inf) with NAs
  mutate_at(., vars (-group_cols()), function(x) ifelse(is.infinite(x), NA, x))

# Combine fig.data
working.fig.data <- working.fig.data %>%
  full_join(., fig.data.1) %>%
  full_join(., fig.data.2) %>%
  full_join(., fig.data.3)

# Remove objects
rm(fig.data.1, fig.data.2, fig.data.3)
```

# Cipriotti 2010

```{r, echo = `FALSE`}
# Cipriotti2010_Fig1.png
fig.data.1 <- fig.data %>%
  filter(filename == "Cipriotti2010_Fig1.png") %>%
  dplyr::rowwise() %>%
  mutate(
    ### Control
    # n
    n.c    = ifelse(group.1 == "Mulinum spinosum"     && group.2 == "grass removal", n, NA),
    n.c    = ifelse(group.1 == "Senecio filaginoides" && group.2 == "grass removal", n, n.c),
    n.c    = ifelse(group.1 == "Adesmia volckmanni"   && group.2 == "grass removal", n, n.c),
    # mean
    mean.c = ifelse(group.1 == "Mulinum spinosum"     && group.2 == "grass removal", mean, NA),
    mean.c = ifelse(group.1 == "Senecio filaginoides" && group.2 == "grass removal", mean, mean.c),
    mean.c = ifelse(group.1 == "Adesmia volckmanni"   && group.2 == "grass removal", mean, mean.c),
    # sd
    sd.c   = ifelse(group.1 == "Mulinum spinosum"     && group.2 == "grass removal", sd, NA),
    sd.c   = ifelse(group.1 == "Senecio filaginoides" && group.2 == "grass removal", sd, sd.c),
    sd.c   = ifelse(group.1 == "Adesmia volckmanni"   && group.2 == "grass removal", sd, sd.c),
    ### Experimental
    # n
    n.e    = ifelse(group.1 == "Mulinum spinosum"     && group.2 == "control", n, NA),
    n.e    = ifelse(group.1 == "Senecio filaginoides" && group.2 == "control", n, n.e),
    n.e    = ifelse(group.1 == "Adesmia volckmanni"   && group.2 == "control", n, n.e),
    # mean
    mean.e = ifelse(group.1 == "Mulinum spinosum"     && group.2 == "control", mean, NA),
    mean.e = ifelse(group.1 == "Senecio filaginoides" && group.2 == "control", mean, mean.e),
    mean.e = ifelse(group.1 == "Adesmia volckmanni"   && group.2 == "control", mean, mean.e),
    # sd
    sd.e   = ifelse(group.1 == "Mulinum spinosum"     && group.2 == "control", sd, NA),
    sd.e   = ifelse(group.1 == "Senecio filaginoides" && group.2 == "control", sd, sd.e),
    sd.e   = ifelse(group.1 == "Adesmia volckmanni"   && group.2 == "control", sd, sd.e)) %>%
  mutate(
    # Remove old stats columns
    mean       = NULL,
    sd         = NULL,
    n          = NULL,
    r          = NULL,
    se         = NULL,
    error_type = NULL,
    plot_type  = NULL) %>%
  # Merge control and experimental rows
  group_by(filename, variable, article.id, group.1) %>%
  dplyr::summarize(n.c     = max(n.c, na.rm = TRUE),
                   mean.c  = max(mean.c, na.rm = TRUE),
                   sd.c    = max(sd.c, na.rm = TRUE),
                   n.e     = max(n.e, na.rm = TRUE),
                   mean.e  = max(mean.e, na.rm = TRUE),
                   sd.e    = max(sd.e, na.rm = TRUE),
                   group.2 = "grass",
                   group.3 = as.character(NA),
                   group.4 = as.character(NA)) %>%
  # Replace any missing values (Inf) with NAs
  mutate_at(., vars (-group_cols()), function(x) ifelse(is.infinite(x), NA, x))

# Cipriotti2010_Fig1.png
fig.data.2 <- fig.data %>%
  filter(filename == "Cipriotti2010_Fig1.png") %>%
  dplyr::rowwise() %>%
  mutate(
    ### Control
    # n
    n.c    = ifelse(group.1 == "Mulinum spinosum"     && group.2 == "shrub removal", n, NA),
    n.c    = ifelse(group.1 == "Senecio filaginoides" && group.2 == "shrub removal", n, n.c),
    n.c    = ifelse(group.1 == "Adesmia volckmanni"   && group.2 == "shrub removal", n, n.c),
    # mean
    mean.c = ifelse(group.1 == "Mulinum spinosum"     && group.2 == "shrub removal", mean, NA),
    mean.c = ifelse(group.1 == "Senecio filaginoides" && group.2 == "shrub removal", mean, mean.c),
    mean.c = ifelse(group.1 == "Adesmia volckmanni"   && group.2 == "shrub removal", mean, mean.c),
    # sd
    sd.c   = ifelse(group.1 == "Mulinum spinosum"     && group.2 == "shrub removal", sd, NA),
    sd.c   = ifelse(group.1 == "Senecio filaginoides" && group.2 == "shrub removal", sd, sd.c),
    sd.c   = ifelse(group.1 == "Adesmia volckmanni"   && group.2 == "shrub removal", sd, sd.c),
    ### Experimental
    # n
    n.e    = ifelse(group.1 == "Mulinum spinosum"     && group.2 == "control", n, NA),
    n.e    = ifelse(group.1 == "Senecio filaginoides" && group.2 == "control", n, n.e),
    n.e    = ifelse(group.1 == "Adesmia volckmanni"   && group.2 == "control", n, n.e),
    # mean
    mean.e = ifelse(group.1 == "Mulinum spinosum"     && group.2 == "control", mean, NA),
    mean.e = ifelse(group.1 == "Senecio filaginoides" && group.2 == "control", mean, mean.e),
    mean.e = ifelse(group.1 == "Adesmia volckmanni"   && group.2 == "control", mean, mean.e),
    # sd
    sd.e   = ifelse(group.1 == "Mulinum spinosum"     && group.2 == "control", sd, NA),
    sd.e   = ifelse(group.1 == "Senecio filaginoides" && group.2 == "control", sd, sd.e),
    sd.e   = ifelse(group.1 == "Adesmia volckmanni"   && group.2 == "control", sd, sd.e)) %>%
  mutate(
    # Remove old stats columns
    mean       = NULL,
    sd         = NULL,
    n          = NULL,
    r          = NULL,
    se         = NULL,
    error_type = NULL,
    plot_type  = NULL) %>%
  # Merge control and experimental rows
  group_by(filename, variable, article.id, group.1) %>%
  dplyr::summarize(n.c     = max(n.c, na.rm = TRUE),
                   mean.c  = max(mean.c, na.rm = TRUE),
                   sd.c    = max(sd.c, na.rm = TRUE),
                   n.e     = max(n.e, na.rm = TRUE),
                   mean.e  = max(mean.e, na.rm = TRUE),
                   sd.e    = max(sd.e, na.rm = TRUE),
                   group.2 = "shrub",
                   group.3 = as.character(NA),
                   group.4 = as.character(NA)) %>%
  # Replace any missing values (Inf) with NAs
  mutate_at(., vars (-group_cols()), function(x) ifelse(is.infinite(x), NA, x))

# Cipriotti2010_Fig3.png
fig.data.3 <- fig.data %>%
  filter(filename == "Cipriotti2010_Fig3.png") %>%
  dplyr::rowwise() %>%
  mutate(
    ### Control
    # n
    n.c    = ifelse(group.1 == "Mulinum spinosum"     && group.2 == "grass removal", n, NA),
    n.c    = ifelse(group.1 == "Senecio" && group.3 == "grass removal", n, n.c),
    n.c    = ifelse(group.1 == "Adesmia volckmanni"   && group.2 == "grass removal", n, n.c),
    # mean
    mean.c = ifelse(group.1 == "Mulinum spinosum"     && group.2 == "grass removal", mean, NA),
    mean.c = ifelse(group.1 == "Senecio" && group.3 == "grass removal", mean, mean.c),
    mean.c = ifelse(group.1 == "Adesmia volckmanni"   && group.2 == "grass removal", mean, mean.c),
    # sd
    sd.c   = ifelse(group.1 == "Mulinum spinosum"     && group.2 == "grass removal", sd, NA),
    sd.c   = ifelse(group.1 == "Senecio" && group.3 == "grass removal", sd, sd.c),
    sd.c   = ifelse(group.1 == "Adesmia volckmanni"   && group.2 == "grass removal", sd, sd.c),
    ### Experimental
    # n
    n.e    = ifelse(group.1 == "Mulinum spinosum"     && group.2 == "control", n, NA),
    n.e    = ifelse(group.1 == "Senecio" && group.3 == "control", n, n.e),
    n.e    = ifelse(group.1 == "Adesmia volckmanni"   && group.2 == "control", n, n.e),
    # mean
    mean.e = ifelse(group.1 == "Mulinum spinosum"     && group.2 == "control", mean, NA),
    mean.e = ifelse(group.1 == "Senecio" && group.3 == "control", mean, mean.e),
    mean.e = ifelse(group.1 == "Adesmia volckmanni"   && group.2 == "control", mean, mean.e),
    # sd
    sd.e   = ifelse(group.1 == "Mulinum spinosum"     && group.2 == "control", sd, NA),
    sd.e   = ifelse(group.1 == "Senecio" && group.3 == "control", sd, sd.e),
    sd.e   = ifelse(group.1 == "Adesmia volckmanni"   && group.2 == "control", sd, sd.e)) %>%
  mutate(group.1 = ifelse(group.1 == "Senecio", "Senecio filaginoides", group.1)) %>%
  mutate(
    # Remove old stats columns
    mean       = NULL,
    sd         = NULL,
    n          = NULL,
    r          = NULL,
    se         = NULL,
    error_type = NULL,
    plot_type  = NULL) %>%
  # Merge control and experimental rows
  group_by(filename, variable, article.id, group.1) %>%
  dplyr::summarize(n.c     = max(n.c, na.rm = TRUE),
                   mean.c  = max(mean.c, na.rm = TRUE),
                   sd.c    = max(sd.c, na.rm = TRUE),
                   n.e     = max(n.e, na.rm = TRUE),
                   mean.e  = max(mean.e, na.rm = TRUE),
                   sd.e    = max(sd.e, na.rm = TRUE),
                   group.2 = "grass",
                   group.3 = as.character(NA),
                   group.4 = as.character(NA)) %>%
  # Replace any missing values (Inf) with NAs
  mutate_at(., vars (-group_cols()), function(x) ifelse(is.infinite(x), NA, x))

# Cipriotti2010_Fig3.png
fig.data.4 <- fig.data %>%
  filter(filename == "Cipriotti2010_Fig3.png") %>%
  dplyr::rowwise() %>%
  mutate(
    ### Control
    # n
    n.c    = ifelse(group.1 == "Mulinum spinosum"     && group.2 == "shrub removal", n, NA),
    n.c    = ifelse(group.1 == "Senecio" && group.3 == "shrub removal", n, n.c),
    n.c    = ifelse(group.1 == "Adesmia volckmanni"   && group.2 == "shrub removal", n, n.c),
    # mean
    mean.c = ifelse(group.1 == "Mulinum spinosum"     && group.2 == "shrub removal", mean, NA),
    mean.c = ifelse(group.1 == "Senecio" && group.3 == "shrub removal", mean, mean.c),
    mean.c = ifelse(group.1 == "Adesmia volckmanni"   && group.2 == "shrub removal", mean, mean.c),
    # sd
    sd.c   = ifelse(group.1 == "Mulinum spinosum"     && group.2 == "shrub removal", sd, NA),
    sd.c   = ifelse(group.1 == "Senecio" && group.3 == "shrub removal", sd, sd.c),
    sd.c   = ifelse(group.1 == "Adesmia volckmanni"   && group.2 == "shrub removal", sd, sd.c),
    ### Experimental
    # n
    n.e    = ifelse(group.1 == "Mulinum spinosum"     && group.2 == "control", n, NA),
    n.e    = ifelse(group.1 == "Senecio" && group.3 == "control", n, n.e),
    n.e    = ifelse(group.1 == "Adesmia volckmanni"   && group.2 == "control", n, n.e),
    # mean
    mean.e = ifelse(group.1 == "Mulinum spinosum"     && group.2 == "control", mean, NA),
    mean.e = ifelse(group.1 == "Senecio" && group.3 == "control", mean, mean.e),
    mean.e = ifelse(group.1 == "Adesmia volckmanni"   && group.2 == "control", mean, mean.e),
    # sd
    sd.e   = ifelse(group.1 == "Mulinum spinosum"     && group.2 == "control", sd, NA),
    sd.e   = ifelse(group.1 == "Senecio" && group.3 == "control", sd, sd.e),
    sd.e   = ifelse(group.1 == "Adesmia volckmanni"   && group.2 == "control", sd, sd.e)) %>%
  mutate(group.1 = ifelse(group.1 == "Senecio", "Senecio filaginoides", group.1)) %>%
  mutate(
    # Remove old stats columns
    mean       = NULL,
    sd         = NULL,
    n          = NULL,
    r          = NULL,
    se         = NULL,
    error_type = NULL,
    plot_type  = NULL) %>%
  # Merge control and experimental rows
  group_by(filename, variable, article.id, group.1) %>%
  dplyr::summarize(n.c     = max(n.c, na.rm = TRUE),
                   mean.c  = max(mean.c, na.rm = TRUE),
                   sd.c    = max(sd.c, na.rm = TRUE),
                   n.e     = max(n.e, na.rm = TRUE),
                   mean.e  = max(mean.e, na.rm = TRUE),
                   sd.e    = max(sd.e, na.rm = TRUE),
                   group.2 = "shrub",
                   group.3 = as.character(NA),
                   group.4 = as.character(NA)) %>%
  # Replace any missing values (Inf) with NAs
  mutate_at(., vars (-group_cols()), function(x) ifelse(is.infinite(x), NA, x))

# Combine fig.data
working.fig.data <- working.fig.data %>%
  full_join(., fig.data.1) %>%
  full_join(., fig.data.2) %>%
  full_join(., fig.data.3) %>%
  full_join(., fig.data.4)

# Remove objects
rm(fig.data.1, fig.data.2, fig.data.3,
   fig.data.4)
```

# Corli 2019

```{r, echo = `FALSE`}
# Corli2019_Fig3a.png
fig.data.1 <- fig.data %>%
  filter(filename == "Corli2019_Fig3a.png") %>%
  dplyr::rowwise() %>%
  mutate(
    ### Control
    # n
    n.c    = ifelse(group.1 == "without pollinators", n, NA),
    # mean
    mean.c = ifelse(group.1 == "without pollinators", mean, NA),
    # sd
    sd.c   = ifelse(group.1 == "without pollinators", sd, NA),
    ### Experimental
    # n
    n.e    = ifelse(group.1 == "with pollinators", n, NA),
    # mean
    mean.e = ifelse(group.1 == "with pollinators", mean, NA),
    # sd
    sd.e   = ifelse(group.1 == "with pollinators", sd, NA)) %>%
  mutate(
    # Remove old stats columns
    mean       = NULL,
    sd         = NULL,
    n          = NULL,
    r          = NULL,
    se         = NULL,
    error_type = NULL,
    plot_type  = NULL) %>%
  # Merge control and experimental rows
  group_by(filename, variable, article.id) %>%
  dplyr::summarize(n.c     = max(n.c, na.rm = TRUE),
                   mean.c  = max(mean.c, na.rm = TRUE),
                   sd.c    = max(sd.c, na.rm = TRUE),
                   n.e     = max(n.e, na.rm = TRUE),
                   mean.e  = max(mean.e, na.rm = TRUE),
                   sd.e    = max(sd.e, na.rm = TRUE),
                   group.1 = "ten species",
                   group.2 = "pollinators",
                   group.3 = as.character(NA),
                   group.4 = as.character(NA)) %>%
  # Replace any missing values (Inf) with NAs
  mutate_at(., vars (-group_cols()), function(x) ifelse(is.infinite(x), NA, x))

# Corli2019_Fig3b.png
fig.data.2 <- fig.data %>%
  filter(filename == "Corli2019_Fig3b.png") %>%
  dplyr::rowwise() %>%
  mutate(
    ### Control
    # n
    n.c    = ifelse(group.1 == "without pollinators", n, NA),
    # mean
    mean.c = ifelse(group.1 == "without pollinators", mean, NA),
    # sd
    sd.c   = ifelse(group.1 == "without pollinators", sd, NA),
    ### Experimental
    # n
    n.e    = ifelse(group.1 == "with pollinators", n, NA),
    # mean
    mean.e = ifelse(group.1 == "with pollinators", mean, NA),
    # sd
    sd.e   = ifelse(group.1 == "with pollinators", sd, NA)) %>%
  mutate(
    # Remove old stats columns
    mean       = NULL,
    sd         = NULL,
    n          = NULL,
    r          = NULL,
    se         = NULL,
    error_type = NULL,
    plot_type  = NULL) %>%
  # Merge control and experimental rows
  group_by(filename, variable, article.id) %>%
  dplyr::summarize(n.c     = max(n.c, na.rm = TRUE),
                   mean.c  = max(mean.c, na.rm = TRUE),
                   sd.c    = max(sd.c, na.rm = TRUE),
                   n.e     = max(n.e, na.rm = TRUE),
                   mean.e  = max(mean.e, na.rm = TRUE),
                   sd.e    = max(sd.e, na.rm = TRUE),
                   group.1 = "ten species",
                   group.2 = "pollinators",
                   group.3 = as.character(NA),
                   group.4 = as.character(NA)) %>%
  # Replace any missing values (Inf) with NAs
  mutate_at(., vars (-group_cols()), function(x) ifelse(is.infinite(x), NA, x))

# Combine fig.data
working.fig.data <- working.fig.data %>%
  full_join(., fig.data.1) %>%
  full_join(., fig.data.2)

# Remove objects
rm(fig.data.1, fig.data.2)
```

# Grass 2018

```{r, echo = `FALSE`}
# Grass2018_Fig3a.png
fig.data.1 <- fig.data %>%
  filter(filename == "Grass2018_Fig3a.png") %>%
  dplyr::rowwise() %>%
  mutate(
    ### Control
    # n
    n.c    = ifelse(group.1 == "pollinator exclusion", n, NA),
    # mean
    mean.c = ifelse(group.1 == "pollinator exclusion", mean, NA),
    # sd
    sd.c   = ifelse(group.1 == "pollinator exclusion", sd, NA),
    ### Experimental
    # n
    n.e    = ifelse(group.1 == "untreated", n, NA),
    # mean
    mean.e = ifelse(group.1 == "untreated", mean, NA),
    # sd
    sd.e   = ifelse(group.1 == "untreated", sd, NA)) %>%
  mutate(
    # Remove old stats columns
    mean       = NULL,
    sd         = NULL,
    n          = NULL,
    r          = NULL,
    se         = NULL,
    error_type = NULL,
    plot_type  = NULL) %>%
  # Merge control and experimental rows
  group_by(filename, variable, article.id) %>%
  dplyr::summarize(n.c     = max(n.c, na.rm = TRUE),
                   mean.c  = max(mean.c, na.rm = TRUE),
                   sd.c    = max(sd.c, na.rm = TRUE),
                   n.e     = max(n.e, na.rm = TRUE),
                   mean.e  = max(mean.e, na.rm = TRUE),
                   sd.e    = max(sd.e, na.rm = TRUE),
                   group.1 = "Sinapis arvensis",
                   group.2 = "pollinators",
                   group.3 = as.character(NA),
                   group.4 = as.character(NA)) %>%
  # Replace any missing values (Inf) with NAs
  mutate_at(., vars (-group_cols()), function(x) ifelse(is.infinite(x), NA, x))

# Grass2018_Fig3b.png
fig.data.2 <- fig.data %>%
  filter(filename == "Grass2018_Fig3b.png") %>%
  dplyr::rowwise() %>%
  mutate(
    ### Control
    # n
    n.c    = ifelse(group.1 == "pollinator exclusion", n, NA),
    # mean
    mean.c = ifelse(group.1 == "pollinator exclusion", mean, NA),
    # sd
    sd.c   = ifelse(group.1 == "pollinator exclusion", sd, NA),
    ### Experimental
    # n
    n.e    = ifelse(group.1 == "untreated", n, NA),
    # mean
    mean.e = ifelse(group.1 == "untreated", mean, NA),
    # sd
    sd.e   = ifelse(group.1 == "untreated", sd, NA)) %>%
  mutate(
    # Remove old stats columns
    mean       = NULL,
    sd         = NULL,
    n          = NULL,
    r          = NULL,
    se         = NULL,
    error_type = NULL,
    plot_type  = NULL) %>%
  # Merge control and experimental rows
  group_by(filename, variable, article.id) %>%
  dplyr::summarize(n.c     = max(n.c, na.rm = TRUE),
                   mean.c  = max(mean.c, na.rm = TRUE),
                   sd.c    = max(sd.c, na.rm = TRUE),
                   n.e     = max(n.e, na.rm = TRUE),
                   mean.e  = max(mean.e, na.rm = TRUE),
                   sd.e    = max(sd.e, na.rm = TRUE),
                   group.1 = "Sinapis arvensis",
                   group.2 = "pollinators",
                   group.3 = as.character(NA),
                   group.4 = as.character(NA)) %>%
  # Replace any missing values (Inf) with NAs
  mutate_at(., vars (-group_cols()), function(x) ifelse(is.infinite(x), NA, x))

# Grass2018_Fig3c.png
fig.data.3 <- fig.data %>%
  filter(filename == "Grass2018_Fig3c.png") %>%
  dplyr::rowwise() %>%
  mutate(
    ### Control
    # n
    n.c    = ifelse(group.1 == "pollinator exclusion", n, NA),
    # mean
    mean.c = ifelse(group.1 == "pollinator exclusion", mean, NA),
    # sd
    sd.c   = ifelse(group.1 == "pollinator exclusion", sd, NA),
    ### Experimental
    # n
    n.e    = ifelse(group.1 == "untreated", n, NA),
    # mean
    mean.e = ifelse(group.1 == "untreated", mean, NA),
    # sd
    sd.e   = ifelse(group.1 == "untreated", sd, NA)) %>%
  mutate(
    # Remove old stats columns
    mean       = NULL,
    sd         = NULL,
    n          = NULL,
    r          = NULL,
    se         = NULL,
    error_type = NULL,
    plot_type  = NULL) %>%
  # Merge control and experimental rows
  group_by(filename, variable, article.id) %>%
  dplyr::summarize(n.c     = max(n.c, na.rm = TRUE),
                   mean.c  = max(mean.c, na.rm = TRUE),
                   sd.c    = max(sd.c, na.rm = TRUE),
                   n.e     = max(n.e, na.rm = TRUE),
                   mean.e  = max(mean.e, na.rm = TRUE),
                   sd.e    = max(sd.e, na.rm = TRUE),
                   group.1 = "Sinapis arvensis",
                   group.2 = "pollinators",
                   group.3 = as.character(NA),
                   group.4 = as.character(NA)) %>%
  # Replace any missing values (Inf) with NAs
  mutate_at(., vars (-group_cols()), function(x) ifelse(is.infinite(x), NA, x))

# Combine fig.data
working.fig.data <- working.fig.data %>%
  full_join(., fig.data.1) %>%
  full_join(., fig.data.2) %>%
  full_join(., fig.data.3)

# Remove objects
rm(fig.data.1, fig.data.2, fig.data.3)
```

# Gundale 2012

```{r, echo = `FALSE`}
# Gundale2012_Fig2a.png
fig.data.1 <- fig.data %>%
  filter(filename == "Gundale2012_Fig2a.png") %>%
  filter(group.3 == "small") %>%
  dplyr::rowwise() %>%
  mutate(
    ### Control
    # n
    n.c    = ifelse(group.2 == "w/o Vaccinium vitisidaea", n, NA),
    # mean
    mean.c = ifelse(group.2 == "w/o Vaccinium vitisidaea", mean, NA),
    # sd
    sd.c   = ifelse(group.2 == "w/o Vaccinium vitisidaea", sd, NA),
    ### Experimental
    # n
    n.e    = ifelse(group.2 == "w/ Vaccinium vitisidaea", n, NA),
    # mean
    mean.e = ifelse(group.2 == "w/ Vaccinium vitisidaea", mean, NA),
    # sd
    sd.e   = ifelse(group.2 == "w/ Vaccinium vitisidaea", sd, NA)) %>%
  mutate(
    # Remove old stats columns
    mean       = NULL,
    sd         = NULL,
    n          = NULL,
    r          = NULL,
    se         = NULL,
    error_type = NULL,
    plot_type  = NULL) %>%
  # Merge control and experimental rows
  group_by(filename, variable, article.id) %>%
  dplyr::summarize(n.c     = max(n.c, na.rm = TRUE),
                   mean.c  = max(mean.c, na.rm = TRUE),
                   sd.c    = max(sd.c, na.rm = TRUE),
                   n.e     = max(n.e, na.rm = TRUE),
                   mean.e  = max(mean.e, na.rm = TRUE),
                   sd.e    = max(sd.e, na.rm = TRUE),
                   group.1 = "Vaccinium myrtillus",
                   group.2 = "Vaccinium vitis-idaea",
                   group.3 = "small",
                   group.4 = as.character(NA)) %>%
  # Replace any missing values (Inf) with NAs
  mutate_at(., vars (-group_cols()), function(x) ifelse(is.infinite(x), NA, x))

# Gundale2012_Fig2a.png
fig.data.2 <- fig.data %>%
  filter(filename == "Gundale2012_Fig2a.png") %>%
  filter(group.3 == "medium") %>%
  dplyr::rowwise() %>%
  mutate(
    ### Control
    # n
    n.c    = ifelse(group.2 == "w/o Vaccinium vitisidaea", n, NA),
    # mean
    mean.c = ifelse(group.2 == "w/o Vaccinium vitisidaea", mean, NA),
    # sd
    sd.c   = ifelse(group.2 == "w/o Vaccinium vitisidaea", sd, NA),
    ### Experimental
    # n
    n.e    = ifelse(group.2 == "w/ Vaccinium vitisidaea", n, NA),
    # mean
    mean.e = ifelse(group.2 == "w/ Vaccinium vitisidaea", mean, NA),
    # sd
    sd.e   = ifelse(group.2 == "w/ Vaccinium vitisidaea", sd, NA)) %>%
  mutate(
    # Remove old stats columns
    mean       = NULL,
    sd         = NULL,
    n          = NULL,
    r          = NULL,
    se         = NULL,
    error_type = NULL,
    plot_type  = NULL) %>%
  # Merge control and experimental rows
  group_by(filename, variable, article.id) %>%
  dplyr::summarize(n.c     = max(n.c, na.rm = TRUE),
                   mean.c  = max(mean.c, na.rm = TRUE),
                   sd.c    = max(sd.c, na.rm = TRUE),
                   n.e     = max(n.e, na.rm = TRUE),
                   mean.e  = max(mean.e, na.rm = TRUE),
                   sd.e    = max(sd.e, na.rm = TRUE),
                   group.1 = "Vaccinium myrtillus",
                   group.2 = "Vaccinium vitis-idaea",
                   group.3 = "medium",
                   group.4 = as.character(NA)) %>%
  # Replace any missing values (Inf) with NAs
  mutate_at(., vars (-group_cols()), function(x) ifelse(is.infinite(x), NA, x))

# Gundale2012_Fig2a.png
fig.data.3 <- fig.data %>%
  filter(filename == "Gundale2012_Fig2a.png") %>%
  filter(group.3 == "large") %>%
  dplyr::rowwise() %>%
  mutate(
    ### Control
    # n
    n.c    = ifelse(group.2 == "w/o Vaccinium vitisidaea", n, NA),
    # mean
    mean.c = ifelse(group.2 == "w/o Vaccinium vitisidaea", mean, NA),
    # sd
    sd.c   = ifelse(group.2 == "w/o Vaccinium vitisidaea", sd, NA),
    ### Experimental
    # n
    n.e    = ifelse(group.2 == "w/ Vaccinium vitisidaea", n, NA),
    # mean
    mean.e = ifelse(group.2 == "w/ Vaccinium vitisidaea", mean, NA),
    # sd
    sd.e   = ifelse(group.2 == "w/ Vaccinium vitisidaea", sd, NA)) %>%
  mutate(
    # Remove old stats columns
    mean       = NULL,
    sd         = NULL,
    n          = NULL,
    r          = NULL,
    se         = NULL,
    error_type = NULL,
    plot_type  = NULL) %>%
  # Merge control and experimental rows
  group_by(filename, variable, article.id) %>%
  dplyr::summarize(n.c     = max(n.c, na.rm = TRUE),
                   mean.c  = max(mean.c, na.rm = TRUE),
                   sd.c    = max(sd.c, na.rm = TRUE),
                   n.e     = max(n.e, na.rm = TRUE),
                   mean.e  = max(mean.e, na.rm = TRUE),
                   sd.e    = max(sd.e, na.rm = TRUE),
                   group.1 = "Vaccinium myrtillus",
                   group.2 = "Vaccinium vitis-idaea",
                   group.3 = "large",
                   group.4 = as.character(NA)) %>%
  # Replace any missing values (Inf) with NAs
  mutate_at(., vars (-group_cols()), function(x) ifelse(is.infinite(x), NA, x))

# Gundale2012_Fig2a.png
fig.data.4 <- fig.data %>%
  filter(filename == "Gundale2012_Fig2a.png") %>%
  filter(group.3 == "small") %>%
  dplyr::rowwise() %>%
  mutate(
    ### Control
    # n
    n.c    = ifelse(group.2 == "w/o Empetrum hermaphroditum", n, NA),
    # mean
    mean.c = ifelse(group.2 == "w/o Empetrum hermaphroditum", mean, NA),
    # sd
    sd.c   = ifelse(group.2 == "w/o Empetrum hermaphroditum", sd, NA),
    ### Experimental
    # n
    n.e    = ifelse(group.2 == "w/ Empetrum hermaphroditum", n, NA),
    # mean
    mean.e = ifelse(group.2 == "w/ Empetrum hermaphroditum", mean, NA),
    # sd
    sd.e   = ifelse(group.2 == "w/ Empetrum hermaphroditum", sd, NA)) %>%
  mutate(
    # Remove old stats columns
    mean       = NULL,
    sd         = NULL,
    n          = NULL,
    r          = NULL,
    se         = NULL,
    error_type = NULL,
    plot_type  = NULL) %>%
  # Merge control and experimental rows
  group_by(filename, variable, article.id) %>%
  dplyr::summarize(n.c     = max(n.c, na.rm = TRUE),
                   mean.c  = max(mean.c, na.rm = TRUE),
                   sd.c    = max(sd.c, na.rm = TRUE),
                   n.e     = max(n.e, na.rm = TRUE),
                   mean.e  = max(mean.e, na.rm = TRUE),
                   sd.e    = max(sd.e, na.rm = TRUE),
                   group.1 = "Vaccinium myrtillus",
                   group.2 = "Empetrum hermaphroditum",
                   group.3 = "small",
                   group.4 = as.character(NA)) %>%
  # Replace any missing values (Inf) with NAs
  mutate_at(., vars (-group_cols()), function(x) ifelse(is.infinite(x), NA, x))

# Gundale2012_Fig2a.png
fig.data.5 <- fig.data %>%
  filter(filename == "Gundale2012_Fig2a.png") %>%
  filter(group.3 == "medium") %>%
  dplyr::rowwise() %>%
  mutate(
    ### Control
    # n
    n.c    = ifelse(group.2 == "w/o Empetrum hermaphroditum", n, NA),
    # mean
    mean.c = ifelse(group.2 == "w/o Empetrum hermaphroditum", mean, NA),
    # sd
    sd.c   = ifelse(group.2 == "w/o Empetrum hermaphroditum", sd, NA),
    ### Experimental
    # n
    n.e    = ifelse(group.2 == "w/ Empetrum hermaphroditum", n, NA),
    # mean
    mean.e = ifelse(group.2 == "w/ Empetrum hermaphroditum", mean, NA),
    # sd
    sd.e   = ifelse(group.2 == "w/ Empetrum hermaphroditum", sd, NA)) %>%
  mutate(
    # Remove old stats columns
    mean       = NULL,
    sd         = NULL,
    n          = NULL,
    r          = NULL,
    se         = NULL,
    error_type = NULL,
    plot_type  = NULL) %>%
  # Merge control and experimental rows
  group_by(filename, variable, article.id) %>%
  dplyr::summarize(n.c     = max(n.c, na.rm = TRUE),
                   mean.c  = max(mean.c, na.rm = TRUE),
                   sd.c    = max(sd.c, na.rm = TRUE),
                   n.e     = max(n.e, na.rm = TRUE),
                   mean.e  = max(mean.e, na.rm = TRUE),
                   sd.e    = max(sd.e, na.rm = TRUE),
                   group.1 = "Vaccinium myrtillus",
                   group.2 = "Empetrum hermaphroditum",
                   group.3 = "medium",
                   group.4 = as.character(NA)) %>%
  # Replace any missing values (Inf) with NAs
  mutate_at(., vars (-group_cols()), function(x) ifelse(is.infinite(x), NA, x))

# Gundale2012_Fig2a.png
fig.data.6 <- fig.data %>%
  filter(filename == "Gundale2012_Fig2a.png") %>%
  filter(group.3 == "large") %>%
  dplyr::rowwise() %>%
  mutate(
    ### Control
    # n
    n.c    = ifelse(group.2 == "w/o Empetrum hermaphroditum", n, NA),
    # mean
    mean.c = ifelse(group.2 == "w/o Empetrum hermaphroditum", mean, NA),
    # sd
    sd.c   = ifelse(group.2 == "w/o Empetrum hermaphroditum", sd, NA),
    ### Experimental
    # n
    n.e    = ifelse(group.2 == "w/ Empetrum hermaphroditum", n, NA),
    # mean
    mean.e = ifelse(group.2 == "w/ Empetrum hermaphroditum", mean, NA),
    # sd
    sd.e   = ifelse(group.2 == "w/ Empetrum hermaphroditum", sd, NA)) %>%
  mutate(
    # Remove old stats columns
    mean       = NULL,
    sd         = NULL,
    n          = NULL,
    r          = NULL,
    se         = NULL,
    error_type = NULL,
    plot_type  = NULL) %>%
  # Merge control and experimental rows
  group_by(filename, variable, article.id) %>%
  dplyr::summarize(n.c     = max(n.c, na.rm = TRUE),
                   mean.c  = max(mean.c, na.rm = TRUE),
                   sd.c    = max(sd.c, na.rm = TRUE),
                   n.e     = max(n.e, na.rm = TRUE),
                   mean.e  = max(mean.e, na.rm = TRUE),
                   sd.e    = max(sd.e, na.rm = TRUE),
                   group.1 = "Vaccinium myrtillus",
                   group.2 = "Empetrum hermaphroditum",
                   group.3 = "large",
                   group.4 = as.character(NA)) %>%
  # Replace any missing values (Inf) with NAs
  mutate_at(., vars (-group_cols()), function(x) ifelse(is.infinite(x), NA, x))

# Gundale2012_Fig2b.png
fig.data.7 <- fig.data %>%
  filter(filename == "Gundale2012_Fig2b.png") %>%
  filter(group.3 == "small") %>%
  dplyr::rowwise() %>%
  mutate(
    ### Control
    # n
    n.c    = ifelse(group.2 == "w/o Vaccinium myrtillus", n, NA),
    # mean
    mean.c = ifelse(group.2 == "w/o Vaccinium myrtillus", mean, NA),
    # sd
    sd.c   = ifelse(group.2 == "w/o Vaccinium myrtillus", sd, NA),
    ### Experimental
    # n
    n.e    = ifelse(group.2 == "w/ Vaccinium myrtillus", n, NA),
    # mean
    mean.e = ifelse(group.2 == "w/ Vaccinium myrtillus", mean, NA),
    # sd
    sd.e   = ifelse(group.2 == "w/ Vaccinium myrtillus", sd, NA)) %>%
  mutate(
    # Remove old stats columns
    mean       = NULL,
    sd         = NULL,
    n          = NULL,
    r          = NULL,
    se         = NULL,
    error_type = NULL,
    plot_type  = NULL) %>%
  # Merge control and experimental rows
  group_by(filename, variable, article.id) %>%
  dplyr::summarize(n.c     = max(n.c, na.rm = TRUE),
                   mean.c  = max(mean.c, na.rm = TRUE),
                   sd.c    = max(sd.c, na.rm = TRUE),
                   n.e     = max(n.e, na.rm = TRUE),
                   mean.e  = max(mean.e, na.rm = TRUE),
                   sd.e    = max(sd.e, na.rm = TRUE),
                   group.1 = "Vaccinium vitis-idaea",
                   group.2 = "Vaccinium myrtillus",
                   group.3 = "small",
                   group.4 = as.character(NA)) %>%
  # Replace any missing values (Inf) with NAs
  mutate_at(., vars (-group_cols()), function(x) ifelse(is.infinite(x), NA, x))

# Gundale2012_Fig2b.png
fig.data.8 <- fig.data %>%
  filter(filename == "Gundale2012_Fig2b.png") %>%
  filter(group.3 == "medium") %>%
  dplyr::rowwise() %>%
  mutate(
    ### Control
    # n
    n.c    = ifelse(group.2 == "w/o Vaccinium myrtillus", n, NA),
    # mean
    mean.c = ifelse(group.2 == "w/o Vaccinium myrtillus", mean, NA),
    # sd
    sd.c   = ifelse(group.2 == "w/o Vaccinium myrtillus", sd, NA),
    ### Experimental
    # n
    n.e    = ifelse(group.2 == "w/ Vaccinium myrtillus", n, NA),
    # mean
    mean.e = ifelse(group.2 == "w/ Vaccinium myrtillus", mean, NA),
    # sd
    sd.e   = ifelse(group.2 == "w/ Vaccinium myrtillus", sd, NA)) %>%
  mutate(
    # Remove old stats columns
    mean       = NULL,
    sd         = NULL,
    n          = NULL,
    r          = NULL,
    se         = NULL,
    error_type = NULL,
    plot_type  = NULL) %>%
  # Merge control and experimental rows
  group_by(filename, variable, article.id) %>%
  dplyr::summarize(n.c     = max(n.c, na.rm = TRUE),
                   mean.c  = max(mean.c, na.rm = TRUE),
                   sd.c    = max(sd.c, na.rm = TRUE),
                   n.e     = max(n.e, na.rm = TRUE),
                   mean.e  = max(mean.e, na.rm = TRUE),
                   sd.e    = max(sd.e, na.rm = TRUE),
                   group.1 = "Vaccinium vitis-idaea",
                   group.2 = "Vaccinium myrtillus",
                   group.3 = "medium",
                   group.4 = as.character(NA)) %>%
  # Replace any missing values (Inf) with NAs
  mutate_at(., vars (-group_cols()), function(x) ifelse(is.infinite(x), NA, x))

# Gundale2012_Fig2b.png
fig.data.9 <- fig.data %>%
  filter(filename == "Gundale2012_Fig2b.png") %>%
  filter(group.3 == "large") %>%
  dplyr::rowwise() %>%
  mutate(
    ### Control
    # n
    n.c    = ifelse(group.2 == "w/o Vaccinium myrtillus", n, NA),
    # mean
    mean.c = ifelse(group.2 == "w/o Vaccinium myrtillus", mean, NA),
    # sd
    sd.c   = ifelse(group.2 == "w/o Vaccinium myrtillus", sd, NA),
    ### Experimental
    # n
    n.e    = ifelse(group.2 == "w/ Vaccinium myrtillus", n, NA),
    # mean
    mean.e = ifelse(group.2 == "w/ Vaccinium myrtillus", mean, NA),
    # sd
    sd.e   = ifelse(group.2 == "w/ Vaccinium myrtillus", sd, NA)) %>%
  mutate(
    # Remove old stats columns
    mean       = NULL,
    sd         = NULL,
    n          = NULL,
    r          = NULL,
    se         = NULL,
    error_type = NULL,
    plot_type  = NULL) %>%
  # Merge control and experimental rows
  group_by(filename, variable, article.id) %>%
  dplyr::summarize(n.c     = max(n.c, na.rm = TRUE),
                   mean.c  = max(mean.c, na.rm = TRUE),
                   sd.c    = max(sd.c, na.rm = TRUE),
                   n.e     = max(n.e, na.rm = TRUE),
                   mean.e  = max(mean.e, na.rm = TRUE),
                   sd.e    = max(sd.e, na.rm = TRUE),
                   group.1 = "Vaccinium vitis-idaea",
                   group.2 = "Vaccinium myrtillus",
                   group.3 = "large",
                   group.4 = as.character(NA)) %>%
  # Replace any missing values (Inf) with NAs
  mutate_at(., vars (-group_cols()), function(x) ifelse(is.infinite(x), NA, x))

# Gundale2012_Fig2b.png
fig.data.10 <- fig.data %>%
  filter(filename == "Gundale2012_Fig2b.png") %>%
  filter(group.3 == "small") %>%
  dplyr::rowwise() %>%
  mutate(
    ### Control
    # n
    n.c    = ifelse(group.2 == "w/o Empetrum hermaphroditums", n, NA),
    # mean
    mean.c = ifelse(group.2 == "w/o Empetrum hermaphroditums", mean, NA),
    # sd
    sd.c   = ifelse(group.2 == "w/o Empetrum hermaphroditums", sd, NA),
    ### Experimental
    # n
    n.e    = ifelse(group.2 == "w/ Empetrum hermaphroditums", n, NA),
    # mean
    mean.e = ifelse(group.2 == "w/ Empetrum hermaphroditums", mean, NA),
    # sd
    sd.e   = ifelse(group.2 == "w/ Empetrum hermaphroditums", sd, NA)) %>%
  mutate(
    # Remove old stats columns
    mean       = NULL,
    sd         = NULL,
    n          = NULL,
    r          = NULL,
    se         = NULL,
    error_type = NULL,
    plot_type  = NULL) %>%
  # Merge control and experimental rows
  group_by(filename, variable, article.id) %>%
  dplyr::summarize(n.c     = max(n.c, na.rm = TRUE),
                   mean.c  = max(mean.c, na.rm = TRUE),
                   sd.c    = max(sd.c, na.rm = TRUE),
                   n.e     = max(n.e, na.rm = TRUE),
                   mean.e  = max(mean.e, na.rm = TRUE),
                   sd.e    = max(sd.e, na.rm = TRUE),
                   group.1 = "Vaccinium vitis-idaea",
                   group.2 = "Empetrum hermaphroditum",
                   group.3 = "small",
                   group.4 = as.character(NA)) %>%
  # Replace any missing values (Inf) with NAs
  mutate_at(., vars (-group_cols()), function(x) ifelse(is.infinite(x), NA, x))

# Gundale2012_Fig2b.png
fig.data.11 <- fig.data %>%
  filter(filename == "Gundale2012_Fig2b.png") %>%
  filter(group.3 == "medium") %>%
  dplyr::rowwise() %>%
  mutate(
    ### Control
    # n
    n.c    = ifelse(group.2 == "w/o Empetrum hermaphroditums", n, NA),
    # mean
    mean.c = ifelse(group.2 == "w/o Empetrum hermaphroditums", mean, NA),
    # sd
    sd.c   = ifelse(group.2 == "w/o Empetrum hermaphroditums", sd, NA),
    ### Experimental
    # n
    n.e    = ifelse(group.2 == "w/ Empetrum hermaphroditums", n, NA),
    # mean
    mean.e = ifelse(group.2 == "w/ Empetrum hermaphroditums", mean, NA),
    # sd
    sd.e   = ifelse(group.2 == "w/ Empetrum hermaphroditums", sd, NA)) %>%
  mutate(
    # Remove old stats columns
    mean       = NULL,
    sd         = NULL,
    n          = NULL,
    r          = NULL,
    se         = NULL,
    error_type = NULL,
    plot_type  = NULL) %>%
  # Merge control and experimental rows
  group_by(filename, variable, article.id) %>%
  dplyr::summarize(n.c     = max(n.c, na.rm = TRUE),
                   mean.c  = max(mean.c, na.rm = TRUE),
                   sd.c    = max(sd.c, na.rm = TRUE),
                   n.e     = max(n.e, na.rm = TRUE),
                   mean.e  = max(mean.e, na.rm = TRUE),
                   sd.e    = max(sd.e, na.rm = TRUE),
                   group.1 = "Vaccinium vitis-idaea",
                   group.2 = "Empetrum hermaphroditum",
                   group.3 = "medium",
                   group.4 = as.character(NA)) %>%
  # Replace any missing values (Inf) with NAs
  mutate_at(., vars (-group_cols()), function(x) ifelse(is.infinite(x), NA, x))

# Gundale2012_Fig2b.png
fig.data.12 <- fig.data %>%
  filter(filename == "Gundale2012_Fig2b.png") %>%
  filter(group.3 == "large") %>%
  dplyr::rowwise() %>%
  mutate(
    ### Control
    # n
    n.c    = ifelse(group.2 == "w/o Empetrum hermaphroditums", n, NA),
    # mean
    mean.c = ifelse(group.2 == "w/o Empetrum hermaphroditums", mean, NA),
    # sd
    sd.c   = ifelse(group.2 == "w/o Empetrum hermaphroditums", sd, NA),
    ### Experimental
    # n
    n.e    = ifelse(group.2 == "w/ Empetrum hermaphroditums", n, NA),
    # mean
    mean.e = ifelse(group.2 == "w/ Empetrum hermaphroditums", mean, NA),
    # sd
    sd.e   = ifelse(group.2 == "w/ Empetrum hermaphroditums", sd, NA)) %>%
  mutate(
    # Remove old stats columns
    mean       = NULL,
    sd         = NULL,
    n          = NULL,
    r          = NULL,
    se         = NULL,
    error_type = NULL,
    plot_type  = NULL) %>%
  # Merge control and experimental rows
  group_by(filename, variable, article.id) %>%
  dplyr::summarize(n.c     = max(n.c, na.rm = TRUE),
                   mean.c  = max(mean.c, na.rm = TRUE),
                   sd.c    = max(sd.c, na.rm = TRUE),
                   n.e     = max(n.e, na.rm = TRUE),
                   mean.e  = max(mean.e, na.rm = TRUE),
                   sd.e    = max(sd.e, na.rm = TRUE),
                   group.1 = "Vaccinium vitis-idaea",
                   group.2 = "Empetrum hermaphroditum",
                   group.3 = "large",
                   group.4 = as.character(NA)) %>%
  # Replace any missing values (Inf) with NAs
  mutate_at(., vars (-group_cols()), function(x) ifelse(is.infinite(x), NA, x))

# Gundale2012_Fig2c.png
fig.data.13 <- fig.data %>%
  filter(filename == "Gundale2012_Fig2c.png") %>%
  filter(group.3 == "small") %>%
  dplyr::rowwise() %>%
  mutate(
    ### Control
    # n
    n.c    = ifelse(group.2 == "w/o Vaccinium myrtillus", n, NA),
    # mean
    mean.c = ifelse(group.2 == "w/o Vaccinium myrtillus", mean, NA),
    # sd
    sd.c   = ifelse(group.2 == "w/o Vaccinium myrtillus", sd, NA),
    ### Experimental
    # n
    n.e    = ifelse(group.2 == "w/ Vaccinium myrtillus", n, NA),
    # mean
    mean.e = ifelse(group.2 == "w/ Vaccinium myrtillus", mean, NA),
    # sd
    sd.e   = ifelse(group.2 == "w/ Vaccinium myrtillus", sd, NA)) %>%
  mutate(
    # Remove old stats columns
    mean       = NULL,
    sd         = NULL,
    n          = NULL,
    r          = NULL,
    se         = NULL,
    error_type = NULL,
    plot_type  = NULL) %>%
  # Merge control and experimental rows
  group_by(filename, variable, article.id) %>%
  dplyr::summarize(n.c     = max(n.c, na.rm = TRUE),
                   mean.c  = max(mean.c, na.rm = TRUE),
                   sd.c    = max(sd.c, na.rm = TRUE),
                   n.e     = max(n.e, na.rm = TRUE),
                   mean.e  = max(mean.e, na.rm = TRUE),
                   sd.e    = max(sd.e, na.rm = TRUE),
                   group.1 = "Empetrum hermaphroditum",
                   group.2 = "Vaccinium myrtillus",
                   group.3 = "small",
                   group.4 = as.character(NA)) %>%
  # Replace any missing values (Inf) with NAs
  mutate_at(., vars (-group_cols()), function(x) ifelse(is.infinite(x), NA, x))

# Gundale2012_Fig2c.png
fig.data.14 <- fig.data %>%
  filter(filename == "Gundale2012_Fig2c.png") %>%
  filter(group.3 == "medium") %>%
  dplyr::rowwise() %>%
  mutate(
    ### Control
    # n
    n.c    = ifelse(group.2 == "w/o Vaccinium myrtillus", n, NA),
    # mean
    mean.c = ifelse(group.2 == "w/o Vaccinium myrtillus", mean, NA),
    # sd
    sd.c   = ifelse(group.2 == "w/o Vaccinium myrtillus", sd, NA),
    ### Experimental
    # n
    n.e    = ifelse(group.2 == "w/ Vaccinium myrtillus", n, NA),
    # mean
    mean.e = ifelse(group.2 == "w/ Vaccinium myrtillus", mean, NA),
    # sd
    sd.e   = ifelse(group.2 == "w/ Vaccinium myrtillus", sd, NA)) %>%
  mutate(
    # Remove old stats columns
    mean       = NULL,
    sd         = NULL,
    n          = NULL,
    r          = NULL,
    se         = NULL,
    error_type = NULL,
    plot_type  = NULL) %>%
  # Merge control and experimental rows
  group_by(filename, variable, article.id) %>%
  dplyr::summarize(n.c     = max(n.c, na.rm = TRUE),
                   mean.c  = max(mean.c, na.rm = TRUE),
                   sd.c    = max(sd.c, na.rm = TRUE),
                   n.e     = max(n.e, na.rm = TRUE),
                   mean.e  = max(mean.e, na.rm = TRUE),
                   sd.e    = max(sd.e, na.rm = TRUE),
                   group.1 = "Empetrum hermaphroditum",
                   group.2 = "Vaccinium myrtillus",
                   group.3 = "medium",
                   group.4 = as.character(NA)) %>%
  # Replace any missing values (Inf) with NAs
  mutate_at(., vars (-group_cols()), function(x) ifelse(is.infinite(x), NA, x))

# Gundale2012_Fig2c.png
fig.data.15 <- fig.data %>%
  filter(filename == "Gundale2012_Fig2c.png") %>%
  filter(group.3 == "large") %>%
  dplyr::rowwise() %>%
  mutate(
    ### Control
    # n
    n.c    = ifelse(group.2 == "w/o Vaccinium myrtillus", n, NA),
    # mean
    mean.c = ifelse(group.2 == "w/o Vaccinium myrtillus", mean, NA),
    # sd
    sd.c   = ifelse(group.2 == "w/o Vaccinium myrtillus", sd, NA),
    ### Experimental
    # n
    n.e    = ifelse(group.2 == "w/ Vaccinium myrtillus", n, NA),
    # mean
    mean.e = ifelse(group.2 == "w/ Vaccinium myrtillus", mean, NA),
    # sd
    sd.e   = ifelse(group.2 == "w/ Vaccinium myrtillus", sd, NA)) %>%
  mutate(
    # Remove old stats columns
    mean       = NULL,
    sd         = NULL,
    n          = NULL,
    r          = NULL,
    se         = NULL,
    error_type = NULL,
    plot_type  = NULL) %>%
  # Merge control and experimental rows
  group_by(filename, variable, article.id) %>%
  dplyr::summarize(n.c     = max(n.c, na.rm = TRUE),
                   mean.c  = max(mean.c, na.rm = TRUE),
                   sd.c    = max(sd.c, na.rm = TRUE),
                   n.e     = max(n.e, na.rm = TRUE),
                   mean.e  = max(mean.e, na.rm = TRUE),
                   sd.e    = max(sd.e, na.rm = TRUE),
                   group.1 = "Empetrum hermaphroditum",
                   group.2 = "Vaccinium myrtillus",
                   group.3 = "large",
                   group.4 = as.character(NA)) %>%
  # Replace any missing values (Inf) with NAs
  mutate_at(., vars (-group_cols()), function(x) ifelse(is.infinite(x), NA, x))


# Gundale2012_Fig2c.png
fig.data.16 <- fig.data %>%
  filter(filename == "Gundale2012_Fig2c.png") %>%
  filter(group.3 == "small") %>%
  dplyr::rowwise() %>%
  mutate(
    ### Control
    # n
    n.c    = ifelse(group.2 == "w/o Vaccinium vitis-idaea", n, NA),
    # mean
    mean.c = ifelse(group.2 == "w/o Vaccinium vitis-idaea", mean, NA),
    # sd
    sd.c   = ifelse(group.2 == "w/o Vaccinium vitis-idaea", sd, NA),
    ### Experimental
    # n
    n.e    = ifelse(group.2 == "w/ Vaccinium vitis-idaea", n, NA),
    # mean
    mean.e = ifelse(group.2 == "w/ Vaccinium vitis-idaea", mean, NA),
    # sd
    sd.e   = ifelse(group.2 == "w/ Vaccinium vitis-idaea", sd, NA)) %>%
  mutate(
    # Remove old stats columns
    mean       = NULL,
    sd         = NULL,
    n          = NULL,
    r          = NULL,
    se         = NULL,
    error_type = NULL,
    plot_type  = NULL) %>%
  # Merge control and experimental rows
  group_by(filename, variable, article.id) %>%
  dplyr::summarize(n.c     = max(n.c, na.rm = TRUE),
                   mean.c  = max(mean.c, na.rm = TRUE),
                   sd.c    = max(sd.c, na.rm = TRUE),
                   n.e     = max(n.e, na.rm = TRUE),
                   mean.e  = max(mean.e, na.rm = TRUE),
                   sd.e    = max(sd.e, na.rm = TRUE),
                   group.1 = "Empetrum hermaphroditum",
                   group.2 = "Vaccinium vitis-idaea",
                   group.3 = "small",
                   group.4 = as.character(NA)) %>%
  # Replace any missing values (Inf) with NAs
  mutate_at(., vars (-group_cols()), function(x) ifelse(is.infinite(x), NA, x))

# Gundale2012_Fig2c.png
fig.data.17 <- fig.data %>%
  filter(filename == "Gundale2012_Fig2c.png") %>%
  filter(group.3 == "medium") %>%
  dplyr::rowwise() %>%
  mutate(
    ### Control
    # n
    n.c    = ifelse(group.2 == "w/o Vaccinium vitis-idaea", n, NA),
    # mean
    mean.c = ifelse(group.2 == "w/o Vaccinium vitis-idaea", mean, NA),
    # sd
    sd.c   = ifelse(group.2 == "w/o Vaccinium vitis-idaea", sd, NA),
    ### Experimental
    # n
    n.e    = ifelse(group.2 == "w/ Vaccinium vitis-idaea", n, NA),
    # mean
    mean.e = ifelse(group.2 == "w/ Vaccinium vitis-idaea", mean, NA),
    # sd
    sd.e   = ifelse(group.2 == "w/ Vaccinium vitis-idaea", sd, NA)) %>%
  mutate(
    # Remove old stats columns
    mean       = NULL,
    sd         = NULL,
    n          = NULL,
    r          = NULL,
    se         = NULL,
    error_type = NULL,
    plot_type  = NULL) %>%
  # Merge control and experimental rows
  group_by(filename, variable, article.id) %>%
  dplyr::summarize(n.c     = max(n.c, na.rm = TRUE),
                   mean.c  = max(mean.c, na.rm = TRUE),
                   sd.c    = max(sd.c, na.rm = TRUE),
                   n.e     = max(n.e, na.rm = TRUE),
                   mean.e  = max(mean.e, na.rm = TRUE),
                   sd.e    = max(sd.e, na.rm = TRUE),
                   group.1 = "Empetrum hermaphroditum",
                   group.2 = "Vaccinium vitis-idaea",
                   group.3 = "medium",
                   group.4 = as.character(NA)) %>%
  # Replace any missing values (Inf) with NAs
  mutate_at(., vars (-group_cols()), function(x) ifelse(is.infinite(x), NA, x))

# Gundale2012_Fig2c.png
fig.data.18 <- fig.data %>%
  filter(filename == "Gundale2012_Fig2c.png") %>%
  filter(group.3 == "large") %>%
  dplyr::rowwise() %>%
  mutate(
    ### Control
    # n
    n.c    = ifelse(group.2 == "w/o Vaccinium vitis-idaea", n, NA),
    # mean
    mean.c = ifelse(group.2 == "w/o Vaccinium vitis-idaea", mean, NA),
    # sd
    sd.c   = ifelse(group.2 == "w/o Vaccinium vitis-idaea", sd, NA),
    ### Experimental
    # n
    n.e    = ifelse(group.2 == "w/ Vaccinium vitis-idaea", n, NA),
    # mean
    mean.e = ifelse(group.2 == "w/ Vaccinium vitis-idaea", mean, NA),
    # sd
    sd.e   = ifelse(group.2 == "w/ Vaccinium vitis-idaea", sd, NA)) %>%
  mutate(
    # Remove old stats columns
    mean       = NULL,
    sd         = NULL,
    n          = NULL,
    r          = NULL,
    se         = NULL,
    error_type = NULL,
    plot_type  = NULL) %>%
  # Merge control and experimental rows
  group_by(filename, variable, article.id) %>%
  dplyr::summarize(n.c     = max(n.c, na.rm = TRUE),
                   mean.c  = max(mean.c, na.rm = TRUE),
                   sd.c    = max(sd.c, na.rm = TRUE),
                   n.e     = max(n.e, na.rm = TRUE),
                   mean.e  = max(mean.e, na.rm = TRUE),
                   sd.e    = max(sd.e, na.rm = TRUE),
                   group.1 = "Empetrum hermaphroditum",
                   group.2 = "Vaccinium vitis-idaea",
                   group.3 = "large",
                   group.4 = as.character(NA)) %>%
  # Replace any missing values (Inf) with NAs
  mutate_at(., vars (-group_cols()), function(x) ifelse(is.infinite(x), NA, x))

# Combine fig.data
working.fig.data <- working.fig.data %>%
  full_join(., fig.data.1) %>%
  full_join(., fig.data.2) %>%
  full_join(., fig.data.3) %>%
  full_join(., fig.data.4) %>%
  full_join(., fig.data.5) %>%
  full_join(., fig.data.6) %>%
  full_join(., fig.data.7) %>%
  full_join(., fig.data.8) %>%
  full_join(., fig.data.9) %>%
  full_join(., fig.data.10) %>%
  full_join(., fig.data.11) %>%
  full_join(., fig.data.12) %>%
  full_join(., fig.data.13) %>%
  full_join(., fig.data.14) %>%
  full_join(., fig.data.15) %>%
  full_join(., fig.data.16) %>%
  full_join(., fig.data.17) %>%
  full_join(., fig.data.18)

# Remove objects
rm(fig.data.1, fig.data.2, fig.data.3,
   fig.data.4, fig.data.5, fig.data.6,
   fig.data.7, fig.data.8, fig.data.9,
   fig.data.10, fig.data.11, fig.data.12,
   fig.data.13, fig.data.14, fig.data.15,
   fig.data.16, fig.data.17, fig.data.18)

# Gundale2012_Fig3a.png
fig.data.1 <- fig.data %>%
  filter(filename == "Gundale2012_Fig3a.png") %>%
  filter(group.1 == "Vaccinium myrtillus") %>%
  dplyr::rowwise() %>%
  mutate(
    ### Control
    # n
    n.c    = ifelse(group.2 == "w/o tree roots", n, NA),
    # mean
    mean.c = ifelse(group.2 == "w/o tree roots", mean, NA),
    # sd
    sd.c   = ifelse(group.2 == "w/o tree roots", sd, NA),
    ### Experimental
    # n
    n.e    = ifelse(group.2 == "w/ tree roots", n, NA),
    # mean
    mean.e = ifelse(group.2 == "w/ tree roots", mean, NA),
    # sd
    sd.e   = ifelse(group.2 == "w/ tree roots", sd, NA)) %>%
  mutate(
    # Remove old stats columns
    mean       = NULL,
    sd         = NULL,
    n          = NULL,
    r          = NULL,
    se         = NULL,
    error_type = NULL,
    plot_type  = NULL) %>%
  # Merge control and experimental rows
  group_by(filename, variable, article.id) %>%
  dplyr::summarize(n.c     = max(n.c, na.rm = TRUE),
                   mean.c  = max(mean.c, na.rm = TRUE),
                   sd.c    = max(sd.c, na.rm = TRUE),
                   n.e     = max(n.e, na.rm = TRUE),
                   mean.e  = max(mean.e, na.rm = TRUE),
                   sd.e    = max(sd.e, na.rm = TRUE),
                   group.1 = "Vaccinium myrtillus",
                   group.2 = "tree roots",
                   group.3 = as.character(NA),
                   group.4 = as.character(NA)) %>%
  # Replace any missing values (Inf) with NAs
  mutate_at(., vars (-group_cols()), function(x) ifelse(is.infinite(x), NA, x))

# Gundale2012_Fig3a.png
fig.data.2 <- fig.data %>%
  filter(filename == "Gundale2012_Fig3a.png") %>%
  filter(group.1 == "Vaccinium vitis-idaea") %>%
  dplyr::rowwise() %>%
  mutate(
    ### Control
    # n
    n.c    = ifelse(group.2 == "w/o tree roots", n, NA),
    # mean
    mean.c = ifelse(group.2 == "w/o tree roots", mean, NA),
    # sd
    sd.c   = ifelse(group.2 == "w/o tree roots", sd, NA),
    ### Experimental
    # n
    n.e    = ifelse(group.2 == "w/ tree roots", n, NA),
    # mean
    mean.e = ifelse(group.2 == "w/ tree roots", mean, NA),
    # sd
    sd.e   = ifelse(group.2 == "w/ tree roots", sd, NA)) %>%
  mutate(
    # Remove old stats columns
    mean       = NULL,
    sd         = NULL,
    n          = NULL,
    r          = NULL,
    se         = NULL,
    error_type = NULL,
    plot_type  = NULL) %>%
  # Merge control and experimental rows
  group_by(filename, variable, article.id) %>%
  dplyr::summarize(n.c     = max(n.c, na.rm = TRUE),
                   mean.c  = max(mean.c, na.rm = TRUE),
                   sd.c    = max(sd.c, na.rm = TRUE),
                   n.e     = max(n.e, na.rm = TRUE),
                   mean.e  = max(mean.e, na.rm = TRUE),
                   sd.e    = max(sd.e, na.rm = TRUE),
                   group.1 = "Vaccinium vitis-idaea",
                   group.2 = "tree roots",
                   group.3 = as.character(NA),
                   group.4 = as.character(NA)) %>%
  # Replace any missing values (Inf) with NAs
  mutate_at(., vars (-group_cols()), function(x) ifelse(is.infinite(x), NA, x))

# Gundale2012_Fig3a.png
fig.data.3 <- fig.data %>%
  filter(filename == "Gundale2012_Fig3a.png") %>%
  filter(group.1 == "Empetrum hermaphroditum") %>%
  dplyr::rowwise() %>%
  mutate(
    ### Control
    # n
    n.c    = ifelse(group.2 == "w/o tree roots", n, NA),
    # mean
    mean.c = ifelse(group.2 == "w/o tree roots", mean, NA),
    # sd
    sd.c   = ifelse(group.2 == "w/o tree roots", sd, NA),
    ### Experimental
    # n
    n.e    = ifelse(group.2 == "w/ tree roots", n, NA),
    # mean
    mean.e = ifelse(group.2 == "w/ tree roots", mean, NA),
    # sd
    sd.e   = ifelse(group.2 == "w/ tree roots", sd, NA)) %>%
  mutate(
    # Remove old stats columns
    mean       = NULL,
    sd         = NULL,
    n          = NULL,
    r          = NULL,
    se         = NULL,
    error_type = NULL,
    plot_type  = NULL) %>%
  # Merge control and experimental rows
  group_by(filename, variable, article.id) %>%
  dplyr::summarize(n.c     = max(n.c, na.rm = TRUE),
                   mean.c  = max(mean.c, na.rm = TRUE),
                   sd.c    = max(sd.c, na.rm = TRUE),
                   n.e     = max(n.e, na.rm = TRUE),
                   mean.e  = max(mean.e, na.rm = TRUE),
                   sd.e    = max(sd.e, na.rm = TRUE),
                   group.1 = "Empetrum hermaphroditum",
                   group.2 = "tree roots",
                   group.3 = as.character(NA),
                   group.4 = as.character(NA)) %>%
  # Replace any missing values (Inf) with NAs
  mutate_at(., vars (-group_cols()), function(x) ifelse(is.infinite(x), NA, x))

# Combine fig.data
working.fig.data <- working.fig.data %>%
  full_join(., fig.data.1) %>%
  full_join(., fig.data.2) %>%
  full_join(., fig.data.3)

# Remove objects
rm(fig.data.1, fig.data.2, fig.data.3)
```

# Guo 2012

```{r, echo = `FALSE`}
# Guo2012_Fig3a_short.png
fig.data.1 <- fig.data %>%
  filter(grepl("Guo2012_Fig3", filename)) %>%
  group_by(group.1, group.3) %>%
  dplyr::rowwise() %>%
  mutate(
    ### Control
    # n
    n.c    = ifelse(group.2 == "w/o neighbor", n, 4),
    # mean
    mean.c = ifelse(group.2 == "w/o neighbor", mean, NA),
    # sd
    sd.c   = ifelse(group.2 == "w/o neighbor", sd, (se * sqrt(4))),
    ### Experimental
    # n
    n.e    = ifelse(group.2 == "w/ neighbor", n, 4),
    # mean
    mean.e = ifelse(group.2 == "w/ neighbor", mean, NA),
    # sd
    sd.e   = ifelse(group.2 == "w/ neighbor", sd, (se * sqrt(4)))) %>%
  mutate(
    # Remove old stats columns
    mean       = NULL,
    sd         = NULL,
    n          = NULL,
    r          = NULL,
    se         = NULL,
    error_type = NULL,
    plot_type  = NULL) %>%
  # Merge control and experimental rows
  group_by(variable, article.id, group.1, group.3) %>%
  dplyr::summarize(filename = "Guo2012.png",
                   n.c      = max(n.c, na.rm = TRUE),
                   mean.c   = max(mean.c, na.rm = TRUE),
                   sd.c     = max(sd.c, na.rm = TRUE),
                   n.e      = max(n.e, na.rm = TRUE),
                   mean.e   = max(mean.e, na.rm = TRUE),
                   sd.e     = max(sd.e, na.rm = TRUE),
                   group.2  = "neighbor",
                   group.4  = as.character(NA)) %>%
  # Replace any missing values (Inf) with NAs
  mutate_at(., vars (-group_cols()), function(x) ifelse(is.infinite(x), NA, x))

# Combine fig.data
working.fig.data <- working.fig.data %>%
  full_join(., fig.data.1)

# Remove objects
rm(fig.data.1)
```

# Jackson 2016

```{r, echo = `FALSE`}
# Jackson2016_Fig4.png
fig.data.1 <- fig.data %>%
  filter(filename == "Jackson2016_Fig4.png") %>%
  group_by(group.1) %>%
  dplyr::rowwise() %>%
  mutate(
    ### Control
    # n
    n.c    = ifelse(group.2 == "w/o crayfish" | group.2 == "w/o crabs", n, NA),
    # mean
    mean.c = ifelse(group.2 == "w/o crayfish" | group.2 == "w/o crabs", mean, NA),
    # sd
    sd.c   = ifelse(group.2 == "w/o crayfish" | group.2 == "w/o crabs", sd, NA),
    ### Experimental
    # n
    n.e    = ifelse(group.2 == "w/ crayfish" | group.2 == "w/ crabs", n, 6),
    # mean
    mean.e = ifelse(group.2 == "w/ crayfish" | group.2 == "w/ crabs", mean, NA),
    # sd
    sd.e   = ifelse(group.2 == "w/ crayfish" | group.2 == "w/ crabs", sd, NA)) %>%
  mutate(
    # Remove old stats columns
    mean       = NULL,
    sd         = NULL,
    n          = NULL,
    r          = NULL,
    se         = NULL,
    error_type = NULL,
    plot_type  = NULL) %>%
  # Merge control and experimental rows
  group_by(filename, variable, article.id, group.1) %>%
  dplyr::summarize(n.c     = max(n.c, na.rm = TRUE),
                   mean.c  = max(mean.c, na.rm = TRUE),
                   sd.c    = max(sd.c, na.rm = TRUE),
                   n.e     = max(n.e, na.rm = TRUE),
                   mean.e  = max(mean.e, na.rm = TRUE),
                   sd.e    = max(sd.e, na.rm = TRUE),
                   group.2 = ifelse(group.1 == "Crabs", "Crayfish", "Crabs"),
                   group.3 = as.character(NA),
                   group.4 = as.character(NA)) %>%
  # Replace any missing values (Inf) with NAs
  mutate_at(., vars (-group_cols()), function(x) ifelse(is.infinite(x), NA, x)) %>%
  distinct(.)

# Combine fig.data
working.fig.data <- working.fig.data %>%
  full_join(., fig.data.1)

# Remove objects
rm(fig.data.1)
```

# Kennison 2016

```{r, echo = FALSE}
# Kenison2016_Fig2a.png
fig.data.1 <- fig.data %>%
  filter(filename == "Kenison2016_Fig2a.png") %>%
  dplyr::rowwise() %>%
  mutate(
    ### Control
    # n
    n.c    = ifelse(group.1 == "Salamander" && group.2 == "no trout", n, NA),
    # mean
    mean.c = ifelse(group.1 == "Salamander" && group.2 == "no trout", mean, NA),
    # sd
    sd.c   = ifelse(group.1 == "Salamander" && group.2 == "no trout", sd, NA),
    ### Experimental
    # n
    n.e    = ifelse(group.1 == "Salamander" && group.2 == "trout", n, NA),
    # mean
    mean.e = ifelse(group.1 == "Salamander" && group.2 == "trout", mean, NA),
    # sd
    sd.e   = ifelse(group.1 == "Salamander" && group.2 == "trout", sd, NA)) %>%
  mutate(
    # Remove old stats columns
    mean       = NULL,
    sd         = NULL,
    n          = NULL,
    r          = NULL,
    se         = NULL,
    error_type = NULL,
    plot_type  = NULL) %>%
  # Merge control and experimental rows
  group_by(filename, variable, article.id, group.1) %>%
  dplyr::summarize(n.c     = max(n.c, na.rm = TRUE),
                   mean.c  = max(mean.c, na.rm = TRUE),
                   sd.c    = max(sd.c, na.rm = TRUE),
                   n.e     = max(n.e, na.rm = TRUE),
                   mean.e  = max(mean.e, na.rm = TRUE),
                   sd.e    = max(sd.e, na.rm = TRUE),
                   group.2 = "trout",
                   group.3 = as.character(NA),
                   group.4 = as.character(NA)) %>%
  # Replace any missing values (Inf) with NAs
  mutate_at(., vars (-group_cols()), function(x) ifelse(is.infinite(x), NA, x))

# Kenison2016_Fig2b.png
fig.data.2 <- fig.data %>%
  filter(filename == "Kenison2016_Fig2b.png") %>%
  dplyr::rowwise() %>%
  mutate(
    ### Control
    # n
    n.c    = ifelse(group.1 == "Salamander" && group.2 == "no trout", n, NA),
    # mean
    mean.c = ifelse(group.1 == "Salamander" && group.2 == "no trout", mean, NA),
    # sd
    sd.c   = ifelse(group.1 == "Salamander" && group.2 == "no trout", sd, NA),
    ### Experimental
    # n
    n.e    = ifelse(group.1 == "Salamander" && group.2 == "trout", n, NA),
    # mean
    mean.e = ifelse(group.1 == "Salamander" && group.2 == "trout", mean, NA),
    # sd
    sd.e   = ifelse(group.1 == "Salamander" && group.2 == "trout", sd, NA)) %>%
  mutate(
    # Remove old stats columns
    mean       = NULL,
    sd         = NULL,
    n          = NULL,
    r          = NULL,
    se         = NULL,
    error_type = NULL,
    plot_type  = NULL) %>%
  # Merge control and experimental rows
  group_by(filename, variable, article.id, group.1) %>%
  dplyr::summarize(n.c     = max(n.c, na.rm = TRUE),
                   mean.c  = max(mean.c, na.rm = TRUE),
                   sd.c    = max(sd.c, na.rm = TRUE),
                   n.e     = max(n.e, na.rm = TRUE),
                   mean.e  = max(mean.e, na.rm = TRUE),
                   sd.e    = max(sd.e, na.rm = TRUE),
                   group.2 = "trout",
                   group.3 = as.character(NA),
                   group.4 = as.character(NA)) %>%
  # Replace any missing values (Inf) with NAs
  mutate_at(., vars (-group_cols()), function(x) ifelse(is.infinite(x), NA, x))

# Kenison2016_Fig2c.png
fig.data.3 <- fig.data %>%
  filter(filename == "Kenison2016_Fig2c.png") %>%
  dplyr::rowwise() %>%
  mutate(
    ### Control
    # n
    n.c    = ifelse(group.1 == "Salamander" && group.2 == "no trout", n, NA),
    # mean
    mean.c = ifelse(group.1 == "Salamander" && group.2 == "no trout", mean, NA),
    # sd
    sd.c   = ifelse(group.1 == "Salamander" && group.2 == "no trout", sd, NA),
    ### Experimental
    # n
    n.e    = ifelse(group.1 == "Salamander" && group.2 == "trout", n, NA),
    # mean
    mean.e = ifelse(group.1 == "Salamander" && group.2 == "trout", mean, NA),
    # sd
    sd.e   = ifelse(group.1 == "Salamander" && group.2 == "trout", sd, NA)) %>%
  mutate(
    # Remove old stats columns
    mean       = NULL,
    sd         = NULL,
    n          = NULL,
    r          = NULL,
    se         = NULL,
    error_type = NULL,
    plot_type  = NULL) %>%
  # Merge control and experimental rows
  group_by(filename, variable, article.id, group.1) %>%
  dplyr::summarize(n.c     = max(n.c, na.rm = TRUE),
                   mean.c  = max(mean.c, na.rm = TRUE),
                   sd.c    = max(sd.c, na.rm = TRUE),
                   n.e     = max(n.e, na.rm = TRUE),
                   mean.e  = max(mean.e, na.rm = TRUE),
                   sd.e    = max(sd.e, na.rm = TRUE),
                   group.2 = "trout",
                   group.3 = as.character(NA),
                   group.4 = as.character(NA)) %>%
  # Replace any missing values (Inf) with NAs
  mutate_at(., vars (-group_cols()), function(x) ifelse(is.infinite(x), NA, x))

# Kenison2016_Fig2d.png
fig.data.4 <- fig.data %>%
  filter(filename == "Kenison2016_Fig2d.png") %>%
  dplyr::rowwise() %>%
  mutate(
    ### Control
    # n
    n.c    = ifelse(group.1 == "Salamander" && group.2 == "no trout", n, NA),
    # mean
    mean.c = ifelse(group.1 == "Salamander" && group.2 == "no trout", mean, NA),
    # sd
    sd.c   = ifelse(group.1 == "Salamander" && group.2 == "no trout", sd, NA),
    ### Experimental
    # n
    n.e    = ifelse(group.1 == "Salamander" && group.2 == "trout", n, NA),
    # mean
    mean.e = ifelse(group.1 == "Salamander" && group.2 == "trout", mean, NA),
    # sd
    sd.e   = ifelse(group.1 == "Salamander" && group.2 == "trout", sd, NA)) %>%
  mutate(
    # Remove old stats columns
    mean       = NULL,
    sd         = NULL,
    n          = NULL,
    r          = NULL,
    se         = NULL,
    error_type = NULL,
    plot_type  = NULL) %>%
  # Merge control and experimental rows
  group_by(filename, variable, article.id, group.1) %>%
  dplyr::summarize(n.c     = max(n.c, na.rm = TRUE),
                   mean.c  = max(mean.c, na.rm = TRUE),
                   sd.c    = max(sd.c, na.rm = TRUE),
                   n.e     = max(n.e, na.rm = TRUE),
                   mean.e  = max(mean.e, na.rm = TRUE),
                   sd.e    = max(sd.e, na.rm = TRUE),
                   group.2 = "trout",
                   group.3 = as.character(NA),
                   group.4 = as.character(NA)) %>%
  # Replace any missing values (Inf) with NAs
  mutate_at(., vars (-group_cols()), function(x) ifelse(is.infinite(x), NA, x))

# Kenison2016_Fig2e.png
fig.data.5 <- fig.data %>%
  filter(filename == "Kenison2016_Fig2e.png") %>%
  dplyr::rowwise() %>%
  mutate(
    ### Control
    # n
    n.c    = ifelse(group.1 == "Salamander" && group.2 == "no trout", n, NA),
    # mean
    mean.c = ifelse(group.1 == "Salamander" && group.2 == "no trout", mean, NA),
    # sd
    sd.c   = ifelse(group.1 == "Salamander" && group.2 == "no trout", sd, NA),
    ### Experimental
    # n
    n.e    = ifelse(group.1 == "Salamander" && group.2 == "trout", n, NA),
    # mean
    mean.e = ifelse(group.1 == "Salamander" && group.2 == "trout", mean, NA),
    # sd
    sd.e   = ifelse(group.1 == "Salamander" && group.2 == "trout", sd, NA)) %>%
  mutate(
    # Remove old stats columns
    mean       = NULL,
    sd         = NULL,
    n          = NULL,
    r          = NULL,
    se         = NULL,
    error_type = NULL,
    plot_type  = NULL) %>%
  # Merge control and experimental rows
  group_by(filename, variable, article.id, group.1) %>%
  dplyr::summarize(n.c     = max(n.c, na.rm = TRUE),
                   mean.c  = max(mean.c, na.rm = TRUE),
                   sd.c    = max(sd.c, na.rm = TRUE),
                   n.e     = max(n.e, na.rm = TRUE),
                   mean.e  = max(mean.e, na.rm = TRUE),
                   sd.e    = max(sd.e, na.rm = TRUE),
                   group.2 = "trout",
                   group.3 = as.character(NA),
                   group.4 = as.character(NA)) %>%
  # Replace any missing values (Inf) with NAs
  mutate_at(., vars (-group_cols()), function(x) ifelse(is.infinite(x), NA, x))

# Kenison2016_Fig2f.png
fig.data.6 <- fig.data %>%
  filter(filename == "Kenison2016_Fig2f.png") %>%
  dplyr::rowwise() %>%
  mutate(
    ### Control
    # n
    n.c    = ifelse(group.1 == "Salamander" && group.2 == "no trout", n, NA),
    # mean
    mean.c = ifelse(group.1 == "Salamander" && group.2 == "no trout", mean, NA),
    # sd
    sd.c   = ifelse(group.1 == "Salamander" && group.2 == "no trout", sd, NA),
    ### Experimental
    # n
    n.e    = ifelse(group.1 == "Salamander" && group.2 == "trout", n, NA),
    # mean
    mean.e = ifelse(group.1 == "Salamander" && group.2 == "trout", mean, NA),
    # sd
    sd.e   = ifelse(group.1 == "Salamander" && group.2 == "trout", sd, NA)) %>%
  mutate(
    # Remove old stats columns
    mean       = NULL,
    sd         = NULL,
    n          = NULL,
    r          = NULL,
    se         = NULL,
    error_type = NULL,
    plot_type  = NULL) %>%
  # Merge control and experimental rows
  group_by(filename, variable, article.id, group.1) %>%
  dplyr::summarize(n.c     = max(n.c, na.rm = TRUE),
                   mean.c  = max(mean.c, na.rm = TRUE),
                   sd.c    = max(sd.c, na.rm = TRUE),
                   n.e     = max(n.e, na.rm = TRUE),
                   mean.e  = max(mean.e, na.rm = TRUE),
                   sd.e    = max(sd.e, na.rm = TRUE),
                   group.2 = "trout",
                   group.3 = as.character(NA),
                   group.4 = as.character(NA)) %>%
  # Replace any missing values (Inf) with NAs
  mutate_at(., vars (-group_cols()), function(x) ifelse(is.infinite(x), NA, x))

# Combine fig.data
working.fig.data <- working.fig.data %>%
  full_join(., fig.data.1) %>%
  full_join(., fig.data.2) %>%
  full_join(., fig.data.3) %>%
  full_join(., fig.data.4) %>%
  full_join(., fig.data.5) %>%
  full_join(., fig.data.6)

# Remove objects
rm(fig.data.1, fig.data.2, fig.data.3,
   fig.data.4, fig.data.5, fig.data.6)
```

# Marino 2013

```{r, echo = FALSE}
# Marino2013_Fig1a.png
fig.data.1 <- fig.data %>%
  filter(filename == "Marino2013_Fig1a.png") %>%
  dplyr::rowwise() %>%
  mutate(
    ### Control
    # n
    n.c    = ifelse(group.1 == "Green frog larvae" && group.2 == "no predator" && group.3 == "no parasite", n, NA),
    # mean
    mean.c = ifelse(group.1 == "Green frog larvae" && group.2 == "no predator" && group.3 == "no parasite", mean, NA),
    # sd
    sd.c   = ifelse(group.1 == "Green frog larvae" && group.2 == "no predator" && group.3 == "no parasite", sd, NA),
    ### Experimental
    # n
    n.e    = ifelse(group.1 == "Green frog larvae" && group.2 == "free predator" && group.3 == "no parasite", n, NA),
    # mean
    mean.e = ifelse(group.1 == "Green frog larvae" && group.2 == "free predator" && group.3 == "no parasite", mean, NA),
    # sd
    sd.e   = ifelse(group.1 == "Green frog larvae" && group.2 == "free predator" && group.3 == "no parasite", sd, NA)) %>%
  mutate(
    # Remove old stats columns
    mean       = NULL,
    sd         = NULL,
    n          = NULL,
    r          = NULL,
    se         = NULL,
    error_type = NULL,
    plot_type  = NULL) %>%
  # Merge control and experimental rows
  group_by(filename, variable, article.id, group.1) %>%
  dplyr::summarize(n.c     = max(n.c, na.rm = TRUE),
                   mean.c  = max(mean.c, na.rm = TRUE),
                   sd.c    = max(sd.c, na.rm = TRUE),
                   n.e     = max(n.e, na.rm = TRUE),
                   mean.e  = max(mean.e, na.rm = TRUE),
                   sd.e    = max(sd.e, na.rm = TRUE),
                   group.2 = "Anax",
                   group.3 = as.character(NA),
                   group.4 = as.character(NA)) %>%
  # Replace any missing values (Inf) with NAs
  mutate_at(., vars (-group_cols()), function(x) ifelse(is.infinite(x), NA, x))

## Marino2013_Fig1b.png
#fig.data.2 <- fig.data %>%
#  filter(filename == "Marino2013_Fig1b.png") %>%
#  dplyr::rowwise() %>%
#  mutate(
#    ### Control
#    # n
#    n.c    = ifelse(group.1 == "Green frog larvae" && group.2 == "caged predator" && group.3 == "no parasite", n, NA),
#    # mean
#    mean.c = ifelse(group.1 == "Green frog larvae" && group.2 == "caged predator" && group.3 == "no parasite", mean, NA),
#    # sd
#    sd.c   = ifelse(group.1 == "Green frog larvae" && group.2 == "caged predator" && group.3 == "no parasite", sd, NA),
#    ### Experimental
#    # n
#    n.e    = ifelse(group.1 == "Green frog larvae" && group.2 == "free predator" && group.3 == "no parasite", n, NA),
#    # mean
#    mean.e = ifelse(group.1 == "Green frog larvae" && group.2 == "free predator" && group.3 == "no parasite", mean, NA),
#    # sd
#    sd.e   = ifelse(group.1 == "Green frog larvae" && group.2 == "free predator" && group.3 == "no parasite", sd, NA)) %>%
#  mutate(
#    # Remove old stats columns
#    mean       = NULL,
#    sd         = NULL,
#    n          = NULL,
#    r          = NULL,
#    se         = NULL,
#    error_type = NULL,
#    plot_type  = NULL) %>%
#  # Merge control and experimental rows
#  group_by(filename, variable, article.id, group.1) %>%
#  dplyr::summarize(n.c     = max(n.c, na.rm = TRUE),
#                   mean.c  = max(mean.c, na.rm = TRUE),
#                   sd.c    = max(sd.c, na.rm = TRUE),
#                   n.e     = max(n.e, na.rm = TRUE),
#                   mean.e  = max(mean.e, na.rm = TRUE),
#                   sd.e    = max(sd.e, na.rm = TRUE),
#                   group.2 = "Anax",
#                   group.3 = as.character(NA),
#                   group.4 = as.character(NA)) %>%
#  # Replace any missing values (Inf) with NAs
#  mutate_at(., vars (-group_cols()), function(x) ifelse(is.infinite(x), NA, x))

# Combine fig.data
working.fig.data <- working.fig.data %>%
  full_join(., fig.data.1)

# Remove objects
rm(fig.data.1)
```

# Martinez 2015

```{r, echo = FALSE}
# Martinez2015_Fig2a.png
fig.data.1 <- fig.data %>%
  filter(filename == "Martinez2015_Fig2a.png") %>%
  dplyr::rowwise() %>%
  mutate(
    ### Control
    # n
    n.c    = ifelse(group.1 == "Acacia myrtifolia" && group.2 == "no ant" && group.4 == "Croajingolong", n, NA),
    # mean
    mean.c = ifelse(group.1 == "Acacia myrtifolia" && group.2 == "no ant" && group.4 == "Croajingolong", mean, NA),
    # sd
    sd.c   = ifelse(group.1 == "Acacia myrtifolia" && group.2 == "no ant" && group.4 == "Croajingolong", sd, NA),
    ### Experimental
    # n
    n.e    = ifelse(group.1 == "Acacia myrtifolia" && group.2 == "ant" && group.4 == "Croajingolong", n, NA),
    # mean
    mean.e = ifelse(group.1 == "Acacia myrtifolia" && group.2 == "ant" && group.4 == "Croajingolong", mean, NA),
    # sd
    sd.e   = ifelse(group.1 == "Acacia myrtifolia" && group.2 == "ant" && group.4 == "Croajingolong", sd, NA)) %>%
  mutate(
    # Remove old stats columns
    mean       = NULL,
    sd         = NULL,
    n          = NULL,
    r          = NULL,
    se         = NULL,
    error_type = NULL,
    plot_type  = NULL) %>%
  filter(group.4 == "Croajingolong") %>%
  # Merge control and experimental rows
  group_by(filename, variable, article.id, group.1) %>%
  dplyr::summarize(n.c     = max(n.c, na.rm = TRUE),
                   mean.c  = max(mean.c, na.rm = TRUE),
                   sd.c    = max(sd.c, na.rm = TRUE),
                   n.e     = max(n.e, na.rm = TRUE),
                   mean.e  = max(mean.e, na.rm = TRUE),
                   sd.e    = max(sd.e, na.rm = TRUE),
                   group.2 = "ant",
                   group.3 = "open pollination",
                   group.4 = "Croajingolong") %>%
  # Replace any missing values (Inf) with NAs
  mutate_at(., vars (-group_cols()), function(x) ifelse(is.infinite(x), NA, x))

# Martinez2015_Fig2a.png
fig.data.2 <- fig.data %>%
  filter(filename == "Martinez2015_Fig2a.png") %>%
  dplyr::rowwise() %>%
  mutate(
    ### Control
    # n
    n.c    = ifelse(group.1 == "Acacia myrtifolia" && group.2 == "no ant" && group.4 == "Grampians", n, NA),
    # mean
    mean.c = ifelse(group.1 == "Acacia myrtifolia" && group.2 == "no ant" && group.4 == "Grampians", mean, NA),
    # sd
    sd.c   = ifelse(group.1 == "Acacia myrtifolia" && group.2 == "no ant" && group.4 == "Grampians", sd, NA),
    ### Experimental
    # n
    n.e    = ifelse(group.1 == "Acacia myrtifolia" && group.2 == "ant" && group.4 == "Grampians", n, NA),
    # mean
    mean.e = ifelse(group.1 == "Acacia myrtifolia" && group.2 == "ant" && group.4 == "Grampians", mean, NA),
    # sd
    sd.e   = ifelse(group.1 == "Acacia myrtifolia" && group.2 == "ant" && group.4 == "Grampians", sd, NA)) %>%
  mutate(
    # Remove old stats columns
    mean       = NULL,
    sd         = NULL,
    n          = NULL,
    r          = NULL,
    se         = NULL,
    error_type = NULL,
    plot_type  = NULL) %>%
  filter(group.4 == "Grampians") %>%
  # Merge control and experimental rows
  group_by(filename, variable, article.id, group.1) %>%
  dplyr::summarize(n.c     = max(n.c, na.rm = TRUE),
                   mean.c  = max(mean.c, na.rm = TRUE),
                   sd.c    = max(sd.c, na.rm = TRUE),
                   n.e     = max(n.e, na.rm = TRUE),
                   mean.e  = max(mean.e, na.rm = TRUE),
                   sd.e    = max(sd.e, na.rm = TRUE),
                   group.2 = "ant",
                   group.3 = "open pollination",
                   group.4 = "Grampians") %>%
  # Replace any missing values (Inf) with NAs
  mutate_at(., vars (-group_cols()), function(x) ifelse(is.infinite(x), NA, x))

# Martinez2015_Fig2a.png
fig.data.3 <- fig.data %>%
  filter(filename == "Martinez2015_Fig2a.png") %>%
  dplyr::rowwise() %>%
  mutate(
    ### Control
    # n
    n.c    = ifelse(group.1 == "Acacia myrtifolia" && group.2 == "no ant" && group.4 == "Brisbane Ranges", n, NA),
    # mean
    mean.c = ifelse(group.1 == "Acacia myrtifolia" && group.2 == "no ant" && group.4 == "Brisbane Ranges", mean, NA),
    # sd
    sd.c   = ifelse(group.1 == "Acacia myrtifolia" && group.2 == "no ant" && group.4 == "Brisbane Ranges", sd, NA),
    ### Experimental
    # n
    n.e    = ifelse(group.1 == "Acacia myrtifolia" && group.2 == "ant" && group.4 == "Brisbane Ranges", n, NA),
    # mean
    mean.e = ifelse(group.1 == "Acacia myrtifolia" && group.2 == "ant" && group.4 == "Brisbane Ranges", mean, NA),
    # sd
    sd.e   = ifelse(group.1 == "Acacia myrtifolia" && group.2 == "ant" && group.4 == "Brisbane Ranges", sd, NA)) %>%
  mutate(
    # Remove old stats columns
    mean       = NULL,
    sd         = NULL,
    n          = NULL,
    r          = NULL,
    se         = NULL,
    error_type = NULL,
    plot_type  = NULL) %>%
  filter(group.4 == "Brisbane Ranges") %>%
  # Merge control and experimental rows
  group_by(filename, variable, article.id, group.1) %>%
  dplyr::summarize(n.c     = max(n.c, na.rm = TRUE),
                   mean.c  = max(mean.c, na.rm = TRUE),
                   sd.c    = max(sd.c, na.rm = TRUE),
                   n.e     = max(n.e, na.rm = TRUE),
                   mean.e  = max(mean.e, na.rm = TRUE),
                   sd.e    = max(sd.e, na.rm = TRUE),
                   group.2 = "ant",
                   group.3 = "open pollination",
                   group.4 = "Brisbane Ranges") %>%
  # Replace any missing values (Inf) with NAs
  mutate_at(., vars (-group_cols()), function(x) ifelse(is.infinite(x), NA, x))

# Martinez2015_Fig2a.png
fig.data.4 <- fig.data %>%
  filter(filename == "Martinez2015_Fig2a.png") %>%
  dplyr::rowwise() %>%
  mutate(
    ### Control
    # n
    n.c    = ifelse(group.1 == "Acacia myrtifolia" && group.2 == "no ant" && group.4 == "Dandenong Ranges", n, NA),
    # mean
    mean.c = ifelse(group.1 == "Acacia myrtifolia" && group.2 == "no ant" && group.4 == "Dandenong Ranges", mean, NA),
    # sd
    sd.c   = ifelse(group.1 == "Acacia myrtifolia" && group.2 == "no ant" && group.4 == "Dandenong Ranges", sd, NA),
    ### Experimental
    # n
    n.e    = ifelse(group.1 == "Acacia myrtifolia" && group.2 == "ant" && group.4 == "Dandenong Ranges", n, NA),
    # mean
    mean.e = ifelse(group.1 == "Acacia myrtifolia" && group.2 == "ant" && group.4 == "Dandenong Ranges", mean, NA),
    # sd
    sd.e   = ifelse(group.1 == "Acacia myrtifolia" && group.2 == "ant" && group.4 == "Dandenong Ranges", sd, NA)) %>%
  mutate(
    # Remove old stats columns
    mean       = NULL,
    sd         = NULL,
    n          = NULL,
    r          = NULL,
    se         = NULL,
    error_type = NULL,
    plot_type  = NULL) %>%
  filter(group.4 == "Dandenong Ranges") %>%
  # Merge control and experimental rows
  group_by(filename, variable, article.id, group.1) %>%
  dplyr::summarize(n.c     = max(n.c, na.rm = TRUE),
                   mean.c  = max(mean.c, na.rm = TRUE),
                   sd.c    = max(sd.c, na.rm = TRUE),
                   n.e     = max(n.e, na.rm = TRUE),
                   mean.e  = max(mean.e, na.rm = TRUE),
                   sd.e    = max(sd.e, na.rm = TRUE),
                   group.2 = "ant",
                   group.3 = "open pollination",
                   group.4 = "Dandenong Ranges") %>%
  # Replace any missing values (Inf) with NAs
  mutate_at(., vars (-group_cols()), function(x) ifelse(is.infinite(x), NA, x))


# Combine fig.data
working.fig.data <- rbind(working.fig.data,
                          fig.data.1, fig.data.2,
                          fig.data.3, fig.data.4)

# Remove objects
rm(fig.data.1, fig.data.2, fig.data.3, fig.data.4)
```

```{r, echo = FALSE}
# Martinez2015_Fig2b.png
fig.data.1 <- fig.data %>%
  filter(filename == "Martinez2015_Fig2b.png") %>%
  dplyr::rowwise() %>%
  mutate(
    ### Control
    # n
    n.c    = ifelse(group.1 == "Acacia myrtifolia" && group.2 == "no ant" && group.4 == "Croajingolong", n, NA),
    # mean
    mean.c = ifelse(group.1 == "Acacia myrtifolia" && group.2 == "no ant" && group.4 == "Croajingolong", mean, NA),
    # sd
    sd.c   = ifelse(group.1 == "Acacia myrtifolia" && group.2 == "no ant" && group.4 == "Croajingolong", sd, NA),
    ### Experimental
    # n
    n.e    = ifelse(group.1 == "Acacia myrtifolia" && group.2 == "ant" && group.4 == "Croajingolong", n, NA),
    # mean
    mean.e = ifelse(group.1 == "Acacia myrtifolia" && group.2 == "ant" && group.4 == "Croajingolong", mean, NA),
    # sd
    sd.e   = ifelse(group.1 == "Acacia myrtifolia" && group.2 == "ant" && group.4 == "Croajingolong", sd, NA)) %>%
  mutate(
    # Remove old stats columns
    mean       = NULL,
    sd         = NULL,
    n          = NULL,
    r          = NULL,
    se         = NULL,
    error_type = NULL,
    plot_type  = NULL) %>%
  filter(group.4 == "Croajingolong") %>%
  # Merge control and experimental rows
  group_by(filename, variable, article.id, group.1) %>%
  dplyr::summarize(n.c     = max(n.c, na.rm = TRUE),
                   mean.c  = max(mean.c, na.rm = TRUE),
                   sd.c    = max(sd.c, na.rm = TRUE),
                   n.e     = max(n.e, na.rm = TRUE),
                   mean.e  = max(mean.e, na.rm = TRUE),
                   sd.e    = max(sd.e, na.rm = TRUE),
                   group.2 = "ant",
                   group.3 = "all pollination exclusion",
                   group.4 = "Croajingolong") %>%
  # Replace any missing values (Inf) with NAs
  mutate_at(., vars (-group_cols()), function(x) ifelse(is.infinite(x), NA, x))

# Martinez2015_Fig2b.png
fig.data.2 <- fig.data %>%
  filter(filename == "Martinez2015_Fig2b.png") %>%
  dplyr::rowwise() %>%
  mutate(
    ### Control
    # n
    n.c    = ifelse(group.1 == "Acacia myrtifolia" && group.2 == "no ant" && group.4 == "Grampians", n, NA),
    # mean
    mean.c = ifelse(group.1 == "Acacia myrtifolia" && group.2 == "no ant" && group.4 == "Grampians", mean, NA),
    # sd
    sd.c   = ifelse(group.1 == "Acacia myrtifolia" && group.2 == "no ant" && group.4 == "Grampians", sd, NA),
    ### Experimental
    # n
    n.e    = ifelse(group.1 == "Acacia myrtifolia" && group.2 == "ant" && group.4 == "Grampians", n, NA),
    # mean
    mean.e = ifelse(group.1 == "Acacia myrtifolia" && group.2 == "ant" && group.4 == "Grampians", mean, NA),
    # sd
    sd.e   = ifelse(group.1 == "Acacia myrtifolia" && group.2 == "ant" && group.4 == "Grampians", sd, NA)) %>%
  mutate(
    # Remove old stats columns
    mean       = NULL,
    sd         = NULL,
    n          = NULL,
    r          = NULL,
    se         = NULL,
    error_type = NULL,
    plot_type  = NULL) %>%
  filter(group.4 == "Grampians") %>%
  # Merge control and experimental rows
  group_by(filename, variable, article.id, group.1) %>%
  dplyr::summarize(n.c     = max(n.c, na.rm = TRUE),
                   mean.c  = max(mean.c, na.rm = TRUE),
                   sd.c    = max(sd.c, na.rm = TRUE),
                   n.e     = max(n.e, na.rm = TRUE),
                   mean.e  = max(mean.e, na.rm = TRUE),
                   sd.e    = max(sd.e, na.rm = TRUE),
                   group.2 = "ant",
                   group.3 = "all pollination exclusion",
                   group.4 = "Grampians") %>%
  # Replace any missing values (Inf) with NAs
  mutate_at(., vars (-group_cols()), function(x) ifelse(is.infinite(x), NA, x))

# Martinez2015_Fig2b.png
fig.data.3 <- fig.data %>%
  filter(filename == "Martinez2015_Fig2b.png") %>%
  dplyr::rowwise() %>%
  mutate(
    ### Control
    # n
    n.c    = ifelse(group.1 == "Acacia myrtifolia" && group.2 == "no ant" && group.4 == "Brisbane Ranges", n, NA),
    # mean
    mean.c = ifelse(group.1 == "Acacia myrtifolia" && group.2 == "no ant" && group.4 == "Brisbane Ranges", mean, NA),
    # sd
    sd.c   = ifelse(group.1 == "Acacia myrtifolia" && group.2 == "no ant" && group.4 == "Brisbane Ranges", sd, NA),
    ### Experimental
    # n
    n.e    = ifelse(group.1 == "Acacia myrtifolia" && group.2 == "ant" && group.4 == "Brisbane Ranges", n, NA),
    # mean
    mean.e = ifelse(group.1 == "Acacia myrtifolia" && group.2 == "ant" && group.4 == "Brisbane Ranges", mean, NA),
    # sd
    sd.e   = ifelse(group.1 == "Acacia myrtifolia" && group.2 == "ant" && group.4 == "Brisbane Ranges", sd, NA)) %>%
  mutate(
    # Remove old stats columns
    mean       = NULL,
    sd         = NULL,
    n          = NULL,
    r          = NULL,
    se         = NULL,
    error_type = NULL,
    plot_type  = NULL) %>%
  filter(group.4 == "Brisbane Ranges") %>%
  # Merge control and experimental rows
  group_by(filename, variable, article.id, group.1) %>%
  dplyr::summarize(n.c     = max(n.c, na.rm = TRUE),
                   mean.c  = max(mean.c, na.rm = TRUE),
                   sd.c    = max(sd.c, na.rm = TRUE),
                   n.e     = max(n.e, na.rm = TRUE),
                   mean.e  = max(mean.e, na.rm = TRUE),
                   sd.e    = max(sd.e, na.rm = TRUE),
                   group.2 = "ant",
                   group.3 = "all pollination exclusion",
                   group.4 = "Brisbane Ranges") %>%
  # Replace any missing values (Inf) with NAs
  mutate_at(., vars (-group_cols()), function(x) ifelse(is.infinite(x), NA, x))

# Martinez2015_Fig2b.png
fig.data.4 <- fig.data %>%
  filter(filename == "Martinez2015_Fig2b.png") %>%
  dplyr::rowwise() %>%
  mutate(
    ### Control
    # n
    n.c    = ifelse(group.1 == "Acacia myrtifolia" && group.2 == "no ant" && group.4 == "Dandenong Ranges", n, NA),
    # mean
    mean.c = ifelse(group.1 == "Acacia myrtifolia" && group.2 == "no ant" && group.4 == "Dandenong Ranges", mean, NA),
    # sd
    sd.c   = ifelse(group.1 == "Acacia myrtifolia" && group.2 == "no ant" && group.4 == "Dandenong Ranges", sd, NA),
    ### Experimental
    # n
    n.e    = ifelse(group.1 == "Acacia myrtifolia" && group.2 == "ant" && group.4 == "Dandenong Ranges", n, NA),
    # mean
    mean.e = ifelse(group.1 == "Acacia myrtifolia" && group.2 == "ant" && group.4 == "Dandenong Ranges", mean, NA),
    # sd
    sd.e   = ifelse(group.1 == "Acacia myrtifolia" && group.2 == "ant" && group.4 == "Dandenong Ranges", sd, NA)) %>%
  mutate(
    # Remove old stats columns
    mean       = NULL,
    sd         = NULL,
    n          = NULL,
    r          = NULL,
    se         = NULL,
    error_type = NULL,
    plot_type  = NULL) %>%
  filter(group.4 == "Dandenong Ranges") %>%
  # Merge control and experimental rows
  group_by(filename, variable, article.id, group.1) %>%
  dplyr::summarize(n.c     = max(n.c, na.rm = TRUE),
                   mean.c  = max(mean.c, na.rm = TRUE),
                   sd.c    = max(sd.c, na.rm = TRUE),
                   n.e     = max(n.e, na.rm = TRUE),
                   mean.e  = max(mean.e, na.rm = TRUE),
                   sd.e    = max(sd.e, na.rm = TRUE),
                   group.2 = "ant",
                   group.3 = "all pollination exclusion",
                   group.4 = "Dandenong Ranges") %>%
  # Replace any missing values (Inf) with NAs
  mutate_at(., vars (-group_cols()), function(x) ifelse(is.infinite(x), NA, x))


# Combine fig.data
working.fig.data <- rbind(working.fig.data,
                          fig.data.1, fig.data.2,
                          fig.data.3, fig.data.4)

# Remove objects
rm(fig.data.1, fig.data.2, fig.data.3, fig.data.4)
```

```{r, echo = FALSE}
# Martinez2015_Fig2c.png
fig.data.1 <- fig.data %>%
  filter(filename == "Martinez2015_Fig2c.png") %>%
  dplyr::rowwise() %>%
  mutate(
    ### Control
    # n
    n.c    = ifelse(group.1 == "Acacia myrtifolia" && group.2 == "no ant" && group.4 == "Croajingolong", n, NA),
    # mean
    mean.c = ifelse(group.1 == "Acacia myrtifolia" && group.2 == "no ant" && group.4 == "Croajingolong", mean, NA),
    # sd
    sd.c   = ifelse(group.1 == "Acacia myrtifolia" && group.2 == "no ant" && group.4 == "Croajingolong", sd, NA),
    ### Experimental
    # n
    n.e    = ifelse(group.1 == "Acacia myrtifolia" && group.2 == "ant" && group.4 == "Croajingolong", n, NA),
    # mean
    mean.e = ifelse(group.1 == "Acacia myrtifolia" && group.2 == "ant" && group.4 == "Croajingolong", mean, NA),
    # sd
    sd.e   = ifelse(group.1 == "Acacia myrtifolia" && group.2 == "ant" && group.4 == "Croajingolong", sd, NA)) %>%
  mutate(
    # Remove old stats columns
    mean       = NULL,
    sd         = NULL,
    n          = NULL,
    r          = NULL,
    se         = NULL,
    error_type = NULL,
    plot_type  = NULL) %>%
  filter(group.4 == "Croajingolong") %>%
  # Merge control and experimental rows
  group_by(filename, variable, article.id, group.1) %>%
  dplyr::summarize(n.c     = max(n.c, na.rm = TRUE),
                   mean.c  = max(mean.c, na.rm = TRUE),
                   sd.c    = max(sd.c, na.rm = TRUE),
                   n.e     = max(n.e, na.rm = TRUE),
                   mean.e  = max(mean.e, na.rm = TRUE),
                   sd.e    = max(sd.e, na.rm = TRUE),
                   group.2 = "ant",
                   group.3 = "large pollination exclusion",
                   group.4 = "Croajingolong") %>%
  # Replace any missing values (Inf) with NAs
  mutate_at(., vars (-group_cols()), function(x) ifelse(is.infinite(x), NA, x))

# Martinez2015_Fig2c.png
fig.data.2 <- fig.data %>%
  filter(filename == "Martinez2015_Fig2c.png") %>%
  dplyr::rowwise() %>%
  mutate(
    ### Control
    # n
    n.c    = ifelse(group.1 == "Acacia myrtifolia" && group.2 == "no ant" && group.4 == "Grampians", n, NA),
    # mean
    mean.c = ifelse(group.1 == "Acacia myrtifolia" && group.2 == "no ant" && group.4 == "Grampians", mean, NA),
    # sd
    sd.c   = ifelse(group.1 == "Acacia myrtifolia" && group.2 == "no ant" && group.4 == "Grampians", sd, NA),
    ### Experimental
    # n
    n.e    = ifelse(group.1 == "Acacia myrtifolia" && group.2 == "ant" && group.4 == "Grampians", n, NA),
    # mean
    mean.e = ifelse(group.1 == "Acacia myrtifolia" && group.2 == "ant" && group.4 == "Grampians", mean, NA),
    # sd
    sd.e   = ifelse(group.1 == "Acacia myrtifolia" && group.2 == "ant" && group.4 == "Grampians", sd, NA)) %>%
  mutate(
    # Remove old stats columns
    mean       = NULL,
    sd         = NULL,
    n          = NULL,
    r          = NULL,
    se         = NULL,
    error_type = NULL,
    plot_type  = NULL) %>%
  filter(group.4 == "Grampians") %>%
  # Merge control and experimental rows
  group_by(filename, variable, article.id, group.1) %>%
  dplyr::summarize(n.c     = max(n.c, na.rm = TRUE),
                   mean.c  = max(mean.c, na.rm = TRUE),
                   sd.c    = max(sd.c, na.rm = TRUE),
                   n.e     = max(n.e, na.rm = TRUE),
                   mean.e  = max(mean.e, na.rm = TRUE),
                   sd.e    = max(sd.e, na.rm = TRUE),
                   group.2 = "ant",
                   group.3 = "large pollination exclusion",
                   group.4 = "Grampians") %>%
  # Replace any missing values (Inf) with NAs
  mutate_at(., vars (-group_cols()), function(x) ifelse(is.infinite(x), NA, x))

# Martinez2015_Fig2c.png
fig.data.3 <- fig.data %>%
  filter(filename == "Martinez2015_Fig2c.png") %>%
  dplyr::rowwise() %>%
  mutate(
    ### Control
    # n
    n.c    = ifelse(group.1 == "Acacia myrtifolia" && group.2 == "no ant" && group.4 == "Brisbane Ranges", n, NA),
    # mean
    mean.c = ifelse(group.1 == "Acacia myrtifolia" && group.2 == "no ant" && group.4 == "Brisbane Ranges", mean, NA),
    # sd
    sd.c   = ifelse(group.1 == "Acacia myrtifolia" && group.2 == "no ant" && group.4 == "Brisbane Ranges", sd, NA),
    ### Experimental
    # n
    n.e    = ifelse(group.1 == "Acacia myrtifolia" && group.2 == "ant" && group.4 == "Brisbane Ranges", n, NA),
    # mean
    mean.e = ifelse(group.1 == "Acacia myrtifolia" && group.2 == "ant" && group.4 == "Brisbane Ranges", mean, NA),
    # sd
    sd.e   = ifelse(group.1 == "Acacia myrtifolia" && group.2 == "ant" && group.4 == "Brisbane Ranges", sd, NA)) %>%
  mutate(
    # Remove old stats columns
    mean       = NULL,
    sd         = NULL,
    n          = NULL,
    r          = NULL,
    se         = NULL,
    error_type = NULL,
    plot_type  = NULL) %>%
  filter(group.4 == "Brisbane Ranges") %>%
  # Merge control and experimental rows
  group_by(filename, variable, article.id, group.1) %>%
  dplyr::summarize(n.c     = max(n.c, na.rm = TRUE),
                   mean.c  = max(mean.c, na.rm = TRUE),
                   sd.c    = max(sd.c, na.rm = TRUE),
                   n.e     = max(n.e, na.rm = TRUE),
                   mean.e  = max(mean.e, na.rm = TRUE),
                   sd.e    = max(sd.e, na.rm = TRUE),
                   group.2 = "ant",
                   group.3 = "large pollination exclusion",
                   group.4 = "Brisbane Ranges") %>%
  # Replace any missing values (Inf) with NAs
  mutate_at(., vars (-group_cols()), function(x) ifelse(is.infinite(x), NA, x))

# Martinez2015_Fig2c.png
fig.data.4 <- fig.data %>%
  filter(filename == "Martinez2015_Fig2c.png") %>%
  dplyr::rowwise() %>%
  mutate(
    ### Control
    # n
    n.c    = ifelse(group.1 == "Acacia myrtifolia" && group.2 == "no ant" && group.4 == "Dandenong Ranges", n, NA),
    # mean
    mean.c = ifelse(group.1 == "Acacia myrtifolia" && group.2 == "no ant" && group.4 == "Dandenong Ranges", mean, NA),
    # sd
    sd.c   = ifelse(group.1 == "Acacia myrtifolia" && group.2 == "no ant" && group.4 == "Dandenong Ranges", sd, NA),
    ### Experimental
    # n
    n.e    = ifelse(group.1 == "Acacia myrtifolia" && group.2 == "ant" && group.4 == "Dandenong Ranges", n, NA),
    # mean
    mean.e = ifelse(group.1 == "Acacia myrtifolia" && group.2 == "ant" && group.4 == "Dandenong Ranges", mean, NA),
    # sd
    sd.e   = ifelse(group.1 == "Acacia myrtifolia" && group.2 == "ant" && group.4 == "Dandenong Ranges", sd, NA)) %>%
  mutate(
    # Remove old stats columns
    mean       = NULL,
    sd         = NULL,
    n          = NULL,
    r          = NULL,
    se         = NULL,
    error_type = NULL,
    plot_type  = NULL) %>%
  filter(group.4 == "Dandenong Ranges") %>%
  # Merge control and experimental rows
  group_by(filename, variable, article.id, group.1) %>%
  dplyr::summarize(n.c     = max(n.c, na.rm = TRUE),
                   mean.c  = max(mean.c, na.rm = TRUE),
                   sd.c    = max(sd.c, na.rm = TRUE),
                   n.e     = max(n.e, na.rm = TRUE),
                   mean.e  = max(mean.e, na.rm = TRUE),
                   sd.e    = max(sd.e, na.rm = TRUE),
                   group.2 = "ant",
                   group.3 = "large pollination exclusion",
                   group.4 = "Dandenong Ranges") %>%
  # Replace any missing values (Inf) with NAs
  mutate_at(., vars (-group_cols()), function(x) ifelse(is.infinite(x), NA, x))


# Combine fig.data
working.fig.data <- rbind(working.fig.data,
                          fig.data.1, fig.data.2,
                          fig.data.3, fig.data.4)

# Remove objects
rm(fig.data.1, fig.data.2, fig.data.3, fig.data.4)
```

### Step 5: Figz

```{r, echo = `FALSE`}
# Calculate effect sizes (yi = es, vi = variance)
es <- working.fig.data %>%
  metafor::escalc(data = .,
                  measure = "SMD",
                  m1i  = mean.e,
                  sd1i = sd.e,
                  n1i  = n.e,
                  m2i  = mean.c,
                  sd2i = sd.c,
                  n2i  = n.c) %>%
  dplyr::rename(es     = yi,
                es.var = vi) %>%
  filter(!is.na(es))

###### Categorize studies

# Plant
comp.plant <- c("Bontrager2014_Fig1.png", "Bontrager2014_Fig1.png",
                "Bontrager2014_Fig1.png", "Bontrager2014_Fig5a.png",
                "Bontrager2014_Fig5a.png", "Bontrager2014_Fig5b.png",
                "Bontrager2014_Fig5b.png", "Bontrager2014_Fig5b.png",
                "Bontrager2014_Fig5c.png", "Bontrager2014_Fig5d.png",
                "Cipriotti2010_Fig1.png", "Cipriotti2010_Fig3.png",
                "Gundale2012_Fig2a.png", "Gundale2012_Fig2b.png",
                "Gundale2012_Fig2c.png", "Gundale2012_Fig3a.png",
                "Guo2012.png")

#######################
### Competition

# Plant/invert
comp.plant.invert <- c("Corli2019_Fig3a.png", "Corli2019_Fig3b.png",
                       "Grass2018_Fig3a.png", "Grass2018_Fig3b.png",
                       "Grass2018_Fig3c.png")

# Invert
comp.invert <- c("Jackson2016_Fig4.png")

#######################
### Mutualism

# Plant/fungi
mut.plant.fungi <- c("Burkle2015_Fig1a.png", "Burkle2015_Fig1b.png",
                     "Burkle2015_Fig1c.png")

# Plant/invert
mut.plant.invert <- c("Martinez2015_Fig2a.png", "Martinez2015_Fig2b.png",
                      "Martinez2015_Fig2c.png")

#######################
### Predation

# Herp/fish
pred.herp.fish <- c("Kenison2016_Fig2a.png", "Kenison2016_Fig2b.png",
                     "Kenison2016_Fig2c.png", "Kenison2016_Fig2d.png",
                     "Kenison2016_Fig2e.png", "Kenison2016_Fig2f.png")

# Invert/herp
pred.invert.herp <-c("Marino2013_Fig1a.png")

#######################
# Categorize studies
es <- es %>%
  mutate(
    # Focal (first) organisms
    organism.first  = ifelse(filename %in% all_of(c(comp.plant,
                                                    mut.plant.fungi,
                                                    comp.plant.invert,
                                                    mut.plant.invert)),
                             "plant",
                             ifelse(filename %in% all_of(c(comp.invert,
                                                           pred.invert.herp)),
                                    "invert",
                                    ifelse(filename %in% all_of(c(pred.herp.fish)),
                                           "herp", NA))),
    # Additive (second) organisms
    organism.second = ifelse(filename %in% all_of(c(comp.plant)),
                             "plant",
                             ifelse(filename %in% all_of(c(mut.plant.fungi)),
                                    "fungi",
                                    ifelse(filename %in% all_of(c(comp.plant.invert,
                                                                  comp.invert,
                                                                  mut.plant.invert)),
                                           "invert",
                                           ifelse(filename %in% all_of(c(pred.invert.herp)),
                                                  "herp",
                                                  ifelse(filename %in% all_of(c(pred.herp.fish)),
                                                         "fish", NA))))),
    # Interaction
    interaction     = ifelse(filename %in% all_of(c(comp.plant,
                                                    comp.plant.invert,
                                                    comp.invert)),
                             "competition",
                             ifelse(filename %in% all_of(c(mut.plant.fungi,
                                                           mut.plant.invert)),
                                    "mutualism",
                                    ifelse(filename %in% all_of(c(pred.invert.herp,
                                                                  pred.herp.fish)),
                                           "predation", NA))))

# Effect sizes by first organism
ggplot(filter(es, abs(es) < 6),
       mapping = aes(x = es,
                     y = interaction,
                     color = organism.first)) +
  scale_color_manual(values = c("mediumslateblue", "black", "darkseagreen3")) +
  geom_vline(xintercept = 0,
             linetype = "dashed") +
  geom_pointrange(mapping = aes(xmin = es - es.var, xmax = es + es.var), 
                  position = position_jitter(width = 0.5),
                  linetype = "solid",
                  size = 0.2) +
  labs(y = "Interaction", x = "Effect size (Cohen's D)",
       color = "Organism (focal)") +
  pretty

# Effect sizes by second organism
ggplot(filter(es, abs(es) < 6),
       mapping = aes(x = es,
                     y = interaction,
                     color = organism.second)) +
  scale_color_manual(values = c("dodgerblue", "gold3", "mediumslateblue", "black", "darkseagreen3")) +
  geom_vline(xintercept = 0,
             linetype = "dashed") +
  geom_pointrange(mapping = aes(xmin = es - es.var, xmax = es + es.var), 
                  position = position_jitter(width = 0.5),
                  linetype = "solid",
                  size = 0.2) +
  labs(y = "Interaction", x = "Effect size (Cohen's D)",
       color = "Organism (interacting)") +
  pretty

# Effect sizes by organism
ggplot(filter(es, abs(es) < 6),
       mapping = aes(x = es,
                     y = organism.first,
                     color = interaction)) +
  scale_color_manual(values = c("dodgerblue", "gold3", "mediumslateblue", "black", "darkseagreen3")) +
  geom_vline(xintercept = 0,
             linetype = "dashed") +
  geom_pointrange(mapping = aes(xmin = es - es.var, xmax = es + es.var), 
                  position = position_jitter(width = 0.5),
                  linetype = "solid",
                  size = 0.2) +
  labs(y = "Organism (focal)", x = "Effect size (Cohen's D)",
       color = "Interaction") +
  pretty


## Do something
#res <- rma.uni(es, es.var, dat = es) 
#class(res)
#
## Forest plot?
#forest_plot <- forest(res)
#
## Manual plot?
#plot(working.es$es)

# Bad plot
#ggplot(filter(es, abs(es) < 6),
#       mapping = aes(x = es,
#                     y = interaction,
#                     color = interaction)) +
#    geom_point(size     = 1,
#               position = position_dodge(width = 0.5)) +
#    geom_errorbar(
#        aes(xmin = es - es.var,
#            xmax = es + es.var),
#        width    = 0.05,
#        linetype = "solid",
#        position = position_dodge(width = 0.5)) +
#    geom_vline(mapping = aes(xintercept = 0),
#               linetype = "solid",
#               color = "black")

```

